---
title: "Figures_script"

---
## Packages to install
```{r}

install.packages("rmarkdown")
install.packages("BiocManager")
BiocManager::install("circlize")
BiocManager::install("dittoSeq")
install.packages("forcats")
install.packages("readr")
install.packages("caret")
install.packages("doMC") 
#install_github("crazyhottommy/scclusteval") #this did not work. In Module 3, for the scclusteval package I used the code lines 18-20:
install.packages("devtools")

#usethis::create_github_token() #to generate a GitHub token
Sys.setenv(GITHUB_PAT = "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")
devtools::install_github("crazyhottommy/scclusteval")
BiocManager::install("ComplexHeatmap")
BiocManager::install("MAST")


```

## Install libraries
```{r}
library(forcats)
library(readr)
library(Seurat)
library(ggplot2)
library(dplyr)
library(pheatmap)
library(reshape2)
library(tibble)
library(dittoSeq)
library(readr)
library(scclusteval)
library(ComplexHeatmap)
library(MAST)
library(tidyr)
library(ggforce)
library(doMC)
```

# Fig 1 ####
```{r}
## B ####
data <- read_excel('Dataset_table.xlsx')
ggplot(data, aes(x = reorder(Dataset, -SST_cells), y = SST_cells, fill = Dataset == "DS17")) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = c("FALSE" = "gray", "TRUE" = "skyblue")) +
  geom_text(aes(label = SST_cells), vjust = -0.5, size = 3) +
  geom_hline(yintercept = 1000, linetype = "dashed", color = "red", size = 0.8) +
  labs(x = "Dataset number", y = "SST+ cells") +
  theme_minimal() +
  theme(legend.position = "none",
        axis.text.x = element_text(angle = 45, hjust = 1))
rm(data)

## C ####
Atlas_v0 <- readRDS('Atlas_v0.rds')
Atlas_v0$Major <- factor(Atlas_v0$major_cluster_label_trained_with_all, levels = sort(levels(factor(Atlas_v0$major_cluster_label_trained_with_all))))
Atlas_v0$Minor <- factor(Atlas_v0$cluster_label_trained_with_all, levels = sort(levels(factor(Atlas_v0$cluster_label_trained_with_all))))

DimPlot(Atlas_v0,group.by = 'sample', cols = c("#1096C5","#6158A6", '#F19B87','#9D1F2B'),
        pt.size=0.1, label = FALSE, alpha = 0.8)
DimPlot(Atlas_v0,group.by = 'Major', cols = c("#20B1BD", "#6C6DAF", "#DA5E6B"),
        pt.size=0.1, label = FALSE, alpha = 0.8)
DimPlot(Atlas_v0,group.by = 'Minor', cols = c('#5BBFAB','#33C0D4',"#1295C7","#1E2D59", '#2A2768','#644E9F','#8459A4','#BB5BA2','#EC697D','#F09F8A','#DF5E36','#9E1D20'),
        pt.size=0.1, label = FALSE, alpha = 0.8)


## D ####
Atlas_v0_Map <- subset(Atlas_v0, subset = subclass_bootstrapping_probability >= 0.9)
Atlas_v0_Map@meta.data <- Atlas_v0_Map@meta.data %>%
  mutate(Subclass = case_when(
    subclass_name %in% c("075 MEA-BST Lhx6 Nr2e1 Gaba", "048 RHP-COA Ndnf Gaba", "057 NDB-SI-MA-STRv Lhx8 Gaba", "055 STR Lhx8 Gaba", "054 STR Prox1 Lhx6 Gaba", 
                         "050 Lamp5 Lhx6 Gaba", "059 GPe-SI Sox6 Cyp26b1 Gaba", "076 MEA-BST Lhx6 Nfib Gaba", "051 Pvalb chandelier Gaba", "300 PARN-MDRNd-NTS Gbx2 Gly-Gaba", 
                         "042 OB-out Frmd7 Gaba", "046 Vip Gaba", "338 Lymphoid NN", "334 Microglia NN", "049 Lamp5 Gaba", "203 LGv-SPFp-SPFm Nkx2-2 Tcf7l2 Gaba",
                         "194 MRN-VTN-PPN Pax5 Cdh23 Gaba", "326 OPC NN", "335 BAM NN", "039 OB Meis2 Thsd7b Gaba", "287 MV-SPIV-PRP Dmbx1 Gly-Gaba", 
                         "327 Oligo NN", "319 Astro-TE NN", "285 MY Lhx1 Gly-Gaba", "036 HPF CR Glut", "201 PAG-RN Nkx2-2 Otx1 Gaba") ~ "Other",
    TRUE ~ subclass_name
  ))
DimPlot(Atlas_v0_Map,group.by = 'Subclass', cols = c("#059E74","#F0E443", '#E6A024','#5BB4E5', '#D36127'), pt.size=0.1, label = FALSE, alpha = 0.8)

FeaturePlot(Atlas_v0, "subclass_bootstrapping_probability") +   scale_color_gradient(low = "white", high = "#3953A4")


## E ####
metadata <- Atlas_v0@meta.data
SubclassPercentages <- metadata %>%
  mutate(Subclass = case_when(
    subclass_name %in% c("075 MEA-BST Lhx6 Nr2e1 Gaba", "048 RHP-COA Ndnf Gaba", "057 NDB-SI-MA-STRv Lhx8 Gaba", "055 STR Lhx8 Gaba", "054 STR Prox1 Lhx6 Gaba", 
                         "050 Lamp5 Lhx6 Gaba", "059 GPe-SI Sox6 Cyp26b1 Gaba", "076 MEA-BST Lhx6 Nfib Gaba", "051 Pvalb chandelier Gaba", "300 PARN-MDRNd-NTS Gbx2 Gly-Gaba", 
                         "042 OB-out Frmd7 Gaba", "046 Vip Gaba", "338 Lymphoid NN", "334 Microglia NN", "049 Lamp5 Gaba", "203 LGv-SPFp-SPFm Nkx2-2 Tcf7l2 Gaba",
                         "194 MRN-VTN-PPN Pax5 Cdh23 Gaba", "326 OPC NN", "335 BAM NN", "039 OB Meis2 Thsd7b Gaba", "287 MV-SPIV-PRP Dmbx1 Gly-Gaba", 
                         "327 Oligo NN", "319 Astro-TE NN", "285 MY Lhx1 Gly-Gaba", "036 HPF CR Glut", "201 PAG-RN Nkx2-2 Otx1 Gaba") ~ "Other",
    TRUE ~ subclass_name)) %>%
  group_by(Subclass) %>%
  summarise(count = n(), .groups = 'drop') %>%
  mutate(percentage = count / sum(count) * 100)
SubclassPercentages <- SubclassPercentages %>%
  mutate(Subclass = fct_reorder(Subclass, percentage, .desc = FALSE))

ggplot(SubclassPercentages, aes(x = 1, y = percentage, fill = Subclass)) +
  geom_bar(stat = "identity", position = "stack") +
  labs(title = "Percentage of Cells by Subclass", x = "Version", y = "Percentage (%)") +
  theme_minimal() +
  scale_fill_manual(values = c("grey","#F0E443", '#059E74','#5BB4E5', '#E6A024')) +
  theme(legend.position = "right")

rm(list = c('perc.sub','Atlas_v0_Map','metadata','SubclassPercentages'))
gc()

```

# Fig 2 ####
```{r}
## B ####
data <- read_csv('DS_anchors.csv')

ggplot(data, aes(x = reorder(dataset_id, -Anchors), y = Anchors)) +
  stat_summary_bin(aes(y = Anchors), fun = "mean", geom = "bar")+
  stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.2, color = "black")+
  geom_jitter()+
  geom_hline(yintercept = 2000, linetype = "dashed", color = "red")

## C ####
#it will generate a .pdf
data<-read.table("data.txt", row.names = 1, header = TRUE)

files <-dir()
files_anchors <- files[grep("Number_of_anchors_",files)]
files_kBET <- files[grep("kBET_batch_estimate_subset_bothDatasets_",files)]
files_kBET<-files_kBET[grep(".RDS",files_kBET)]

rejection_rate<-c()
mean_anchors <- c()
mean_anchors_scaled <- c()
for (i in rownames(data)){
  n_anchors <- read.table(paste0(files_anchors[grep(i,files_anchors)]), row.names = 1)
  mean_anchors <- c(mean_anchors,mean(n_anchors[,1]))
  mean_anchors_scaled <- c(mean_anchors_scaled,mean(n_anchors[,1])/data[i,2])
  kbet<-readRDS(paste0(files_kBET[grep(i,files_kBET)]))
  rejection_rate<-c(rejection_rate,median(kbet$stats$kBET.observed))
}
colors_ditto<-dittoColors()

files_knn <- files[grep("Percentages_of_cells_in_",files)]
files_knn<-files_knn[grep(".RDS",files_knn)]
pdf('plot_percentages_of_cells_with_mixed_neighborhood_in_kNN_allSamples.pdf', width=8)
par(cex=0.5,cex.lab=1.2)
i=1
knn_perc<-readRDS(paste0(files_knn[grep(rownames(data)[i],files_knn)]))
plot(seq(5,200,5),knn_perc, xlab="k",ylab="fraction of cells with mixed neighborhood", pch=0,cex=0.8,col=colors_ditto[i], ylim=c(0,1.05),xaxt='n',yaxt='n')
legend("top",legend=paste0(data[,1]," - ",rownames(data[,])),fill=colors_ditto,horiz = FALSE,cex=0.915,pch = c(0:12),ncol=6)
axis(2,cex.axis=1.5,)
axis(1,cex.axis=1.5)
print(lines(seq(5,200,5),knn_perc,col=colors_ditto[i]))
for (i in 2:nrow(data)){
  knn_perc<-readRDS(paste0(files_knn[grep(rownames(data)[i],files_knn)]))
  print(points(seq(5,200,5),knn_perc, pch=i-1,col=colors_ditto[i]))
  print(lines(seq(5,200,5),knn_perc,col=colors_ditto[i]))
}
abline(v=60,col="blue4",lty=2)
abline(h=0.45,col="red3",lty=2)
dev.off()

knn_perc_k60<-c()
for (i in 1:nrow(data)){
  knn_perc<-readRDS(paste0(files_knn[grep(rownames(data)[i],files_knn)]))
  knn_perc_k60<-c(knn_perc_k60,knn_perc["60"])
}

## D ####
#it will generate a .pdf
data_plot1<-data.frame(data, mean_anchors = mean_anchors, kBET_rejection_rate=rejection_rate, knn_perc_k60=knn_perc_k60)
pdf('Mean_ancors_vs_k60_fraction_of_cells.pdf')
plot(data_plot1$knn_perc_k60, data_plot1$mean_anchors, col=colors_ditto, pch=c(0:12),cex=1.5,lwd=2, xlab ="fraction of cells with mixed neighborhood", ylab="Mean anchors")
legend("topleft",legend=paste0(data[,1]," - ",rownames(data[,])),fill=colors_ditto,horiz = FALSE,cex=0.6,pch = c(0:12))
abline(v=0.45,col="red3",lty=2)
abline(h=2000,col="red3",lty=2)
dev.off()

## E ####
#it will generate a .pdf
data_plot1<-data.frame(data,mean_anchors = mean_anchors, mean_anchors_scaled = mean_anchors_scaled, kBET_rejection_rate=rejection_rate)
pdf('Mean_anchors_vs_kBET_RejectionRate.pdf')
plot(data_plot1$kBET_rejection_rate, data_plot1$mean_anchors, col=colors_ditto, pch=c(0:12),cex=1.5,lwd=2, xlab ="kBET rejection rate", ylab="Mean anchors")
legend("topright",legend=paste0(data[,1]," - ",rownames(data)),fill=colors_ditto,horiz = FALSE,cex=0.6,pch = c(0:12))
abline(v=0.5,col="red3",lty=2)
abline(h=2000,col="red3",lty=2)
dev.off()

rm(list = c("data", "data_plot1", "kbet", "n_anchors", "colors_ditto", 
            "files", "files_anchors", "files_kBET", "files_knn", "i", 
            "knn_perc", "knn_perc_k60", "mean_anchors", "mean_anchors_scaled", 
            "rejection_rate"))
gc()
```

# Fig 3 ####
```{r}
## A ####
data <- read_excel('Sequencing_depth.xlsx')
data <- data %>% mutate(Reads_in_Thousands = Reads / 1000)
means <- data %>% group_by(Sequencing) %>% summarise(Mean_Reads = mean(Reads_in_Thousands))
ggplot() +
  geom_bar(data = means, aes(x = Sequencing, y = Mean_Reads, fill = Sequencing),
           stat = "identity", width = 0.6, alpha = 0.8) +
  geom_point(data = data, aes(x = Sequencing, y = Reads_in_Thousands),
             color = "black", size = 3, position = position_jitter(width = 0.1, height = 0)) +
  scale_fill_manual(values = c("shallow" = "gray", "deep" = "darkgreen")) +
  labs(title = "Number of Reads per Cell",
       x = "",
       y = "Number of reads per cell (*10Â³)") +
  theme_minimal() +
  theme(legend.position = "none")

## C ####
#it will generate a .pdf
data<-read.table("data.txt", row.names = 1, header = TRUE)

files <-dir()
files_anchors <- files[grep("Number_of_anchors_",files)]
files_kBET <- files[grep("kBET_batch_estimate_subset_bothDatasets_",files)]
files_kBET<-files_kBET[grep(".RDS",files_kBET)]

rejection_rate<-c()
mean_anchors <- c()
mean_anchors_scaled <- c()
for (i in rownames(data)){
  n_anchors <- read.table(paste0(files_anchors[grep(i,files_anchors)]), row.names = 1)
  mean_anchors <- c(mean_anchors,mean(n_anchors[,1]))
  mean_anchors_scaled <- c(mean_anchors_scaled,mean(n_anchors[,1])/data[i,2])
  kbet<-readRDS(paste0(files_kBET[grep(i,files_kBET)]))
  rejection_rate<-c(rejection_rate,median(kbet$stats$kBET.observed))
}
colors_ditto<-dittoColors()

files_knn <- files[grep("Percentages_of_cells_in_",files)]
files_knn<-files_knn[grep(".RDS",files_knn)]
pdf('plot_percentages_of_cells_with_mixed_neighborhood_in_kNN_allSamples.pdf', width=8)
par(cex=0.5,cex.lab=1.2)
i=1
knn_perc<-readRDS(paste0(files_knn[grep(rownames(data)[i],files_knn)]))
plot(seq(5,200,5),knn_perc, xlab="k",ylab="fraction of cells with mixed neighborhood", pch=0,cex=0.8,col=colors_ditto[i], ylim=c(0,1.05),xaxt='n',yaxt='n')
legend("top",legend=paste0(data[,1]," - ",rownames(data[,])),fill=colors_ditto,horiz = FALSE,cex=0.915,pch = c(0:12),ncol=6)
axis(2,cex.axis=1.5,)
axis(1,cex.axis=1.5)
print(lines(seq(5,200,5),knn_perc,col=colors_ditto[i]))
for (i in 2:nrow(data)){
  knn_perc<-readRDS(paste0(files_knn[grep(rownames(data)[i],files_knn)]))
  print(points(seq(5,200,5),knn_perc, pch=i-1,col=colors_ditto[i]))
  print(lines(seq(5,200,5),knn_perc,col=colors_ditto[i]))
}
abline(v=60,col="blue4",lty=2)
abline(h=0.45,col="red3",lty=2)
dev.off()

knn_perc_k60<-c()
for (i in 1:nrow(data)){
  knn_perc<-readRDS(paste0(files_knn[grep(rownames(data)[i],files_knn)]))
  knn_perc_k60<-c(knn_perc_k60,knn_perc["60"])
}

## D ####
#it will generate a .pdf
data_plot1<-data.frame(data, mean_anchors = mean_anchors, kBET_rejection_rate=rejection_rate, knn_perc_k60=knn_perc_k60)
pdf('Mean_ancors_vs_k60_fraction_of_cells.pdf')
plot(data_plot1$knn_perc_k60, data_plot1$mean_anchors, col=colors_ditto, pch=c(0:12),cex=1.5,lwd=2, xlab ="fraction of cells with mixed neighborhood", ylab="Mean anchors")
legend("topleft",legend=paste0(data[,1]," - ",rownames(data[,])),fill=colors_ditto,horiz = FALSE,cex=0.6,pch = c(0:12))
abline(v=0.45,col="red3",lty=2)
abline(h=2000,col="red3",lty=2)
dev.off()

## E ####
#it will generate a .pdf
data_plot1<-data.frame(data,mean_anchors = mean_anchors, mean_anchors_scaled = mean_anchors_scaled, kBET_rejection_rate=rejection_rate)
pdf('Mean_anchors_vs_kBET_RejectionRate.pdf')
plot(data_plot1$kBET_rejection_rate, data_plot1$mean_anchors, col=colors_ditto, pch=c(0:12),cex=1.5,lwd=2, xlab ="kBET rejection rate", ylab="Mean anchors")
legend("topright",legend=paste0(data[,1]," - ",rownames(data)),fill=colors_ditto,horiz = FALSE,cex=0.6,pch = c(0:12))
abline(v=0.5,col="red3",lty=2)
abline(h=2000,col="red3",lty=2)
dev.off()


rm(list = c("data", "data_plot1", "kbet", "n_anchors", "means", "colors_ditto", 
            "files", "files_anchors", "files_kBET", "files_knn", "i", 
            "knn_perc", "knn_perc_k60", "mean_anchors", "mean_anchors_scaled", 
            "rejection_rate"))
gc()

```

# Fig 4 ####
```{r}
## A ####
#Open Atlas_v2
Atlas_v2 <- readRDS('L:/GBW-0086_LYLIM/Manuscripts/2025_Atlas_development/Datasets/Seurat_objects/Atlas old/Atlas_v2.rds')

DimPlot(Atlas_v2,group.by = 'sample', cols = c("grey",'grey','grey','grey', '#28BAA3','#C574AF','#F067A6','#2AB45D','#489DD7','#0077B6','#00B9DE'),
             pt.size=0.1, label = FALSE, alpha = 0.8)
DimPlot(Atlas_v2,group.by = 'major_label_transferAnchors_BaseAtlas', cols = c("#20B1BD", "#6C6DAF", "#DA5E6B"),
             pt.size=0.1, label = FALSE, alpha = 0.8)
DimPlot(Atlas_v2,group.by = 'minor_label_transferAnchors_BaseAtlas', cols = c('#5BBFAB','#33C0D4',"#1295C7","#1E2D59", '#2A2768','#644E9F','#8459A4','#BB5BA2','#EC697D','#F09F8A','#DF5E36','#9E1D20'),
             pt.size=0.1, label = FALSE, alpha = 0.8)

## B ####
#For accuracy calculation, source the file 'Compute_AccuracyMatrix_RF.R) and install Seurat v4
remotes::install_version('SeuratObject','4.1.4', repos = c('https://satijalab.r-universe.dev', getOption('repos')))

Idents(Atlas_v2) <- "major_label_transferAnchors_BaseAtlas"
DefaultAssay(Atlas_v2) <- "RNA"

majorClassMarkers <- FindAllMarkers(Atlas_v2, only.pos = TRUE, assay = 'RNA', logfc.threshold = log(1.5))
majorClassMarkers <- majorClassMarkers[which(majorClassMarkers$p_val_adj <= 0.01),]

Atlas_Matrix <- CreateSeuratObject(Atlas_v2@assays$RNA@counts[unique(majorClassMarkers$gene), drop=FALSE])
Atlas_Matrix <- AddMetaData(Atlas_Matrix, Atlas_v2@meta.data[colnames(Atlas_Matrix),])

accuracy_matrix <- ComputeClassificationAccuracy(Atlas_Matrix, "major_label_transferAnchors_BaseAtlas")

gplots::heatmap.2(accuracy_matrix, margins=c(7,7),
                  key = FALSE, # Explicitly set key to TRUE
                  keysize=1, key.xlab="", key.title="Accuracy", trace = "none", density.info = "none",
                  col = colorRampPalette(colors = c('black','#EE9A3A'))(1000), breaks = seq(0, 0.5, length.out = 1001), 
                  offsetRow=0.1, offsetCol=0.1, cexRow = 0.8, cexCol = 0.8, cellnote = accuracy_matrix, notecex = 0.5, notecol = 'white', Colv = F, Rowv = F,dendrogram = "none")


## C ####
Atlas_v2_Map <- subset(Atlas_v2, subset = subclass_bootstrapping_probability >= 0.9)
Atlas_v2_Map@meta.data <- Atlas_v2_Map@meta.data %>%
  mutate(Subclass = case_when(
    subclass_name %in% c("075 MEA-BST Lhx6 Nr2e1 Gaba", "048 RHP-COA Ndnf Gaba", 
                         "057 NDB-SI-MA-STRv Lhx8 Gaba", "054 STR Prox1 Lhx6 Gaba", "050 Lamp5 Lhx6 Gaba", 
                         "059 GPe-SI Sox6 Cyp26b1 Gaba", "076 MEA-BST Lhx6 Nfib Gaba", 
                         "055 STR Lhx8 Gaba", "042 OB-out Frmd7 Gaba", "046 Vip Gaba", 
                         "051 Pvalb chandelier Gaba", "338 Lymphoid NN", "049 Lamp5 Gaba", 
                         "326 OPC NN", "039 OB Meis2 Thsd7b Gaba", "287 MV-SPIV-PRP Dmbx1 Gly-Gaba", 
                         "065 IA Mgp Gaba", "038 DG-PIR Ex IMN", "036 HPF CR Glut", "073 MEA-BST Sox6 Gaba", 
                         "203 LGv-SPFp-SPFm Nkx2-2 Tcf7l2 Gaba", "103 PVHd-DMH Lhx6 Gaba", 
                         "044 OB Dopa-Gaba") ~ "Other",
    TRUE ~ subclass_name
  ))
DimPlot(Atlas_v2_Map,group.by = 'Subclass', cols = c("#059E74","#F0E443", '#E6A024','#5BB4E5', '#D36127'), pt.size=0.1, label = FALSE, alpha = 0.8)

## D ####
Atlas_v0$Version <- "v0"
Atlas_v2$Version <- "v2"
Atlas_Merged <- merge(Atlas_v0, Atlas_v2)
metadata <- Atlas_Merged@meta.data
SubclassPercentages <- metadata %>%
  mutate(Subclass = case_when(
    subclass_name %in% c("075 MEA-BST Lhx6 Nr2e1 Gaba", "048 RHP-COA Ndnf Gaba", "057 NDB-SI-MA-STRv Lhx8 Gaba", "054 STR Prox1 Lhx6 Gaba", "050 Lamp5 Lhx6 Gaba", 
                         "059 GPe-SI Sox6 Cyp26b1 Gaba", "076 MEA-BST Lhx6 Nfib Gaba", "055 STR Lhx8 Gaba", "042 OB-out Frmd7 Gaba", "046 Vip Gaba", 
                         "051 Pvalb chandelier Gaba", "338 Lymphoid NN", "049 Lamp5 Gaba", "326 OPC NN", "039 OB Meis2 Thsd7b Gaba", "287 MV-SPIV-PRP Dmbx1 Gly-Gaba","065 IA Mgp Gaba", "038 DG-PIR Ex IMN", "036 HPF CR Glut", "073 MEA-BST Sox6 Gaba", "203 LGv-SPFp-SPFm Nkx2-2 Tcf7l2 Gaba", "103 PVHd-DMH Lhx6 Gaba", 
                         "044 OB Dopa-Gaba") ~ "Other",
    TRUE ~ subclass_name)) %>%
  group_by(Version, Subclass) %>%
  summarise(count = n(), .groups = 'drop') %>%
  group_by(Version ) %>%
  mutate(percentage = count / sum(count) * 100)

ggplot(SubclassPercentages, aes(x = Version, y = percentage, fill = Subclass)) +
  geom_bar(stat = "identity", position = "stack") +
  labs(title = "Percentage of Cells by Subclass",
       x = "Version",
       y = "Percentage (%)") +
  theme_minimal() +
  scale_fill_manual(values = c("#059E74",'grey',"#F0E443", '#E6A024','#5BB4E5', 'grey', 'grey', 'grey', 'grey', 'grey', 'grey', 'grey'
                                                     , 'grey', 'grey', 'grey', 'grey', 'grey')) +
  theme(legend.position = "right")

## E ####
DimPlot(Atlas_v2, group.by = 'final_clusters_renamed', cols = c("#CCCCCC", "#989898",
                                                               "#2EBAD1", "#5BC2AE", "#60BD6E",
                                                               "#001d3d","#2a6f97", "#0d00a4", "#3C3CF0", "#3f8efc", "#449dd1", "#00a7e1", "#7fc8f8", "#81c3d7", "#1B1B75", "#023e8a", 
                                                               "#6E2DB7", "#d283ff", 
                                                               "#52b69a", "#007f5f", "#006400", "#6a994e", "#7cb518", "#a5be00", 
                                                               "#F0B572", "#e9c46a", 
                                                               "#A45F10", 
                                                               "#D14545", 
                                                               "#AC4758", 
                                                               "#ea698b", "#ffacc5", "#ff5d8f", "#c05299", "#ffa69e", "#f7cad0"),
        pt.size=0.1, label = FALSE, alpha = 0.8)


rm(list = c("Atlas_Matrix", "Atlas_Merged", "Atlas_v2_Map", "majorClassMarkers", 
            "metadata", "SubclassPercentages", "ComputeClassificationAccuracy"))
gc()
```

# Fig 5 ####
```{r}
#Open Atlas_v2 and subset only interneurons 
Atlas_v2_IN <- subset(Atlas_v2, subset = final_clusters_renamed %in% c("MC1.1", "MC1.2", "MC1.3",  "MC1.4",  "MC1.5",  "MC1.6",  "MC1.7",
                                                                       "MC1.8",  "MC1.9", "MC1.10", "MC1.11",  "MC2.1",  "MC2.2", "MC3.1", 
                                                                       "MC3.2" , "MC3.3",  "MC3.4" , "MC3.5" , "MC3.6",
                                                                       "nMC1.1","nMC1.2", "nMC2.0", "nMC3.0", "nMC4.0", 
                                                                       "nMC5.1", "nMC5.2", "nMC5.3", "nMC5.4" ,"nMC5.5" ,"nMC5.6"))

#Subset interneurons that have a high mapping probability to adult at the supertype level
Atlas_v2_IN_09 <- subset(Atlas_v2_IN, subset = supertype_bootstrapping_probability >= 0.9)

#Modify metadata for easier plotting and visualization
Atlas_v2_IN_09@meta.data <- Atlas_v2_IN_09@meta.data %>% mutate(SupertypeClean_Other = case_when(
    supertype_name %in% c("0135 HPF CR Glut_1", "0142 DG-PIR Ex IMN_3", "0144 OB Meis2 Thsd7b Gaba_2", "0160 OB-out Frmd7 Gaba_3", "0165 OB Dopa-Gaba_4", "0173 Vip Gaba_1", "0176 Vip Gaba_4",
                          "0177 Vip Gaba_5", "0181 Vip Gaba_9", "0187 Sncg Gaba_3",
                          "0193 RHP-COA Ndnf Gaba_1", "0194 RHP-COA Ndnf Gaba_2", "0195 RHP-COA Ndnf Gaba_3", "0198 RHP-COA Ndnf Gaba_6", "0199 Lamp5 Gaba_1", "0200 Lamp5 Gaba_2",
                          "0201 Lamp5 Gaba_3","0203 Lamp5 Lhx6 Gaba_1", "0204 Pvalb chandelier Gaba_1","0233 STR Prox1 Lhx6 Gaba_1", "0234 STR Prox1 Lhx6 Gaba_2",
                          "0235 STR Prox1 Lhx6 Gaba_3", "0236 STR Lhx8 Gaba_1","0237 STR Lhx8 Gaba_2", "0238 Sst Chodl Gaba_1", "0239 Sst Chodl Gaba_2","0240 Sst Chodl Gaba_3",
                          "0241 Sst Chodl Gaba_4","0242 Sst Chodl Gaba_5", "0243 Sst Chodl Gaba_6", "0244 NDB-SI-MA-STRv Lhx8 Gaba_1", "0253 NDB-SI-MA-STRv Lhx8 Gaba_10", "0258 GPe-SI Sox6 Cyp26b1 Gaba_2",
                          "0294 IA Mgp Gaba_5", "0341 MEA-BST Sox6 Gaba_5", "0355 MEA-BST Lhx6 Nr2e1 Gaba_2", "0356 MEA-BST Lhx6 Nr2e1 Gaba_3","0357 MEA-BST Lhx6 Nfib Gaba_1", 
                          "0358 MEA-BST Lhx6 Nfib Gaba_2", "0362 MEA-BST Lhx6 Nfib Gaba_6", "0476 PVHd-DMH Lhx6 Gaba_3", "0799 MRN-VTN-PPN Pax5 Cdh23 Gaba_2","1056 LDT-PCG-CS Gata3 Lhx1 Gaba_1",
                          "1091 MY Lhx1 Gly-Gaba_3", "1180 OPC NN_2","1195 BAM NN_1", "1198 B cells NN_1") ~ "Other",
    TRUE ~ supertype_name))

#Combine Major and SUpertype to plot only the mappings from MC and nMC independently
Atlas_v2_IN_09$SupertypeCleanMajorCombined <- paste(
  Atlas_v2_IN_09$major_label_transferAnchors_BaseAtlas, 
  Atlas_v2_IN_09$SupertypeClean_Other, 
  sep = "_"
)
MC <- c("Martinotti_0168 OB-STR-CTX Inh IMN_3", "Martinotti_0171 OB-STR-CTX Inh IMN_6",  "Martinotti_0170 OB-STR-CTX Inh IMN_5", "Martinotti_0215 Sst Gaba_2",  "Martinotti_0230 Sst Gaba_17", "Martinotti_0228 Sst Gaba_15",  "Martinotti_0225 Sst Gaba_12", "Martinotti_Other",  "Martinotti_0223 Sst Gaba_10", "Martinotti_0231 Sst Gaba_18",  "Martinotti_0219 Sst Gaba_6", "Martinotti_0217 Sst Gaba_4",  "Martinotti_0224 Sst Gaba_11", "Martinotti_0216 Sst Gaba_3",  "Martinotti_0226 Sst Gaba_13", "Martinotti_0172 OB-STR-CTX Inh IMN_7",  "Martinotti_0232 Sst Gaba_19", "Martinotti_0229 Sst Gaba_16",  "Martinotti_0214 Sst Gaba_1", "Martinotti_0222 Sst Gaba_9",  "Martinotti_0218 Sst Gaba_5", "Martinotti_0220 Sst Gaba_7",  "Martinotti_0221 Sst Gaba_8", "Martinotti_0227 Sst Gaba_14",  "Martinotti_0210 Pvalb Gaba_6", "Martinotti_0209 Pvalb Gaba_5",  "Martinotti_0208 Pvalb Gaba_4", "Martinotti_0167 OB-STR-CTX Inh IMN_2")
nMC <- c( "Non-Martinotti_0210 Pvalb Gaba_6", "Non-Martinotti_0171 OB-STR-CTX Inh IMN_6","Non-Martinotti_0168 OB-STR-CTX Inh IMN_3", "Non-Martinotti_0170 OB-STR-CTX Inh IMN_5","Non-Martinotti_0231 Sst Gaba_18", "Non-Martinotti_0230 Sst Gaba_17",          "Non-Martinotti_0215 Sst Gaba_2", "Non-Martinotti_0206 Pvalb Gaba_2",          "Non-Martinotti_0228 Sst Gaba_15", "Non-Martinotti_0225 Sst Gaba_12",          "Non-Martinotti_0214 Sst Gaba_1", "Non-Martinotti_0211 Pvalb Gaba_7",          "Non-Martinotti_0209 Pvalb Gaba_5", "Non-Martinotti_Other",          "Non-Martinotti_0229 Sst Gaba_16", "Non-Martinotti_0226 Sst Gaba_13",          "Non-Martinotti_0208 Pvalb Gaba_4", "Non-Martinotti_0216 Sst Gaba_3",          "Non-Martinotti_0221 Sst Gaba_8", "Non-Martinotti_0224 Sst Gaba_11",          "Non-Martinotti_0223 Sst Gaba_10", "Non-Martinotti_0219 Sst Gaba_6",          "Non-Martinotti_0227 Sst Gaba_14", "Non-Martinotti_0218 Sst Gaba_5",          "Non-Martinotti_0169 OB-STR-CTX Inh IMN_4", "Non-Martinotti_0232 Sst Gaba_19",          "Non-Martinotti_0217 Sst Gaba_4", "Non-Martinotti_0205 Pvalb Gaba_1",          "Non-Martinotti_0212 Pvalb Gaba_8", "Non-Martinotti_0207 Pvalb Gaba_3",          "Non-Martinotti_0213 Pvalb Gaba_9", "Non-Martinotti_0220 Sst Gaba_7",          "Non-Martinotti_0166 OB-STR-CTX Inh IMN_1", "Non-Martinotti_0222 Sst Gaba_9")

## A ####
#UMAPs of MC main group and final Martinotti developmental clusters 
dittoDimPlot(Atlas_v2_IN, "major_label_transferAnchors_BaseAtlas", reduction.use = "umap", color.panel = c("#8D8DC1", "grey"), legend.show = FALSE, opacity=0.7,size = 0.5)
dittoDimPlot(Atlas_v2_IN, "final_clusters_renamed", reduction.use = "umap",
             color.panel = c("#001d3d","#2a6f97", "#0d00a4", "#3C3CF0", "#3f8efc", "#449dd1", "#00a7e1", "#7fc8f8", "#81c3d7", "#1B1B75", "#023e8a", "#6E2DB7", "#d283ff", "#52b69a", "#007f5f", "#006400", "#6a994e", "#7cb518", "#a5be00", "grey","grey", "grey", "grey", "grey", "grey", "grey", "grey", "grey", "grey", "grey"),
                   legend.show = FALSE,opacity=0.7,size = 0.5)

#UMAP of adult mapped supertypes
Atlas_v2_IN_09@meta.data <- Atlas_v2_IN_09@meta.data %>% mutate(SupertypeCleanMajorCombined_MC = case_when( SupertypeCleanMajorCombined %in% nMC ~ "non-Martinotti", TRUE ~ SupertypeCleanMajorCombined))
dittoDimPlot(Atlas_v2_IN_09, "SupertypeCleanMajorCombined_MC", reduction.use = "umap", 
                   color.panel = c("#059E74","#059E74", "#059E74", "#059E74", "#059E74",
                                   "#F0E443", "#F0E443", "#F0E443",
                                   "#81c3d7", "#3C3CF0", "#3f8efc", "#001d3d", "#00a7e1", "#6a994e", "#A01061", "#B23C25", "#2a6f97", "#6E2DB7", "#52b69a",
                                   "#ffa69e", "#5111B2", "#B58C00", "#d283ff", "#2E004B", "#A45F10", "#007f5f", "#006400",
                                   "grey50", "grey"),
             legend.show = FALSE, opacity=0.7,size = 0.5)

## B ####
'%!in%' <- function(x,y)!('%in%'(x,y))

martinotti<-subset(Atlas_v2, subset= major_label_transferAnchors_BaseAtlas %in% "Martinotti" & subclass_name %in% "053 Sst Gaba" & supertype_bootstrapping_probability > 0.9)
data<-martinotti@meta.data[,c("final_clusters_renamed","supertype_name","final_clusters_palette")]
data$supertype_name <- paste("Sst_",unlist(lapply(as.character(data$supertype_name), function(x) strsplit(x," ")[[1]][3])),sep="")
### add blank layer
data2<-data.frame(final_clusters_renamed=rep("MC0.0",2000),supertype_name=rep("Sst_Gaba_0",2000),final_clusters_palette=rep("gray",2000))
data<-rbind(data,data2)
data$final_clusters_renamed <- factor(data$final_clusters_renamed, levels = c("MC2.1","MC1.1","MC3.2","MC3.3","MC3.4","MC1.5","MC1.7","MC1.9","MC0.0","MC1.2","MC3.1","MC1.4","MC3.5","MC2.2","MC1.3","MC3.6","MC1.6","MC1.8","MC1.10","MC1.11"))
data$supertype_name <- factor(data$supertype_name, levels = c("Sst_Gaba_10","Sst_Gaba_4","Sst_Gaba_18","Sst_Gaba_19","Sst_Gaba_6","Sst_Gaba_3","Sst_Gaba_5","Sst_Gaba_1","Sst_Gaba_0","Sst_Gaba_9","Sst_Gaba_11","Sst_Gaba_16","Sst_Gaba_15","Sst_Gaba_13","Sst_Gaba_2","Sst_Gaba_12","Sst_Gaba_7","Sst_Gaba_8","Sst_Gaba_17","Sst_Gaba_14"))
edges  <- data.frame(data$final_clusters_renamed, data$supertype_name)
edges<-edges %>% group_by(across(everything())) %>% count()
colnames(edges) <- c("Cluster", "Supertype","Freq")  
edges <- as.data.frame(edges)
### set colors
tmp<-data[!duplicated(data$final_clusters_renamed),]
color1<-tmp$final_clusters_palette
names(color1)<-tmp$final_clusters_renamed
### create plot table
edges2<-edges
dat_ggforce <- edges2  %>%
  gather_set_data(1:2) %>%        # <- ggforce helper function
  arrange(x,Cluster,desc(Supertype))
## make plot
p<-ggplot(dat_ggforce, aes(x = x, id = id, split = y, value = Freq)) +
  geom_parallel_sets(aes(fill = Cluster), alpha = 0.45, axis.width = 0.2, n=100, strength = 0.5) +
  geom_parallel_sets_axes(axis.width = 0.05, fill = "white",color = "white", size = 0.15) +
  geom_parallel_sets_labels(colour = 'black', size = 3, angle = 0) +
  scale_fill_manual(values  = color1) + 
  theme(panel.background = element_rect(fill='white', colour='black')) + theme(legend.position="none") + theme(axis.title.x=element_blank(),axis.text.x=element_blank(),axis.ticks.x=element_blank(),axis.title.y=element_blank(),axis.text.y=element_blank(),axis.ticks.y=element_blank())
### modify blank layer to make it invisible
q<-ggplot_build(p)
tmp<-q$data[[3]]
tmp<-tmp[which(tmp$label %!in% c("Sst_Gaba_0","MC0.0")),]
q$data[[3]] <- tmp
tmp<-q$data[[1]]
tmp[which(tmp$split == "MC0.0"),"alpha"] <-0
q$data[[1]] <-tmp
q <- ggplot_gtable(q)
plot(q)

## C ####
##UMAPs of nMC main group and final non-Martinotti developmental clusters 
dittoDimPlot(Atlas_v2_IN, "major_label_transferAnchors_BaseAtlas", reduction.use = "umap", color.panel = c("grey", "#E0767F"), legend.show = FALSE, opacity=0.7,size = 0.5)
dittoDimPlot(Atlas_v2_IN, "final_clusters_renamed", reduction.use = "umap",
             color.panel = c("grey","grey","grey","grey","grey","grey","grey","grey","grey","grey","grey", "grey","grey", "grey","grey","grey","grey","grey","grey",
                             "#F0B572", "#e9c46a", "#A45F10", "#D14545", "#AC4758", "#ea698b", "#ffacc5", "#ff5d8f", "#c05299", "#ffa69e", "#f7cad0"),
             legend.show = FALSE, opacity=0.7,size = 0.5)

#UMAP of adult mapped supertypes
Atlas_v2_IN_09@meta.data <- Atlas_v2_IN_09@meta.data %>% mutate(SupertypeCleanMajorCombined_nMC = case_when(SupertypeCleanMajorCombined %in% MC ~ "Martinotti", TRUE ~ SupertypeCleanMajorCombined))
dittoDimPlot(Atlas_v2_IN_09, "SupertypeCleanMajorCombined_nMC", reduction.use = "umap",
             color.panel = c("grey", "grey50", "grey50", "grey50", "grey50", "grey50",
                             "#F0E443", "#F0E443", "#F0E443","#F0E443", "#AC4758", "#F0E443","#F0E443", "#F0E443", "#F0E443",
                             "#EA698B","#FFACC9", "#D14545","#FFACC1", "#FF5D8F", "#AC4758", "#A01061", "#F0B572",
                             "#2a6f97", "#F0B411","#52b69a","#ffa69e","#FFDDD5","#FFACC5","#D15545", "#F1C572",
                             "#A45F10", "#AC4AA8", "#AC4AA9", "grey50"),
             legend.show = TRUE, opacity=0.7,size = 0.5)
## D ####
'%!in%' <- function(x,y)!('%in%'(x,y))

non_martinotti<-subset(Atlas_v2, subset=major_label_transferAnchors_BaseAtlas %in% "Non-Martinotti" & subclass_name %in% "053 Sst Gaba" & supertype_bootstrapping_probability > 0.9)
non_martinotti<-subset(non_martinotti, subset=supertype_name %in% c( "0214 Sst Gaba_1","0215 Sst Gaba_2","0216 Sst Gaba_3","0217 Sst Gaba_4","0218 Sst Gaba_5","0219 Sst Gaba_6","0221 Sst Gaba_8","0223 Sst Gaba_10","0225 Sst Gaba_12","0226 Sst Gaba_13","0227 Sst Gaba_14","0228 Sst Gaba_15","0229 Sst Gaba_16","0230 Sst Gaba_17","0231 Sst Gaba_18","0232 Sst Gaba_19" ))
data<-non_martinotti@meta.data[,c("final_clusters_renamed","supertype_name","final_clusters_palette")]
data$supertype_name <- paste("Sst_",unlist(lapply(as.character(data$supertype_name), function(x) strsplit(x," ")[[1]][3])),sep="")
### add blank layer
data2<-data.frame(final_clusters_renamed=rep("nMC0.0",2000),supertype_name=rep("Sst_Gaba_0",1000),final_clusters_palette=rep("white",2000))
data<-rbind(data,data2)
data$final_clusters_renamed <- factor(data$final_clusters_renamed, levels = c("nMC1.1","nMC4.0","nMC3.0","nMC5.2","nMC5.1","nMC5.3","nMC5.4","nMC5.6","nMC1.2","nMC0.0","nMC2.0","nMC5.5"))
data$supertype_name <- factor(data$supertype_name, levels = c("Sst_Gaba_14","Sst_Gaba_13","Sst_Gaba_1","Sst_Gaba_5","Sst_Gaba_8","Sst_Gaba_10","Sst_Gaba_16","Sst_Gaba_6","Sst_Gaba_18","Sst_Gaba_19","Sst_Gaba_2","Sst_Gaba_4","Sst_Gaba_3","Sst_Gaba_15","Sst_Gaba_0","Sst_Gaba_17","Sst_Gaba_12"))
edges  <- data.frame(data$final_clusters_renamed, data$supertype_name)
edges<-edges %>% group_by(across(everything())) %>% count()
colnames(edges) <- c("Cluster", "Supertype","Freq")  
edges <- as.data.frame(edges)
### set colors
tmp<-data[!duplicated(data$final_clusters_renamed),]
color1<-tmp$final_clusters_palette
names(color1)<-tmp$final_clusters_renamed
### create table for the plot
edges2<-edges
dat_ggforce <- edges2  %>%
  gather_set_data(1:2) %>%        # <- ggforce helper function
  arrange(x,Cluster,desc(Supertype))
### make plot
p<-ggplot(dat_ggforce, aes(x = x, id = id, split = y, value = Freq)) +
  geom_parallel_sets(aes(fill = Cluster), alpha = 0.45, axis.width = 0.2,
                     n=100, strength = 0.5) +
  geom_parallel_sets_axes(axis.width = 0.05, fill = "white",
                          color = "white", size = 0.15) +
  geom_parallel_sets_labels(colour = 'black', size = 3, angle = 0) +
  scale_fill_manual(values  = color1) + 
  theme(panel.background = element_rect(fill='white', colour='black')) + theme(legend.position="none") + theme(axis.title.x=element_blank(),axis.text.x=element_blank(),axis.ticks.x=element_blank(),axis.title.y=element_blank(),axis.text.y=element_blank(),axis.ticks.y=element_blank())
### modify blank layer to make it invisible
q<-ggplot_build(p)
tmp<-q$data[[3]]
tmp<-tmp[which(tmp$label %!in% c("Sst_Gaba_0","nMC0.0")),]
q$data[[3]] <- tmp
tmp<-q$data[[1]]
tmp[which(tmp$split == "nMC0.0"),"alpha"] <- 0
q$data[[1]] <-tmp
q <- ggplot_gtable(q)
plot(q)

## E ####
metadatav2 <- Atlas_v2@meta.data
metadatav2 <- metadatav2 %>% mutate(Supertype_Other = case_when( supertype_name %in% c("0135 HPF CR Glut_1", "0142 DG-PIR Ex IMN_3", "0144 OB Meis2 Thsd7b Gaba_2", "0160 OB-out Frmd7 Gaba_3", "0165 OB Dopa-Gaba_4", "0166 OB-STR-CTX Inh IMN_1", "0167 OB-STR-CTX Inh IMN_2", "0168 OB-STR-CTX Inh IMN_3", "0169 OB-STR-CTX Inh IMN_4", "0170 OB-STR-CTX Inh IMN_5", "0171 OB-STR-CTX Inh IMN_6", "0172 OB-STR-CTX Inh IMN_7", "0173 Vip Gaba_1", "0174 Vip Gaba_2", "0176 Vip Gaba_4", "0177 Vip Gaba_5", "0181 Vip Gaba_9", "0187 Sncg Gaba_3", "0193 RHP-COA Ndnf Gaba_1", "0194 RHP-COA Ndnf Gaba_2", "0195 RHP-COA Ndnf Gaba_3", "0196 RHP-COA Ndnf Gaba_4", "0197 RHP-COA Ndnf Gaba_5", "0198 RHP-COA Ndnf Gaba_6", "0199 Lamp5 Gaba_1", "0200 Lamp5 Gaba_2", "0201 Lamp5 Gaba_3", "0202 Lamp5 Gaba_4", "0203 Lamp5 Lhx6 Gaba_1", "0204 Pvalb chandelier Gaba_1", "0205 Pvalb Gaba_1", "0206 Pvalb Gaba_2", "0207 Pvalb Gaba_3", "0208 Pvalb Gaba_4", "0209 Pvalb Gaba_5", "0210 Pvalb Gaba_6", "0211 Pvalb Gaba_7", "0212 Pvalb Gaba_8", "0213 Pvalb Gaba_9", "0233 STR Prox1 Lhx6 Gaba_1", "0234 STR Prox1 Lhx6 Gaba_2", "0235 STR Prox1 Lhx6 Gaba_3", "0236 STR Lhx8 Gaba_1", "0237 STR Lhx8 Gaba_2", "0238 Sst Chodl Gaba_1", "0239 Sst Chodl Gaba_2", "0240 Sst Chodl Gaba_3", "0241 Sst Chodl Gaba_4", "0242 Sst Chodl Gaba_5", "0243 Sst Chodl Gaba_6", "0244 NDB-SI-MA-STRv Lhx8 Gaba_1", "0251 NDB-SI-MA-STRv Lhx8 Gaba_8", "0253 NDB-SI-MA-STRv Lhx8 Gaba_10", "0258 GPe-SI Sox6 Cyp26b1 Gaba_2", "0294 IA Mgp Gaba_5", "0339 MEA-BST Sox6 Gaba_3", "0341 MEA-BST Sox6 Gaba_5", "0355 MEA-BST Lhx6 Nr2e1 Gaba_2", "0356 MEA-BST Lhx6 Nr2e1 Gaba_3", "0357 MEA-BST Lhx6 Nfib Gaba_1", "0358 MEA-BST Lhx6 Nfib Gaba_2",  "0361 MEA-BST Lhx6 Nfib Gaba_5", "0362 MEA-BST Lhx6 Nfib Gaba_6", "0476 PVHd-DMH Lhx6 Gaba_3", "0799 MRN-VTN-PPN Pax5 Cdh23 Gaba_2", "0835 LGv-SPFp-SPFm Nkx2-2 Tcf7l2 Gaba_2", "0836 LGv-SPFp-SPFm Nkx2-2 Tcf7l2 Gaba_3",  "0846 SC-PAG Lef1 Emx2 Gaba_4", "1056 LDT-PCG-CS Gata3 Lhx1 Gaba_1", "1091 MY Lhx1 Gly-Gaba_3", "1099 MV-SPIV-PRP Dmbx1 Gly-Gaba_1", "1120 PARN-MDRNd-NTS Gbx2 Gly-Gaba_1", "1180 OPC NN_2", "1195 BAM NN_1",  "1198 B cells NN_1")
                                                                 ~ "Other", TRUE ~ supertype_name)) %>%
  filter(!final_clusters_renamed %in% c("HIn1", "HIn2", "LRP1.0", "LRP2.1", "LRP2.2")) %>% filter(supertype_bootstrapping_probability >= 0.9) %>% filter(Supertype_Other != "Other")

summary_df <- metadatav2 %>% group_by(Supertype_Other, major_label_transferAnchors_BaseAtlas) %>% summarise(cell_count = n(), .groups = 'drop') %>% group_by(Supertype_Other) %>% mutate(proportion = cell_count / sum(cell_count)) %>% ungroup()
supertype_order <- summary_df %>% filter(major_label_transferAnchors_BaseAtlas == "Martinotti") %>% arrange(desc(proportion)) %>% pull(Supertype_Other)
summary_df$Supertype_Other <- factor(summary_df$Supertype_Other, levels = supertype_order)

#Bar plot
ggplot(summary_df, aes(x = Supertype_Other, y = proportion, fill = major_label_transferAnchors_BaseAtlas)) +
  geom_bar(stat = "identity", position = "stack") +
  scale_fill_manual(values = c("Martinotti" = "#8E8EC1", "Non-Martinotti" = "#E0777F"))+
  labs(title = "Distribution of Supertypes by Major Group (MC and nMC)",
       x = "Supertype_Other",
       y = "Proportion of Cells") +
  scale_y_continuous(breaks = seq(0, 1, by = 0.2)) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))

#Dot plot
ggplot(summary_df, aes(x = Supertype_Other, y = major_label_transferAnchors_BaseAtlas)) +
  geom_point(aes(size = cell_count, color = proportion)) +
  scale_color_gradient(low = "grey", high = "red")+
  labs(x = "Adult Supertypes", y = "Main types", color = "Proportion", size = "Cell Count") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))


## F ####
#Adjust metadata for easier handling and plotting
Yao <- readRDS('Yao_2023_Sst_Gaba_053_056_data.RDS')

metadataYao <- Yao@meta.data
metadatav2 <- Atlas_v2@meta.data
metadatav2 <- metadatav2 %>% filter(!final_clusters_renamed %in% c("HIn1", "HIn2", "LRP1.0", "LRP2.1", "LRP2.2"))
metadatav2 <- metadatav2 %>% mutate(Supertype_Other = case_when( supertype_name %in% c("0135 HPF CR Glut_1", "0142 DG-PIR Ex IMN_3", "0144 OB Meis2 Thsd7b Gaba_2", "0160 OB-out Frmd7 Gaba_3", "0165 OB Dopa-Gaba_4", "0166 OB-STR-CTX Inh IMN_1", "0167 OB-STR-CTX Inh IMN_2", "0168 OB-STR-CTX Inh IMN_3", "0169 OB-STR-CTX Inh IMN_4", "0170 OB-STR-CTX Inh IMN_5", "0171 OB-STR-CTX Inh IMN_6", "0172 OB-STR-CTX Inh IMN_7", "0173 Vip Gaba_1", "0174 Vip Gaba_2", "0176 Vip Gaba_4", "0177 Vip Gaba_5", "0181 Vip Gaba_9", "0187 Sncg Gaba_3", "0193 RHP-COA Ndnf Gaba_1", "0194 RHP-COA Ndnf Gaba_2", "0195 RHP-COA Ndnf Gaba_3", "0196 RHP-COA Ndnf Gaba_4", "0197 RHP-COA Ndnf Gaba_5", "0198 RHP-COA Ndnf Gaba_6", "0199 Lamp5 Gaba_1", "0200 Lamp5 Gaba_2", "0201 Lamp5 Gaba_3", "0202 Lamp5 Gaba_4", "0203 Lamp5 Lhx6 Gaba_1", "0204 Pvalb chandelier Gaba_1", "0205 Pvalb Gaba_1",  "0206 Pvalb Gaba_2", "0207 Pvalb Gaba_3", "0208 Pvalb Gaba_4", "0209 Pvalb Gaba_5", "0210 Pvalb Gaba_6", "0211 Pvalb Gaba_7", "0212 Pvalb Gaba_8", "0213 Pvalb Gaba_9", "0233 STR Prox1 Lhx6 Gaba_1", "0234 STR Prox1 Lhx6 Gaba_2", "0235 STR Prox1 Lhx6 Gaba_3", "0236 STR Lhx8 Gaba_1", "0237 STR Lhx8 Gaba_2", "0238 Sst Chodl Gaba_1", "0239 Sst Chodl Gaba_2", "0240 Sst Chodl Gaba_3", "0241 Sst Chodl Gaba_4", "0242 Sst Chodl Gaba_5", "0243 Sst Chodl Gaba_6", "0244 NDB-SI-MA-STRv Lhx8 Gaba_1", "0251 NDB-SI-MA-STRv Lhx8 Gaba_8", "0253 NDB-SI-MA-STRv Lhx8 Gaba_10", "0258 GPe-SI Sox6 Cyp26b1 Gaba_2", "0294 IA Mgp Gaba_5", "0339 MEA-BST Sox6 Gaba_3", "0341 MEA-BST Sox6 Gaba_5", "0355 MEA-BST Lhx6 Nr2e1 Gaba_2", "0356 MEA-BST Lhx6 Nr2e1 Gaba_3", "0357 MEA-BST Lhx6 Nfib Gaba_1", "0358 MEA-BST Lhx6 Nfib Gaba_2", "0361 MEA-BST Lhx6 Nfib Gaba_5", "0362 MEA-BST Lhx6 Nfib Gaba_6", "0476 PVHd-DMH Lhx6 Gaba_3", "0799 MRN-VTN-PPN Pax5 Cdh23 Gaba_2", "0835 LGv-SPFp-SPFm Nkx2-2 Tcf7l2 Gaba_2", "0836 LGv-SPFp-SPFm Nkx2-2 Tcf7l2 Gaba_3", "0846 SC-PAG Lef1 Emx2 Gaba_4", "1056 LDT-PCG-CS Gata3 Lhx1 Gaba_1", "1091 MY Lhx1 Gly-Gaba_3", "1099 MV-SPIV-PRP Dmbx1 Gly-Gaba_1", "1120 PARN-MDRNd-NTS Gbx2 Gly-Gaba_1", "1180 OPC NN_2", "1195 BAM NN_1", "1198 B cells NN_1")
                          ~ "Other", TRUE ~ supertype_name))

metadatav2 <- metadatav2 %>% filter(Supertype_Other != "Other") %>% mutate(supertype_category = ifelse(supertype_bootstrapping_probability >= 0.9, Supertype_Other, "Not assigned")) 

plot_data_v2 <- metadatav2 %>% group_by(Stage, Supertype_Other) %>% summarise(count = n()) %>% mutate(proportion = count / sum(count)) %>% ungroup() %>% filter(!Stage == "E16")
plot_data_Yao <- metadataYao %>% rename(Supertype_Other = supertype) %>% mutate(Supertype_Other = case_when(Supertype_Other == "Sst Gaba_1"  ~ "0214 Sst Gaba_1", Supertype_Other == "Sst Gaba_2"  ~ "0215 Sst Gaba_2", Supertype_Other == "Sst Gaba_3"  ~ "0216 Sst Gaba_3", Supertype_Other == "Sst Gaba_4"  ~ "0217 Sst Gaba_4", Supertype_Other == "Sst Gaba_5"  ~ "0218 Sst Gaba_5", Supertype_Other == "Sst Gaba_6"  ~ "0219 Sst Gaba_6",Supertype_Other == "Sst Gaba_7"  ~ "0220 Sst Gaba_7", Supertype_Other == "Sst Gaba_8"  ~ "0221 Sst Gaba_8", Supertype_Other == "Sst Gaba_9"  ~ "0222 Sst Gaba_9", Supertype_Other == "Sst Gaba_10" ~ "0223 Sst Gaba_10", Supertype_Other == "Sst Gaba_11" ~ "0224 Sst Gaba_11", Supertype_Other == "Sst Gaba_12" ~ "0225 Sst Gaba_12", Supertype_Other == "Sst Gaba_13" ~ "0226 Sst Gaba_13", Supertype_Other == "Sst Gaba_14" ~ "0227 Sst Gaba_14",Supertype_Other == "Sst Gaba_15" ~ "0228 Sst Gaba_15", Supertype_Other == "Sst Gaba_16" ~ "0229 Sst Gaba_16", Supertype_Other == "Sst Gaba_17" ~ "0230 Sst Gaba_17",Supertype_Other == "Sst Gaba_18" ~ "0231 Sst Gaba_18",Supertype_Other == "Sst Gaba_19" ~ "0232 Sst Gaba_19",
                                                                                                            TRUE ~ Supertype_Other)) %>%
filter(!Supertype_Other %in% c("Sst Chodl Gaba_1", "Sst Chodl Gaba_2", "Sst Chodl Gaba_3", "Sst Chodl Gaba_4", "Sst Chodl Gaba_5", "Sst Chodl Gaba_6")) %>% mutate(Stage = "P56") %>% group_by(Stage, Supertype_Other) %>% summarise(count = n()) %>% mutate(proportion = count / sum(count)) %>% ungroup()
plot_data <- bind_rows(plot_data_v2, plot_data_Yao)

p1_proportions <- plot_data_v2 %>% filter(Stage == "P1") %>% select(Supertype_Other, proportion) %>% rename(p1_proportion = proportion)
plot_data <- plot_data %>% left_join(p1_proportions, by = "Supertype_Other") %>% mutate(relative_proportion_toP1 = proportion / p1_proportion)

plot_data <- plot_data %>% filter(!Stage == "P1")%>% mutate(Supertype_Other = factor(Supertype_Other, levels = rev(sort(unique(Supertype_Other), decreasing = FALSE, na.last = TRUE)))) %>% 
  mutate(group = factor(Stage, levels = rev(sort(unique(Stage), decreasing = FALSE, na.last = TRUE))))

#Bar plot
ggplot(plot_data, aes(x = Supertype_Other, y = relative_proportion_toP1,  fill = Stage)) +
  geom_bar(stat = "identity", position = position_dodge()) +
  labs( title = "Proportion of Supertypes by Group", x = "Supertype", y = "Relative Proportion to P1", fill = "Group") +
  scale_fill_manual(values = c("coral", "lightgreen")) +
  coord_flip() +
  theme_minimal()

```

# Fig 6 ####
```{r}
LRP <- readRDS('Atlas_v2_LRPsubset_Pseudotime.rds')
DimPlot(LRP,group.by = 'clusters', cols = c("#2EBAD1", "#5BC2AE"),
        pt.size=0.1, label = FALSE, alpha = 0.8)
DimPlot(LRP,group.by = 'supertype_name', cols = c("grey", "grey",'grey','grey','grey','grey','grey','grey','grey','#AC4758','grey','#2EBAD1','grey','grey','grey'),
        pt.size=0.1, label = FALSE, alpha = 0.8)

## B ####
metadata <-readRDS("metadata_MERFISH_Zhuang.RDS")
metadata_list<-lapply(names(table(metadata$brain_section_label)), function(x) metadata[which(metadata$brain_section_label == x),])
names(metadata_list) <- names(table(metadata$brain_section_label))
supertype<-c("Sst Chodl Gaba_2", 'Sst Chodl Gaba_4')  #Insert the class/subclass/supertype that you need without the number
section<-c("Zhuang-ABCA-2.027")  #Insert the section that you want to see
meta<-metadata_list[[section]]
cells <- rownames(meta[which(meta$supertype_label == supertype),]) #Change here class/subclass/supertype_label (meta$X)
sub<-meta[cells,]


##Plot the section with the spatial distribution of the supertype of interest
ggplot(meta, aes(x, y)) + 
  geom_point(colour = "gray90", size = 0.6) +  # Background points
  theme_void() +
  geom_point(data = sub, aes(x, y, colour = supertype_label)) +  
  scale_colour_manual(values = c("Sst Chodl Gaba_2" = "red", "Sst Chodl Gaba_4" = "blue")) +  # Define colors for supertypes
  scale_y_reverse() + 
  ggtitle(paste0("Section: ", section, ", Supertypes: ", paste(supertype, collapse = " and "))) + 
  theme(plot.title = element_text(hjust = 1))

## C ####
plot_data <- LRP@meta.data[, c("DC1", "DC2", "Pseudotime")]
ggplot(plot_data, aes(x = DC1, y = DC2, color = Pseudotime)) +
  geom_point(size = 1) +
  scale_color_viridis_c(option = "plasma") + 
  labs(title = "Pseudotime Plot on DC1-DC2", x = "DC1", y = "DC2") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5))

plot_data <- LRP@meta.data[, c("DC1", "DC2", "time_point")]
ggplot(plot_data, aes(x = DC1, y = DC2, color = time_point)) +
  geom_point(size = 1) +
  #scale_color_viridis_c(option = "plasma") + # You can choose different color scales
  scale_color_manual(values = c('#F067A6','#7FD2E8','#489DD7'))+
  labs(title = "Pseudotime Plot on DC1-DC2", x = "DC1", y = "DC2") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5))

plot_data <- LRP@meta.data[, c("DC1", "DC2", "clusters")]
ggplot(plot_data, aes(x = DC1, y = DC2, color = clusters)) +
  geom_point(size = 1) +
  #scale_color_viridis_c(option = "plasma") + # You can choose different color scales
  scale_color_manual(values = c("#2EBAD1", "#5BC2AE"))+
  labs(title = "Pseudotime Plot on DC1-DC2", x = "DC1", y = "DC2") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5))

## D ####
genes <- c("Mafb", "Crhbp", "Lingo1","Car10","Igf1","Lypd6b","Ncam2","Gad2","Oxtr",'Clmp') #LRP1 genes
genes <- c("Cited1","Hhip","Igsf21","Mgll","Selenom","Atp2b4","Brinp1","Elmod1","Rarb", 'Hs6st2') #LRP2.1 genes
genes <- c("Sox4", "Olfm3", "Cntnap4") #common genes
#Arrange data
plot_data <- FetchData(LRP, vars = c("Pseudotime", "clusters", genes))
cluster1_data <- plot_data %>% filter(clusters == "LRP1.0") %>% arrange(Pseudotime)
cluster2_data <- plot_data %>% filter(clusters == "LRP2.1") %>% arrange(Pseudotime)
# Combine ordered data, with cluster 1 cells first and then cluster 2 cells
ordered_data <- rbind(cluster1_data, cluster2_data)

#Create and arrange the matrix
heatmap_data <- as.matrix(ordered_data[, genes, drop = FALSE])
heatmap_data <- t(heatmap_data)  
rownames(heatmap_data) <- genes                     
colnames(heatmap_data) <- rownames(ordered_data)
annotation_col <- data.frame(Cluster = ordered_data$clusters, Pseudotime = ordered_data$Pseudotime)
rownames(annotation_col) <- rownames(ordered_data)

# Plot graph
pheatmap(
  heatmap_data,
  cluster_rows = FALSE,            # Keep gene order as specified
  cluster_cols = FALSE,            # Avoid reordering cells
  annotation_col = annotation_col, # Color columns by cluster
  gaps_col = nrow(cluster1_data),  # Add gap between clusters
  color = colorRampPalette(colors = c("#110807", "#926026", "#EE9A3A","#E5CC4D"))(2000),  # Custom color palette
  main = "Gene Expression Across Clusters by Pseudotime"
)

## E ####
LRP_numbers <- data.frame(
  E16 = c(304, 128),
  P1 = c(1348,   871),
  P5 = c(1980,   2206))
rownames(LRP_numbers) <- c("LRP1", "LRP2.1")
sst_chodl_props <- sweep(LRP_numbers, 2, colSums(LRP_numbers), FUN = "/")
sst_chodl_long <- as.data.frame(sst_chodl_props) %>%
  rownames_to_column(var = "Cell_Type") %>%
  melt(id.vars = "Cell_Type", variable.name = "Time_Point", value.name = "Proportion")
sst_chodl_long$Time_Point <- factor(sst_chodl_long$Time_Point, levels = c("E16",'P1',"P5"))
overall_proportion <- sst_chodl_long %>%
  group_by(Time_Point) %>%
  summarize(Proportion = sum(Proportion), .groups = 'drop') 
lrp2_only$Time_Point_numeric <- as.numeric(lrp2_only$Time_Point)

#plot
ggplot() +
  geom_bar(data = sst_chodl_long, aes(x = Time_Point, y = Proportion, fill = Cell_Type), 
           stat = "identity", position = "fill", alpha = 0.5) +  # Stacked bar plot
  scale_y_continuous(labels = scales::percent) +  # Y-axis as percentage
  labs(title = "Proportion of LRP1 and LRP2.1 Across Time Points",
       y = "Proportion",
       x = "Time Point") +
  scale_fill_manual(values = c("LRP1" = "blue2", "LRP2.1" = "#57C0AC")) +  # Custom colors
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

## F ####
#open Vanderhagen/Stein and Gao et al dataset and check LRP numbers
Pierre <- readRDS('SST_trajectory_PV.rds')
table(Pierre$Predicted_cluster, Pierre$Stage)
Gao <- readRDS()
table(Gao$predicted.id, Gao$Stage)
sst_chodl_counts <- data.frame(
  E15_5_Emir = c(48,  15),
  E17_5_Emir = c(46, 46),
  P1_Emir =    c(53,  52),
  P3_Emir =    c(45,  36),
  P5_Emir =    c(17,  23),
  P7_Emir =    c(47,  9), 
  P10_Emir =   c(30, 2),
  P14_Emir =   c(15,  4),
  P21_Emir =   c(19,  4),
  E18_P5_Gao =   c(18, 8),
  P6_P14_Gao =   c(89,  3),
  P15_P56_Gao =   c(57,  2))
rownames(sst_chodl_counts) <- c("LRP1", "LRP2.1")
sst_chodl_props <- sweep(sst_chodl_counts, 2, colSums(sst_chodl_counts), FUN = "/")
sst_chodl_long <- as.data.frame(sst_chodl_props) %>%
  rownames_to_column(var = "Cell_Type") %>%
  melt(id.vars = "Cell_Type", variable.name = "Time_Point", value.name = "Proportion")
sst_chodl_long$Time_Point <- factor(sst_chodl_long$Time_Point, levels = c('E15_5_Emir',"E17_5_Emir", "P1_Emir", "P3_Emir", "P5_Emir", "P7_Emir","P10_Emir", "P10_Mayer", "P14_Emir",'P21_Emir', 'E18_P5_Gao','P6_P14_Gao','P15_P56_Gao'))
overall_proportion <- sst_chodl_long %>%
  group_by(Time_Point) %>%
  summarize(Proportion = sum(Proportion), .groups = 'drop') 
lrp2_only$Time_Point_numeric <- as.numeric(lrp2_only$Time_Point)

 
ggplot() +
  geom_bar(data = sst_chodl_long, aes(x = Time_Point, y = Proportion, fill = Cell_Type), 
           stat = "identity", position = "fill", alpha = 0.5) +  # Stacked bar plot
  scale_y_continuous(labels = scales::percent) +  # Y-axis as percentage
  labs(title = "Proportion of LRP1 and LRP2.1 Across Time Points",
       y = "Proportion",
       x = "Time Point") +
  scale_fill_manual(values = c("LRP1" = "blue2", "LRP2.1" = "#57C0AC")) +  # Custom colors
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

## J ####
#Chcek Ext. Fig 6 for more detailed information
LRP_numbers <- data.frame(
  P5 = c(1401, 1126),
  P7 = c(1025,   371),
  P21 = c(1701,   373))
rownames(LRP_numbers) <- c("LRP1", "LRP2.1")
sst_chodl_props <- sweep(LRP_numbers, 2, colSums(LRP_numbers), FUN = "/")
sst_chodl_long <- as.data.frame(sst_chodl_props) %>%
  rownames_to_column(var = "Cell_Type") %>%
  melt(id.vars = "Cell_Type", variable.name = "Time_Point", value.name = "Proportion")
sst_chodl_long$Time_Point <- factor(sst_chodl_long$Time_Point, levels = c("P5",'P7',"P21"))
overall_proportion <- sst_chodl_long %>%
  group_by(Time_Point) %>%
  summarize(Proportion = sum(Proportion), .groups = 'drop') 
lrp2_only$Time_Point_numeric <- as.numeric(lrp2_only$Time_Point)

#plot
ggplot() +
  geom_bar(data = sst_chodl_long, aes(x = Time_Point, y = Proportion, fill = Cell_Type), 
           stat = "identity", position = "fill", alpha = 0.5) +  # Stacked bar plot
  scale_y_continuous(labels = scales::percent) +  # Y-axis as percentage
  labs(title = "Proportion of LRP1 and LRP2.1 Across Time Points",
       y = "Proportion",
       x = "Time Point") +
  scale_fill_manual(values = c("LRP1" = "blue2", "LRP2.1" = "#57C0AC")) +  # Custom colors
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

## K ###
Wu <- readRDS('Wu_Sst_Chodl_annotated.rde')
table(Wu$predicted.id, Wu$Stage)
sst_chodl_counts <- data.frame(
  Wu_P2 = c(49,  21),
  Wu_P7 = c(119, 21),
  Wu_P20 =  c(113,  1),
  Wu_P14wt =    c(108,  1),
  Wu_P14ko =    c(151,  55))
rownames(sst_chodl_counts) <- c("LRP1", "LRP2.1")
sst_chodl_props <- sweep(sst_chodl_counts, 2, colSums(sst_chodl_counts), FUN = "/")
verall_proportion <- sst_chodl_long %>% group_by(Time_Point) %>% summarize(Proportion = sum(Proportion), .groups = 'drop') 
sst_chodl_long <- as.data.frame(sst_chodl_props) %>% rownames_to_column(var = "Cell_Type") %>%
  melt(id.vars = "Cell_Type", variable.name = "Time_Point", value.name = "Proportion")
sst_chodl_long$Time_Point <- factor(sst_chodl_long$Time_Point, levels = c('Wu_P2',"Wu_P7", "Wu_P20", "Wu_P14wt", "Wu_P14ko"))
lrp2_only$Time_Point_numeric <- as.numeric(lrp2_only$Time_Point)

ggplot() +
  geom_bar(data = sst_chodl_long, aes(x = Time_Point, y = Proportion, fill = Cell_Type), 
           stat = "identity", position = "fill", alpha = 0.5) +  # Stacked bar plot
  scale_y_continuous(labels = scales::percent) +  # Y-axis as percentage
  labs(title = "Proportion of LRP1 and LRP2.1 Across Time Points",
       y = "Proportion",
       x = "Time Point") +
  scale_fill_manual(values = c("LRP1" = "blue2", "LRP2.1" = "#57C0AC")) +  # Custom colors
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

## L ####
table(Badhuri$predicted.id, Badhuri$Stage)
table(Velmeshev$predicted.id, Velmeshev$Stage)
sst_chodl_counts <- data.frame(
  GW18_20 = c(32,  31),
  GW22_25 = c(32, 13),
  Postnatal =  c(41,  12),
  GW14_27 =    c(151,  55),
  GW28_40 =    c(108,  1),
  PN_010 =    c(108,  1),
  PN_10 =    c(151,  55))
rownames(sst_chodl_counts) <- c("LRP1", "LRP2.1")
sst_chodl_props <- sweep(sst_chodl_counts, 2, colSums(sst_chodl_counts), FUN = "/")
verall_proportion <- sst_chodl_long %>% group_by(Time_Point) %>% summarize(Proportion = sum(Proportion), .groups = 'drop') 
sst_chodl_long <- as.data.frame(sst_chodl_props) %>% rownames_to_column(var = "Cell_Type") %>%
  melt(id.vars = "Cell_Type", variable.name = "Time_Point", value.name = "Proportion")
sst_chodl_long$Time_Point <- factor(sst_chodl_long$Time_Point, levels = c('GW18_20',"GW22_25", "Postnatal", "GW14_27", "GW28_40", 'PN_010', 'PN_10'))
lrp2_only$Time_Point_numeric <- as.numeric(lrp2_only$Time_Point)

ggplot() +
  geom_bar(data = sst_chodl_long, aes(x = Time_Point, y = Proportion, fill = Cell_Type), 
           stat = "identity", position = "fill", alpha = 0.5) +  # Stacked bar plot
  scale_y_continuous(labels = scales::percent) +  # Y-axis as percentage
  labs(title = "Proportion of LRP1 and LRP2.1 Across Time Points",
       y = "Proportion",
       x = "Time Point") +
  scale_fill_manual(values = c("LRP1" = "blue2", "LRP2.1" = "#57C0AC")) +  # Custom colors
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

## M ###
Bershteyn <- readRDS('Bershteyn')
table(Bershteyn$cluster_prediction, Bershteyn$Stage)
sst_chodl_counts <- data.frame(
  MPT_1 = c(507,  189),
  MPT_3 = c(405, 46),
  MPT_6 =  c(2115,  272),
  MPT_12_18 =    c(908,  117))
rownames(sst_chodl_counts) <- c("LRP1", "LRP2.1")
sst_chodl_props <- sweep(sst_chodl_counts, 2, colSums(sst_chodl_counts), FUN = "/")
verall_proportion <- sst_chodl_long %>% group_by(Time_Point) %>% summarize(Proportion = sum(Proportion), .groups = 'drop') 
sst_chodl_long <- as.data.frame(sst_chodl_props) %>% rownames_to_column(var = "Cell_Type") %>%
  melt(id.vars = "Cell_Type", variable.name = "Time_Point", value.name = "Proportion")
sst_chodl_long$Time_Point <- factor(sst_chodl_long$Time_Point, levels = c('MPT_1',"MPT_3", "MPT_6", "MPT_12_18"))
lrp2_only$Time_Point_numeric <- as.numeric(lrp2_only$Time_Point)

ggplot() +
  geom_bar(data = sst_chodl_long, aes(x = Time_Point, y = Proportion, fill = Cell_Type), 
           stat = "identity", position = "fill", alpha = 0.5) +  # Stacked bar plot
  scale_y_continuous(labels = scales::percent) +  # Y-axis as percentage
  labs(title = "Proportion of LRP1 and LRP2.1 Across Time Points",
       y = "Proportion",
       x = "Time Point") +
  scale_fill_manual(values = c("LRP1" = "blue2", "LRP2.1" = "#57C0AC")) +  # Custom colors
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

## N ####
DimPlot(Bershteyn, group.by = 'cluster_prediction', cols = c("blue2", "#57C0AC"))

rm(list = c("annotation_col", "cluster1_data", "cluster2_data", "heatmap_data", 
            "LRP_numbers", "lrp2_only", "meta", "metadata", "metadata_list", 
            "ordered_data", "overall_proportion", "plot_data", "sst_chodl_counts", 
            "sst_chodl_long", "sst_proportion", "sst_chodl_props", "sub", 
            "cells", "desired_order", "genes", "section", "supertype", "LRP"))
gc()

```

# Ext Fig 2 ####
```{r}
## B ####
#it gives a .pdf output
data<-read.table("data.txt", row.names = 1, header = TRUE)

files <- dir()
files_anchors <- files[grep("Number_of_anchors_",files)]
files_kBET <- files[grep("kBET_batch_estimate_",files)]
files_kBET <- files_kBET[grep(".RDS",files_kBET)]

rejection_rate <- c()
mean_anchors <- c()
mean_anchors_scaled <- c()
for (i in rownames(data)){
  n_anchors <- read.table(paste0(files_anchors[grep(i,files_anchors)]), row.names = 1)
  mean_anchors <- c(mean_anchors,mean(n_anchors[,1]))
  mean_anchors_scaled <- c(mean_anchors_scaled,mean(n_anchors[,1])/data[i,2])
  kbet<-readRDS(paste0(files_kBET[grep(i,files_kBET)]))
  rejection_rate<-c(rejection_rate,median(kbet$stats$kBET.observed))
}
colors_ditto <- dittoColors()

files_knn <- files[grep("Percentages_of_cells_in_",files)]
files_knn <- files_knn[grep(".RDS",files_knn)]
pdf('plot_percentages_of_cells_with_mixed_neighborhood_in_kNN_allSamples.pdf', width=8)
par(cex=0.5,cex.lab=1.2)
i=1
knn_perc<-readRDS(paste0(files_knn[grep(rownames(data)[i],files_knn)]))
plot(seq(5,200,5),knn_perc, xlab="k",ylab="fraction of cells with mixed neighborhood", pch=0,cex=0.8,col=colors_ditto[i], ylim=c(0,1.05),xaxt='n',yaxt='n')
legend("top",legend=paste0(data[,1]," - ",rownames(data[,])),fill=colors_ditto,horiz = FALSE,cex=0.915,pch = c(0:12),ncol=6)
axis(2,cex.axis=1.5,)
axis(1,cex.axis=1.5)
print(lines(seq(5,200,5),knn_perc,col=colors_ditto[i]))
for (i in 2:nrow(data)){
  knn_perc<-readRDS(paste0(files_knn[grep(rownames(data)[i],files_knn)]))
  print(points(seq(5,200,5),knn_perc, pch=i-1,col=colors_ditto[i]))
  print(lines(seq(5,200,5),knn_perc,col=colors_ditto[i]))
}
abline(v=60, col = "blue4", lty=2)
abline(h=0.45, col = "red3", lty=2)
dev.off()

knn_perc_k60<-c()
for (i in 1:nrow(data)){
  knn_perc<-readRDS(paste0(files_knn[grep(rownames(data)[i],files_knn)]))
  knn_perc_k60<-c(knn_perc_k60,knn_perc["60"])
}

## C ##
data <- read_csv('DS_anchors.csv')
ggplot(data, aes(x = reorder(dataset_id, -Anchors), y = Anchors)) +
  stat_summary_bin(aes(y = Anchors), fun = "mean", geom = "bar")+
  stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.2, color = "black")+
  geom_jitter()+
  geom_hline(yintercept = 15000, linetype = "dashed", color = "red")

## D ####
#it gives a .pdf output
data_plot1 <- data.frame(data, mean_anchors = mean_anchors, kBET_rejection_rate=rejection_rate, knn_perc_k60=knn_perc_k60)
pdf('Mean_ancors_vs_k60_fraction_of_cells.pdf')
plot(data_plot1$knn_perc_k60, data_plot1$mean_anchors, col=colors_ditto, pch=c(0:12),cex=1.5,lwd=2, xlab ="fraction of cells with mixed neighborhood", ylab="Mean anchors")
legend("topleft",legend=paste0(data[,1]," - ",rownames(data[,])),fill=colors_ditto,horiz = FALSE, cex = 0.6, pch = c(0:12))
abline(v=0.45, col="red3", lty=2)
abline(h=2000, col="red3", lty=2)
dev.off()

## E ####
#it gives a .pdf output
data_plot1<-data.frame(data,mean_anchors = mean_anchors, mean_anchors_scaled = mean_anchors_scaled, kBET_rejection_rate=rejection_rate)
pdf('Mean_anchors_vs_kBET_RejectionRate.pdf')
plot(data_plot1$kBET_rejection_rate, data_plot1$mean_anchors, col=colors_ditto, pch=c(0:12),cex=1.5,lwd=2, xlab ="kBET rejection rate", ylab="Mean anchors")
legend("topright",legend=paste0(data[,1]," - ",rownames(data)),fill=colors_ditto,horiz = FALSE,cex=0.6,pch = c(0:12))
abline(v=0.5, col = "red3",lty = 2)
abline(h=2000, col = "red3",lty = 2)
dev.off()

rm(list = c("data", "data_plot1", "kbet", "n_anchors", "colors_ditto", 
            "files", "files_anchors", "files_kBET", "files_knn", "i", 
            "knn_perc", "knn_perc_k60", "mean_anchors", "mean_anchors_scaled", 
            "rejection_rate"))
gc()
```

# Ext Fig 3 ####
```{r}
## A ####
conserved_clusters <- readRDS('MN_clusters_conserved_finalClusters.RDS')
cluster_number<-list()
for (perc.sub in seq(0.05, 0.95, 0.05)){
  cluster_number[[as.character(perc.sub)]]<-unlist(lapply(conserved_clusters[[as.character(perc.sub)]], function(x) {
    sum(diag(x) >0.9)
  }))
}
cluster_number_all<-35

avg<-rev(unlist(lapply(cluster_number, mean))/cluster_number_all)
x<-round(seq(0.05, 0.95, 0.05)*51205)
err <- unlist(lapply(cluster_number, function(x) sd(x/cluster_number_all)/sqrt(length(x))))
pdf('Number_ofcluster_finalClustersMarkers_MAST_v2.pdf', width = 8, height = 6)
par(las=2, mar=c(6, 5, 4, 3))
plot(seq(0.05, 0.95, 0.05), rev(unlist(lapply(cluster_number, mean))/cluster_number_all), pch=20, xaxt='n', xlab="", ylab="Fraction of conserved cluster", ylim=c(0.54,1.02))
axis(1, at=seq(0.05, 0.95, 0.05), labels = rev(x), cex.lab=1.2, cex.axis=1.2)
lines(seq(0.05, 0.95, 0.05), avg)
title(xlab="Number of cells",mgp=c(4,1,0))
abline(v=0.8, col="red3", lty=2) #inflection point
arrows(seq(0.05, 0.95, 0.05), avg-rev(err), seq(0.05, 0.95, 0.05), avg+rev(err), length=0.05, angle=90, code=3)

cluster_number<-list()
for (perc.sub in seq(0.05, 0.95, 0.05)){
  cluster_number[[as.character(perc.sub)]]<-unlist(lapply(conserved_clusters[[as.character(perc.sub)]], function(x) {
    sum(diag(x) >0.88)
  }))
}
avg<-rev(unlist(lapply(cluster_number, mean))/cluster_number_all)
x<-round(seq(0.05, 0.95, 0.05)*51205)
err <- unlist(lapply(cluster_number, function(x) sd(x/cluster_number_all)/sqrt(length(x))))
points(seq(0.05, 0.95, 0.05), rev(unlist(lapply(cluster_number, mean))/cluster_number_all), pch=20, col="red4")
lines(seq(0.05, 0.95, 0.05), avg,col="red4")
arrows(seq(0.05, 0.95, 0.05), avg-rev(err), seq(0.05, 0.95, 0.05), avg+rev(err), length=0.05, angle=90, code=3, col = "red4")

cluster_number<-list()
for (perc.sub in seq(0.05, 0.95, 0.05)){
  cluster_number[[as.character(perc.sub)]]<-unlist(lapply(conserved_clusters[[as.character(perc.sub)]], function(x) {
    sum(diag(x) >0.86)
  }))
}
avg<-rev(unlist(lapply(cluster_number, mean))/cluster_number_all)
x<-round(seq(0.05, 0.95, 0.05)*51205)
err <- unlist(lapply(cluster_number, function(x) sd(x/cluster_number_all)/sqrt(length(x))))
points(seq(0.05, 0.95, 0.05), rev(unlist(lapply(cluster_number, mean))/cluster_number_all), pch=20, col="blue4")
lines(seq(0.05, 0.95, 0.05), avg,col="blue4")
arrows(seq(0.05, 0.95, 0.05), avg-rev(err), seq(0.05, 0.95, 0.05), avg+rev(err), length=0.05, angle=90, code=3, col = "blue4")

legend("bottom",x.intersp = c(0.5,0.5,0.5),pch = 20,lty=1, legend = c(expression(AUROC > 0.9),expression(AUROC > 0.88),expression(AUROC > 0.86)), col = c("black","red4","blue4"), horiz = TRUE,inset=c(0,1), xpd=TRUE, bty="n" )
dev.off()

## B ###
majorClassMarkers <- readRDS("allDEGs_final_clusters_subsets_Atlas_v2.RDS")
DEGs_list <- readRDS("allDEGs_final_clusters_subsets.RDS")

DEGs_number<-list()
for (perc.sub in seq(0.05, 0.95, 0.05)){
  DEGs_number[[as.character(perc.sub)]]<-unlist(lapply(DEGs_list[[as.character(perc.sub)]], function(x) {
    x <- subset(x,pct.1 > 0.2 & avg_log2FC>log2(1.8))
    dim(x)[1]
  }))
}
DEGs_number_all <- dim(subset(majorClassMarkers,pct.1 > 0.2 & avg_log2FC>log2(1.8)))[1]

avg<-rev(unlist(lapply(DEGs_number, mean))/DEGs_number_all)
x<-round(seq(0.05, 0.95, 0.05)*51205)
err <- unlist(lapply(DEGs_number, function(x) sd(x/DEGs_number_all)/sqrt(length(x))))
pdf('Number_ofDEGs_finalClustersMarkers_MAST.pdf', width = 8, height = 6)
par(las=2, mar=c(6, 5, 4, 3))
plot(seq(0.05, 0.95, 0.05), rev(unlist(lapply(DEGs_number, mean))/DEGs_number_all), pch=20, xaxt='n', xlab="", ylab="Fraction of conserved DEGs", ylim=c(0.48,1.02))
axis(1, at=seq(0.05, 0.95, 0.05), labels = rev(x), cex.lab=1.2, cex.axis=1.2)
lines(seq(0.05, 0.95, 0.05), avg)
title(xlab="Number of cells",mgp=c(4,1,0))
abline(v=0.75, col="red3", lty=2) #inflection point
arrows(seq(0.05, 0.95, 0.05), avg-rev(err), seq(0.05, 0.95, 0.05), avg+rev(err), length=0.05, angle=90, code=3)

DEGs_number<-list()
for (perc.sub in seq(0.05, 0.95, 0.05)){
  DEGs_number[[as.character(perc.sub)]]<-unlist(lapply(DEGs_list[[as.character(perc.sub)]], function(x) {
    x <- subset(x,pct.1 > 0.2 & avg_log2FC>log2(2))
    dim(x)[1]
  }))
}
DEGs_number_all <- dim(subset(majorClassMarkers,pct.1 > 0.2 & avg_log2FC>log2(2)))[1]
avg<-rev(unlist(lapply(DEGs_number, mean))/DEGs_number_all)
x<-round(seq(0.05, 0.95, 0.05)*51205)
err <- unlist(lapply(DEGs_number, function(x) sd(x/DEGs_number_all)/sqrt(length(x))))
points(seq(0.05, 0.95, 0.05), rev(unlist(lapply(DEGs_number, mean))/DEGs_number_all), pch=20, col="red4")
lines(seq(0.05, 0.95, 0.05), avg,col="red4")
arrows(seq(0.05, 0.95, 0.05), avg-rev(err), seq(0.05, 0.95, 0.05), avg+rev(err), length=0.05, angle=90, code=3, col = "red4")

DEGs_number <- list()
for (perc.sub in seq(0.05, 0.95, 0.05)){
  DEGs_number[[as.character(perc.sub)]]<-unlist(lapply(DEGs_list[[as.character(perc.sub)]], function(x) {
    x <- subset(x,pct.1 > 0.2 & avg_log2FC>log2(2.2))
    dim(x)[1]
  }))
}
DEGs_number_all <- dim(subset(majorClassMarkers,pct.1 > 0.2 & avg_log2FC>log2(2.2)))[1]
avg<-rev(unlist(lapply(DEGs_number, mean))/DEGs_number_all)
x<-round(seq(0.05, 0.95, 0.05)*51205)
err <- unlist(lapply(DEGs_number, function(x) sd(x/DEGs_number_all)/sqrt(length(x))))
points(seq(0.05, 0.95, 0.05), rev(unlist(lapply(DEGs_number, mean))/DEGs_number_all), pch=20, col="blue4")
lines(seq(0.05, 0.95, 0.05), avg,col="blue4")
arrows(seq(0.05, 0.95, 0.05), avg-rev(err), seq(0.05, 0.95, 0.05), avg+rev(err), length=0.05, angle=90, code=3, col = "blue4")

legend("bottom",x.intersp = c(0.5,0.5,0.5),pch = 20,lty=1, legend = c(expression(log[2](FC) > log[2](1.8)),expression(log[2](FC) > log[2](2.0)),expression(log[2](FC) > log[2](2.2))), col = c("black","red4","blue4"), horiz = TRUE,inset=c(0,1), xpd=TRUE, bty="n" )
dev.off()

## C ####
diff_mean <- readRDS('diff_mean_subsampling_DEGs.RDS')
x<-round(seq(0.05, 0.95, 0.05)*51205)
avg<-rev(unlist(lapply(diff_mean, mean)))
err <- unlist(lapply(diff_mean, function(x) sd(x)/sqrt(length(x))))
pdf('Differences_MeanExpression_of_finalClustersMarkers_MAST_v2.pdf', width = 8, height = 6)
par(las=2, mar=c(6, 5, 3, 3))
plot(seq(0.05, 0.95, 0.05), rev(unlist(lapply(diff_mean, mean))), pch=20, xaxt='n', xlab="", ylab="Absolute Difference (mean expression)")
axis(1, at=seq(0.05, 0.95, 0.05), labels = rev(x), cex.lab=1.2, cex.axis=1.2)
lines(seq(0.05, 0.95, 0.05), rev(unlist(lapply(diff_mean, mean))))
title(xlab="Number of cells",mgp=c(4,1,0))
abline(v=0.75, col="red3", lty=2) #inflection point
arrows(seq(0.05, 0.95, 0.05), avg-rev(err), seq(0.05, 0.95, 0.05), avg+rev(err), length=0.05, angle=90, code=3)
dev.off()

## D ####
HausdorffDistance <- readRDS('HausdorffDistance_q50.RDS')
x<-round(seq(0.05, 0.95, 0.05)*51205)
avg<-rev(unlist(lapply(HausdorffDistance, mean)))
err <- unlist(lapply(HausdorffDistance, function(x) sd(x)/sqrt(length(x))))
pdf('HausdorffDistance_of_subsampling_q50_v2.pdf', width = 8, height = 6)
par(las=2, mar=c(6, 5, 3, 3))
plot(seq(0.05, 0.95, 0.05), rev(unlist(lapply(HausdorffDistance, mean))), pch=20, xaxt='n', xlab="", ylab="Robust Hausdorff Distance")
axis(1, at=seq(0.05, 0.95, 0.05), labels = rev(x), cex.lab=1.2, cex.axis=1.2)
lines(seq(0.05, 0.95, 0.05), rev(unlist(lapply(HausdorffDistance, mean))))
title(xlab="Number of cells",mgp=c(4,1,0))
abline(v=0.8, col="red3",lty=2) #inflection point
arrows(seq(0.05, 0.95, 0.05), avg-rev(err), seq(0.05, 0.95, 0.05), avg+rev(err), length=0.05, angle=90, code=3)
dev.off()

## E ####
#load data
jc_to_plot_all <- readRDS("Jaccard_allClusters.RDS")
colors_plot<-readRDS("palette_clusters.RDS")

pdf('JC_ClusterStability_plot_all_clusters.pdf', width=15, height = 10)
jc_to_plot_all %>% ggplot2::ggplot(ggplot2::aes(x = cluster, y = jaccard, fill = cluster)) +
  geom_flat_violin(position = position_nudge(x = .2, y = 0), alpha = .7, width = 3) +
  ggplot2::geom_point(ggplot2::aes(y = jaccard, color = cluster), position = position_jitter(width = .15), size = .5, alpha = 0.7) +
  ggplot2::geom_boxplot(width = .2, outlier.shape = NA, alpha = 0.5) +
  ggplot2::theme_classic() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + ylim(0,1) +
  ggplot2::theme(legend.position="none") +  scale_colour_manual(values=colors_plot) +  scale_fill_manual(values=colors_plot) + geom_hline(yintercept = c(0.75), linetype = 2,color="red2")
dev.off()


rm(list = c("cluster_number", "conserved_clusters", "DEGs_list", "DEGs_number", 
            "diff_mean", "HausdorffDistance", "jc_to_plot_all", "majorClassMarkers", 
            "avg", "cluster_number_all", "colors_plot", "DEGs_number_all", 
            "err", "perc.sub", "x"))
gc()

```

# Ext Fig 4 ####
```{r}
## A ####
LRP <- readRDS('Atlas_v2_LRPsubset_IterativeClustering_and_MapMyCell.RDS')
LRPmetadata <- LRP@meta.data
cellpercentages <- LRPmetadata %>%
  group_by(final_clusters) %>%
  summarise(count = n(), .groups = 'drop') %>%
  mutate(percentage = count / sum(count) * 100)
ggplot(cellpercentages, aes(x = final_clusters, y = percentage, fill = final_clusters)) +
  geom_bar(stat = "identity", position = "stack") +
  labs(title = "Percentage of Cells by cluster",
       x = "Cluster",
       y = "Percentage (%)") +
  theme_minimal() +
  scale_fill_manual(values = c("#30BBD1","#5DC3AF", 'darkgreen')) +
  theme(legend.position = "right")

## B ####
DimPlot(LRP,group.by = 'final_clusters', cols = c("#30BBD1","#5DC3AF", 'darkgreen'),
        pt.size=1.2, label = FALSE, alpha = 0.8)

## C, D, E ####
##Subset LRP, MC and nMC from Atlas_v2 as in Figure 5 and reannotate the subset as 'obj
obj <- LRP

##### Compute markers and select top genes
DefaultAssay(obj) <-'RNA'
Idents(obj) <- 'final_clusters'
markers <- FindAllMarkers(obj,logfc.threshold = 1,min.pct = 0.2,only.pos = TRUE, test.use = 'MAST', latent.vars = 'sample')
markers<-markers[order(markers$avg_log2FC, decreasing = TRUE),]
markers<-markers[which(markers$p_val_adj < 0.01),]
markers<-markers[which(markers$pct.1 -markers$pct.2 > 0.15),]
markers<-markers %>% group_by(cluster) %>% top_n(n = 10)
## Dot plot
p <- DotPlot(object = obj, assay="RNA", features=unique(markers$gene),scale=TRUE) + ylab("Clusters") + scale_colour_gradient2(low = "blue2", mid = "gray90", high = "red2", midpoint=0) + theme(axis.text.x = element_text(angle = 90, hjust = 1))
print(p)

## Add hierarchical clustering
df<- p$data
exp_mat<-df %>% select(-pct.exp, -avg.exp) %>%  pivot_wider(names_from = id, values_from = avg.exp.scaled) %>% as.data.frame() 
row.names(exp_mat) <- exp_mat$features.plot  
exp_mat <- exp_mat[,-1] %>% as.matrix()
percent_mat<-df %>% select(-avg.exp, -avg.exp.scaled) %>%  pivot_wider(names_from = id, values_from = pct.exp) %>% as.data.frame() 
row.names(percent_mat) <- percent_mat$features.plot  
percent_mat <- percent_mat[,-1] %>% as.matrix()

col_fun = circlize::colorRamp2(c(-1, 0, 2.5), c("blue2","gray90","red2"))
layer_fun = function(j, i, x, y, w, h, fill){
  grid.rect(x = x, y = y, width = w, height = h, 
            gp = gpar(col = NA, fill = NA))
  grid.circle(x=x,y=y,r= pindex(percent_mat, i, j)/100 * unit(1.5, "mm"),
              gp = gpar(fill = col_fun(pindex(exp_mat, i, j)), col = NA))}
lgd_list = list(
  Legend( labels = c(0,0.25,0.5,0.75,1), title = "pt",
          graphics = list(
            function(x, y, w, h) grid.circle(x = x, y = y, r = 0 * unit(1.5, "mm"),
                                             gp = gpar(fill = "black")),
            function(x, y, w, h) grid.circle(x = x, y = y, r = 0.25 * unit(1.5, "mm"),
                                             gp = gpar(fill = "black")),
            function(x, y, w, h) grid.circle(x = x, y = y, r = 0.5 * unit(1.5, "mm"),
                                             gp = gpar(fill = "black")),
            function(x, y, w, h) grid.circle(x = x, y = y, r = 0.75 * unit(1.5, "mm"),
                                             gp = gpar(fill = "black")),
            function(x, y, w, h) grid.circle(x = x, y = y, r = 1 * unit(1.5, "mm"),
                                             gp = gpar(fill = "black")))
  ))
set.seed(123)
# compute heatmap and extract hierarchical clustering
hp<- Heatmap(exp_mat,
             heatmap_legend_param=list(title="expression"),
             column_title = "clustered dotplot", 
             col=col_fun,
             rect_gp = gpar(type = "none"),
             layer_fun = layer_fun,
             row_names_gp = gpar(fontsize = 7),
             border = "black",
             cluster_rows = FALSE)
cluster_order<-hp@column_names_param$labels[column_order(hp)]
# Reorder genes
tmp<-markers %>% arrange(factor(cluster, levels = cluster_order))
ordered_genes <- unique(tmp$gene)
exp_mat<-exp_mat[ordered_genes,]
percent_mat<-percent_mat[ordered_genes,]
## print heamtmap in dot plot format
hp<- Heatmap(exp_mat,
             heatmap_legend_param=list(title="expression"),
             column_title = "clustered dotplot", 
             col=col_fun,
             rect_gp = gpar(type = "none"),
             layer_fun = layer_fun,
             row_names_gp = gpar(fontsize = 7),
             border = "black",
             cluster_rows = FALSE)
draw( hp, annotation_legend_list = lgd_list)

rm(list = c("cellpercentages", "df", "exp_mat", "hp", "lgd_list", "LRP", 
            "LRPmetadata", "markers", "obj", "p", "percent_mat", "tmp", 
            "cluster_order", "ordered_genes", "col_fun", "layer_fun"))
gc()

```

# Ext Fig 5 ####
```{r}
## A ####
library(dplyr)
        
metadata <- Atlas_v2@meta.data
metadata <- metadata %>%
  filter(!final_clusters_renamed %in% c("HIn1", "HIn2", "LRP1.0", "LRP2.1", "LRP2.2")) %>%
  mutate(Category = ifelse(subclass_bootstrapping_probability >= 0.9, ">= 0.9", "<0.9"))

percentage_data <- metadata %>%
  group_by(major_label_transferAnchors_BaseAtlas, Category) %>%
  summarise(count = n()) %>%
  ungroup() %>%
  group_by(major_label_transferAnchors_BaseAtlas) %>%
  mutate(Percentage = (count / sum(count)) * 100)
percentage_long <- percentage_data %>%
  dplyr::select(major_label_transferAnchors_BaseAtlas, Percentage, Category) %>%
  pivot_longer(cols = Percentage, names_to = "Probability_Range", values_to = "Percentage")
ggplot(percentage_long, aes(x = major_label_transferAnchors_BaseAtlas, y = Percentage, fill = Category)) +
  geom_bar(stat = "identity") +
  labs(title = "Percentage of Cells by Subclass Bootstrapping Probability Range",
       x = "Major",
       y = "Percentage of Cells") +
  scale_fill_manual(values = c(">=0.9" = "blue", "<0.9" = "lightblue")) +
  theme_minimal() +
  theme(legend.title = element_blank())

## B,C ####
metadata <- Atlas_v2@meta.data
metadata <- metadata %>%
  filter(!final_clusters_renamed %in% c("HIn1", "HIn2", "LRP1.0", "LRP2.1", "LRP2.2")) %>%
  mutate(Category = ifelse(supertype_bootstrapping_probability >= 0.9, ">=0.9", "<0.9"))
percentage_data <- metadata %>%
  group_by(major_label_transferAnchors_BaseAtlas, Category, group) %>%
  summarise(count = n(), .groups = 'drop') %>%
  group_by(major_label_transferAnchors_BaseAtlas, group) %>%
  mutate(Percentage = (count / sum(count)) * 100) %>%
  ungroup()
percentage_long <- percentage_data %>%
  select(major_label_transferAnchors_BaseAtlas, Percentage, Category, group) %>%
  pivot_longer(cols = Percentage, names_to = "Probability_Range", values_to = "Percentage")
ggplot(percentage_long, aes(x = group, y = Percentage, fill = Category)) +
  geom_bar(stat = "identity") +  # Use position = "dodge" for side-by-side bars
  labs(title = "Percentage of Cells by Supertype Bootstrapping Probability Range",
       x = "group",
       y = "Percentage of Cells") +
  scale_fill_manual(values = c(">=0.9" = "blue", "<0.9" = "lightblue")) +
  facet_wrap(~ major_label_transferAnchors_BaseAtlas) +  # Create separate panels for each group
  theme_minimal() +
  theme(legend.title = element_blank())

## D ####
metadatav2 <- Atlas_v2@meta.data
metadatav2 <- metadatav2 %>%
  filter(!final_clusters_renamed %in% c("HIn1", "HIn2", "LRP1.0", "LRP2.1", "LRP2.2"))
metadatav2 <- metadatav2 %>% mutate(SupertypeClean_AllOther = case_when(
  supertype_name %in% c("0135 HPF CR Glut_1", "0142 DG-PIR Ex IMN_3", "0144 OB Meis2 Thsd7b Gaba_2", "0160 OB-out Frmd7 Gaba_3", "0165 OB Dopa-Gaba_4", "0166 OB-STR-CTX Inh IMN_1", 
                        "0167 OB-STR-CTX Inh IMN_2", "0168 OB-STR-CTX Inh IMN_3", "0169 OB-STR-CTX Inh IMN_4", "0170 OB-STR-CTX Inh IMN_5", "0171 OB-STR-CTX Inh IMN_6",
                        "0172 OB-STR-CTX Inh IMN_7", "0173 Vip Gaba_1", "0174 Vip Gaba_2", "0176 Vip Gaba_4", "0177 Vip Gaba_5", 
                        "0181 Vip Gaba_9", "0187 Sncg Gaba_3", "0193 RHP-COA Ndnf Gaba_1", "0194 RHP-COA Ndnf Gaba_2", "0195 RHP-COA Ndnf Gaba_3", "0196 RHP-COA Ndnf Gaba_4", 
                        "0197 RHP-COA Ndnf Gaba_5", "0198 RHP-COA Ndnf Gaba_6", "0199 Lamp5 Gaba_1", "0200 Lamp5 Gaba_2", "0201 Lamp5 Gaba_3", "0202 Lamp5 Gaba_4", 
                        "0203 Lamp5 Lhx6 Gaba_1", "0204 Pvalb chandelier Gaba_1", "0205 Pvalb Gaba_1", "0206 Pvalb Gaba_2", "0207 Pvalb Gaba_3", "0208 Pvalb Gaba_4", 
                        "0209 Pvalb Gaba_5", "0210 Pvalb Gaba_6", "0211 Pvalb Gaba_7",  "0212 Pvalb Gaba_8", "0213 Pvalb Gaba_9","0233 STR Prox1 Lhx6 Gaba_1",
                        "0234 STR Prox1 Lhx6 Gaba_2", "0235 STR Prox1 Lhx6 Gaba_3", "0236 STR Lhx8 Gaba_1",  "0237 STR Lhx8 Gaba_2", "0238 Sst Chodl Gaba_1", "0239 Sst Chodl Gaba_2", 
                        "0240 Sst Chodl Gaba_3", "0241 Sst Chodl Gaba_4", "0242 Sst Chodl Gaba_5",  "0243 Sst Chodl Gaba_6", "0244 NDB-SI-MA-STRv Lhx8 Gaba_1", 
                        "0251 NDB-SI-MA-STRv Lhx8 Gaba_8", "0253 NDB-SI-MA-STRv Lhx8 Gaba_10", "0258 GPe-SI Sox6 Cyp26b1 Gaba_2", "0294 IA Mgp Gaba_5", "0339 MEA-BST Sox6 Gaba_3",
                        "0341 MEA-BST Sox6 Gaba_5", "0355 MEA-BST Lhx6 Nr2e1 Gaba_2", "0356 MEA-BST Lhx6 Nr2e1 Gaba_3", "0357 MEA-BST Lhx6 Nfib Gaba_1", "0358 MEA-BST Lhx6 Nfib Gaba_2", 
                        "0361 MEA-BST Lhx6 Nfib Gaba_5", "0362 MEA-BST Lhx6 Nfib Gaba_6", "0476 PVHd-DMH Lhx6 Gaba_3", "0799 MRN-VTN-PPN Pax5 Cdh23 Gaba_2", 
                        "0835 LGv-SPFp-SPFm Nkx2-2 Tcf7l2 Gaba_2", "0836 LGv-SPFp-SPFm Nkx2-2 Tcf7l2 Gaba_3", "0846 SC-PAG Lef1 Emx2 Gaba_4", "1056 LDT-PCG-CS Gata3 Lhx1 Gaba_1", 
                        "1091 MY Lhx1 Gly-Gaba_3", "1099 MV-SPIV-PRP Dmbx1 Gly-Gaba_1",  "1120 PARN-MDRNd-NTS Gbx2 Gly-Gaba_1", "1180 OPC NN_2", "1195 BAM NN_1", "1198 B cells NN_1") ~ "Other",
  TRUE ~ supertype_name))
metadatav2 <- metadatav2 %>%
  filter(SupertypeClean_AllOther != "Other") %>%
  mutate(supertype_categoryAllOther = ifelse(
    supertype_bootstrapping_probability >= 0.9, 
    SupertypeClean_AllOther, 
    "Not assigned")) 
plot_data_v2 <- metadatav2 %>%
  group_by(group, supertype_categoryAllOther) %>% summarise(count = n()) %>% mutate(proportion = count / sum(count)) %>% ungroup()
Yao <- readRDS('Yao_2023_Sst_Gaba_053_056_data.RDS')
metadataYao <- Yao@meta.data
plot_data_Yao <- metadataYao %>% dplyr::rename(supertype_categoryAllOther = supertype) %>%
  mutate(supertype_categoryAllOther = case_when(
    supertype_categoryAllOther == "Sst Gaba_1"  ~ "0214 Sst Gaba_1",
    supertype_categoryAllOther == "Sst Gaba_2"  ~ "0215 Sst Gaba_2",
    supertype_categoryAllOther == "Sst Gaba_3"  ~ "0216 Sst Gaba_3",
    supertype_categoryAllOther == "Sst Gaba_4"  ~ "0217 Sst Gaba_4",
    supertype_categoryAllOther == "Sst Gaba_5"  ~ "0218 Sst Gaba_5",
    supertype_categoryAllOther == "Sst Gaba_6"  ~ "0219 Sst Gaba_6",
    supertype_categoryAllOther == "Sst Gaba_7"  ~ "0220 Sst Gaba_7",
    supertype_categoryAllOther == "Sst Gaba_8"  ~ "0221 Sst Gaba_8",
    supertype_categoryAllOther == "Sst Gaba_9"  ~ "0222 Sst Gaba_9",
    supertype_categoryAllOther == "Sst Gaba_10" ~ "0223 Sst Gaba_10",
    supertype_categoryAllOther == "Sst Gaba_11" ~ "0224 Sst Gaba_11",
    supertype_categoryAllOther == "Sst Gaba_12" ~ "0225 Sst Gaba_12",
    supertype_categoryAllOther == "Sst Gaba_13" ~ "0226 Sst Gaba_13",
    supertype_categoryAllOther == "Sst Gaba_14" ~ "0227 Sst Gaba_14",
    supertype_categoryAllOther == "Sst Gaba_15" ~ "0228 Sst Gaba_15",
    supertype_categoryAllOther == "Sst Gaba_16" ~ "0229 Sst Gaba_16",
    supertype_categoryAllOther == "Sst Gaba_17" ~ "0230 Sst Gaba_17",
    supertype_categoryAllOther == "Sst Gaba_18" ~ "0231 Sst Gaba_18",
    supertype_categoryAllOther == "Sst Gaba_19" ~ "0232 Sst Gaba_19",
    TRUE ~ supertype_categoryAllOther)) %>%
  filter(!supertype_categoryAllOther %in% c("Sst Chodl Gaba_1", "Sst Chodl Gaba_2", "Sst Chodl Gaba_3", "Sst Chodl Gaba_4", "Sst Chodl Gaba_5", "Sst Chodl Gaba_6")) %>%
  mutate(group = "P56") %>% group_by(group, supertype_categoryAllOther) %>% summarise(count = n()) %>% mutate(proportion = count / sum(count)) %>% ungroup()
plot_data <- bind_rows(plot_data_v2, plot_data_Yao)
ggplot(plot_data, aes(x = group, y = proportion, fill = supertype_categoryAllOther)) +
  geom_bar(stat = "identity", position = "stack") +
  labs( title = "Proportion of Supertypes by Group", x = "Group", y = "Proportion", fill = "Supertype") +
  scale_fill_manual(values = c("#81c3d7","#3C3CF0","#3f8efc","#001d3d","#00a7e1","#6a994e","#A01061","#B23C25","#2a6f97","#6E2DB7","#52b69a","#ffa69e",
                               "#5111B2","#B58C00","#d283ff","#2E004B","#A45F10","#007f5f","#006400","grey")) +
  theme_minimal()

```

# Ext Fig 6 ####
```{r}
## A ####
Yao <- readRDS('Yao2021_Sst_Chodl_MapMyCell_to_Yao2023_supertypes.rds')
DimPlot(Yao, group.by= 'cluster_label')
DimPlot(Yao, group.by= 'supertype_name_Yao2023')

## D-H ####
#For accuracy calculation, source the file 'Compute_AccuracyMatrix_RF.R) and install Seurat v4
remotes::install_version('SeuratObject','4.1.4', repos = c('https://satijalab.r-universe.dev', getOption('repos')))

atlas<-readRDS('Integrated_Atlas_V2_FinalClusters_and_MapMyCell.RDS')
gene_list<- c('Nos1', 'Sst', 'Mafb') #or Dach1

P5_samples<-c("BaseAtlas_P5","BaseAtlas_lim_P5_fixed_sorted","P5_WT1_Lim1","P5_WT23_Lim2")
atlas <- subset(atlas,subset=sample %in% P5_samples)
atlas <- subset(atlas,subset=final_clusters_renamed %!in% c("HIn1","HIn2","LRP2.2"))
atlas <- subset(atlas,subset=major_label_transferAnchors_BaseAtlas == "LRP")
LRP_sub<-CreateSeuratObject(atlas@assays$RNA@counts[gene_list,])
LRP_sub<-AddMetaData(LRP_sub, atlas@meta.data[colnames(LRP_sub),])

accuracy_matrix<-ComputeClassificationAccuracy(LRP_sub, "final_clusters_renamed")
breaks <-  seq(0, 0.5, length.out=1001);
cols <- colorRampPalette(colors = c('black','#EE9A3A'))(1000)
gplots::heatmap.2(accuracy_matrix,
                  margins=c(7,7),
                  key = FALSE, # Explicitly set key to TRUE
                  keysize=1,
                  key.xlab="",
                  key.title="Accuracy",
                  trace = "none",
                  density.info = "none",
                  # density.info = "density",
                  col = cols,
                  breaks = breaks,
                  offsetRow=0.1,
                  offsetCol=0.1,
                  cexRow = 0.8,
                  cexCol = 0.8,
                  cellnote = accuracy_matrix,
                  notecex = 0.5,
                  notecol = 'white',
                  Colv = F,
                  Rowv = F,
                  dendrogram = "none")

## G-Dach1 ####
data <- read.csv("Dach1_GFP_quantification.csv")
data$Stage <- factor(data$Stage, levels = c('P5',"P7", "P21"))
data$Celltype <- factor(data$Celltype, levels = c('YFP-Dach1-low',"YFP-Dach1-high"))

summary_data2 <- data %>%
  group_by(Stage, Celltype) %>%
  summarise(total_count = sum(cellcount), .groups = 'drop') %>%
  group_by(Stage) %>%
  mutate(relative_count = total_count / sum(total_count) * 100)
# Plotting the relative distribution as a barplot
ggplot(summary_data2, aes(x = Stage, y = relative_count, fill = Celltype)) +
  geom_bar(stat = "identity", position = "stack") +
  labs(title = "Relative Distribution of LRP1/LRP2 by Stage",
       y = "Relative Percentage (%)",
       x = "Region") +
  theme_minimal() +
  scale_fill_manual(values = c("YFP-Dach1-high" = "#51B885", "YFP-Dach1-low" = "#81D2E7")) +
  geom_text(aes(label = total_count), position = position_stack(vjust = 0.5), color = "black", size = 6)+
  geom_text(aes(label = paste0(round(relative_count, 1), "%")), position = position_stack(vjust = 0.9), color = "black", size = 4) 

## K-Mafb ####
data <- read.csv("Mafb__GFP.csv")
data$Stage <- factor(data$Stage, levels = c('P5',"P7", "P21"))
data$Celltype <- factor(data$Celltype, levels = c('YFP-MafB-high',"YFP-MafB-low"))

summary_data2 <- data %>%
  group_by(Stage, Celltype) %>%
  summarise(total_count = sum(cellcount), .groups = 'drop') %>%
  group_by(Stage) %>%
  mutate(relative_count = total_count / sum(total_count) * 100)
# Plotting the relative distribution as a barplot
ggplot(summary_data2, aes(x = Stage, y = relative_count, fill = Celltype)) +
  geom_bar(stat = "identity", position = "stack") +
  labs(title = "Relative Distribution of LRP1/LRP2 by Stage",
       y = "Relative Percentage (%)",
       x = "Region") +
  theme_minimal() +
  scale_fill_manual(values = c("YFP-MafB-high" = "#81D2E7", "YFP-MafB-low" = "#51B885")) +
  geom_text(aes(label = total_count), position = position_stack(vjust = 0.5), color = "black", size = 6)+
  geom_text(aes(label = paste0(round(relative_count, 1), "%")), position = position_stack(vjust = 0.9), color = "black", size = 4) 


## L ###
Wu_IN <- readRDS('Wu_P14_interneurons_053subclass.rds')
meta <- Wu_IN@meta.data

meta <- meta %>%
  mutate(orig.ident_combined = case_when(
    orig.ident %in% c("GSM8409503_P14MGE_BaxcKO_rep1", "GSM8409504_P14MGE_BaxcKO_rep2") ~ "BaxcKO",
    TRUE ~ orig.ident  # Keep other conditions as-is
  ))

proportions <- meta %>%
  group_by(orig.ident_combined, supertype_name) %>%
  summarise(count = n(), .groups = "drop") %>%
  group_by(orig.ident_combined) %>%
  mutate(proportion = count / sum(count) * 100)
proportions$orig.ident_combined <- factor(proportions$orig.ident_combined, levels = c('GSM8409502_P14MGE_BaxcHET',"BaxcKO"))

ggplot(proportions, aes(x = supertype_name, y = proportion, fill = orig.ident_combined)) +
  geom_bar(stat = "identity", position = "dodge") +
  theme_classic() +
  scale_fill_manual(values = c("GSM8409502_P14MGE_BaxcHET" = "lightgray", "BaxcKO" = "black"))+
  labs(
    x = "Supertype",
    y = "Relative Proportion (%)",
    fill = "Condition"
  ) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))


## G ####
Bershteyn <- readRDS('Bershteyn')
DimPlot(Bershteyn, group.by = 'Stage', cols = c("grey", '#28BAA3','#C574AF','#F067A6','#2AB45D','#489DD7','#0077B6','#00B9DE'))

```

