---
title: "Figures_script"

---
## Packages to install
```{r}
install.packages("rmarkdown")
install.packages("BiocManager")
BiocManager::install("circlize")
BiocManager::install("dittoSeq")


install.packages("forcats")
install.packages("readr")
install.packages("caret")
install.packages("devtools")

#usethis::create_github_token() #to generate a GitHub token
Sys.setenv(GITHUB_PAT = "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")
devtools::install_github("crazyhottommy/scclusteval")

BiocManager::install("ComplexHeatmap")
BiocManager::install("MAST")
```

## Install libraries
```{r}
library(forcats)
library(readr)
library(Seurat)
library(ggplot2)
library(dplyr)
library(pheatmap)
library(reshape2)
library(tibble)
library(dittoSeq)
library(scclusteval)
library(ComplexHeatmap)
library(MAST)
library(tidyr)
library(ggforce)
library(ssclusteval)
```

# Fig 1 ####
```{r}
## B ####
data <- read_excel('Dataset_table.xlsx')
ggplot(data, aes(x = reorder(Dataset, -SST_cells), y = SST_cells, fill = Dataset == "DS17")) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = c("FALSE" = "gray", "TRUE" = "skyblue")) +
  geom_text(aes(label = SST_cells), vjust = -0.5, size = 3) +
  geom_hline(yintercept = 1000, linetype = "dashed", color = "red", size = 0.8) +
  labs(x = "Dataset number", y = "SST+ cells") +
  theme_minimal() +
  theme(legend.position = "none",
        axis.text.x = element_text(angle = 45, hjust = 1))
rm(data)

## C ####
umap_data <- read.csv("UMAP_Metadata.csv")

#Samples
ggplot(umap_data, aes(x = UMAP_1, y = UMAP_2, color = sample)) +
  geom_point(size = 0.1, alpha = 0.8) +
  scale_color_manual(values = c("#1096C5","#6158A6", '#F19B87','#9D1F2B')) +
  theme_minimal()
#Major clusters
ggplot(umap_data, aes(x = UMAP_1, y = UMAP_2, color = major_cluster_label_trained_with_all)) +
  geom_point(size = 0.1, alpha = 0.8) +
  scale_color_manual(values = c("#20B1BD", "#6C6DAF", "#DA5E6B")) +
  theme_minimal()
#Minor clusters
ggplot(umap_data, aes(x = UMAP_1, y = UMAP_2, color = cluster_label_trained_with_all)) +
  geom_point(size = 0.1, alpha = 0.8) +
  scale_color_manual(values = c('#5BBFAB','#33C0D4',"#1295C7","#1E2D59", '#2A2768',
                                '#644E9F','#8459A4','#BB5BA2','#EC697D','#F09F8A',
                                '#DF5E36','#9E1D20')) +
  theme_minimal()


## D ####
umap_filtered <- umap_data %>% filter(subclass_bootstrapping_probability >= 0.9)

#Subclass
ggplot(umap_filtered, aes(x = UMAP_1, y = UMAP_2, color = Subclass)) +
  geom_point(size = 0.1, alpha = 0.8) +
  scale_color_manual(values = c("#059E74","#F0E443", '#E6A024','#5BB4E5', '#D36127')) +
  theme_minimal()
#Probability
ggplot(umap_data, aes(x = UMAP_1, y = UMAP_2, color = subclass_bootstrapping_probability)) +
  geom_point(size = 0.1, alpha = 0.8) +
  scale_color_gradient(low = "white", high = "#3953A4") +
  theme_minimal()

## E ####
stacked_bar_data <- read_csv("Stacked_Bar_Data.csv")
ggplot(stacked_bar_data, aes(x = "Atlas-v0", y = Fraction, fill = subclass_name)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = c("#E6A024", "#5BB4E5", "#059E74", "#F0E443")) +  # Use specific colors
  theme_minimal() +
  labs(title = "Fraction of Cells by Subclass",
       x = NULL,  # Remove x-axis label
       y = "Fraction of cells",
       fill = "Subclass") +
  theme(axis.text.x = element_blank(),  # Hide x-axis text
        axis.ticks.x = element_blank())


rm(list = c('metadata','stacked_bar_data','subclass_data','subclass_distribution','umap_coords','umap_data','umap_filtered'))
gc()

```

# Fig 2 ####
```{r}
## B ####
data <- read_csv('DS_anchors.csv')

ggplot(data, aes(x = reorder(dataset_id, -Anchors), y = Anchors)) +
  stat_summary_bin(aes(y = Anchors), fun = "mean", geom = "bar")+
  stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.2, color = "black")+
  geom_jitter()+
  geom_hline(yintercept = 2000, linetype = "dashed", color = "red")

## C ####
#it will generate a .pdf
data<-read.table("data.txt", row.names = 1, header = TRUE)

files <-dir()
files_anchors <- files[grep("Number_of_anchors_",files)]
files_kBET <- files[grep("kBET_batch_estimate_subset_bothDatasets_",files)]
files_kBET<-files_kBET[grep(".RDS",files_kBET)]

rejection_rate<-c()
mean_anchors <- c()
mean_anchors_scaled <- c()
for (i in rownames(data)){
  n_anchors <- read.table(paste0(files_anchors[grep(i,files_anchors)]), row.names = 1)
  mean_anchors <- c(mean_anchors,mean(n_anchors[,1]))
  mean_anchors_scaled <- c(mean_anchors_scaled,mean(n_anchors[,1])/data[i,2])
  kbet<-readRDS(paste0(files_kBET[grep(i,files_kBET)]))
  rejection_rate<-c(rejection_rate,median(kbet$stats$kBET.observed))
}
colors_ditto<-dittoColors()

files_knn <- files[grep("Percentages_of_cells_in_",files)]
files_knn<-files_knn[grep(".RDS",files_knn)]
pdf('plot_percentages_of_cells_with_mixed_neighborhood_in_kNN_allSamples.pdf', width=8)
par(cex=0.5,cex.lab=1.2)
i=1
knn_perc<-readRDS(paste0(files_knn[grep(rownames(data)[i],files_knn)]))
plot(seq(5,200,5),knn_perc, xlab="k",ylab="fraction of cells with mixed neighborhood", pch=0,cex=0.8,col=colors_ditto[i], ylim=c(0,1.05),xaxt='n',yaxt='n')
legend("top",legend=paste0(data[,1]," - ",rownames(data[,])),fill=colors_ditto,horiz = FALSE,cex=0.915,pch = c(0:12),ncol=6)
axis(2,cex.axis=1.5,)
axis(1,cex.axis=1.5)
print(lines(seq(5,200,5),knn_perc,col=colors_ditto[i]))
for (i in 2:nrow(data)){
  knn_perc<-readRDS(paste0(files_knn[grep(rownames(data)[i],files_knn)]))
  print(points(seq(5,200,5),knn_perc, pch=i-1,col=colors_ditto[i]))
  print(lines(seq(5,200,5),knn_perc,col=colors_ditto[i]))
}
abline(v=60,col="blue4",lty=2)
abline(h=0.45,col="red3",lty=2)
dev.off()

knn_perc_k60<-c()
for (i in 1:nrow(data)){
  knn_perc<-readRDS(paste0(files_knn[grep(rownames(data)[i],files_knn)]))
  knn_perc_k60<-c(knn_perc_k60,knn_perc["60"])
}

## D ####
#it will generate a .pdf
data_plot1<-data.frame(data, mean_anchors = mean_anchors, kBET_rejection_rate=rejection_rate, knn_perc_k60=knn_perc_k60)
pdf('Mean_ancors_vs_k60_fraction_of_cells.pdf')
plot(data_plot1$knn_perc_k60, data_plot1$mean_anchors, col=colors_ditto, pch=c(0:12),cex=1.5,lwd=2, xlab ="fraction of cells with mixed neighborhood", ylab="Mean anchors")
legend("topleft",legend=paste0(data[,1]," - ",rownames(data[,])),fill=colors_ditto,horiz = FALSE,cex=0.6,pch = c(0:12))
abline(v=0.45,col="red3",lty=2)
abline(h=2000,col="red3",lty=2)
dev.off()

## E ####
#it will generate a .pdf
data_plot1<-data.frame(data,mean_anchors = mean_anchors, mean_anchors_scaled = mean_anchors_scaled, kBET_rejection_rate=rejection_rate)
pdf('Mean_anchors_vs_kBET_RejectionRate.pdf')
plot(data_plot1$kBET_rejection_rate, data_plot1$mean_anchors, col=colors_ditto, pch=c(0:12),cex=1.5,lwd=2, xlab ="kBET rejection rate", ylab="Mean anchors")
legend("topright",legend=paste0(data[,1]," - ",rownames(data)),fill=colors_ditto,horiz = FALSE,cex=0.6,pch = c(0:12))
abline(v=0.5,col="red3",lty=2)
abline(h=2000,col="red3",lty=2)
dev.off()

rm(list = c("data", "data_plot1", "kbet", "n_anchors", "colors_ditto", 
            "files", "files_anchors", "files_kBET", "files_knn", "i", 
            "knn_perc", "knn_perc_k60", "mean_anchors", "mean_anchors_scaled", 
            "rejection_rate"))
gc()
```

# Fig 3 ####
```{r}
## A ####
data <- read_excel('Sequencing_depth.xlsx')
data <- data %>% mutate(Reads_in_Thousands = Reads / 1000)
means <- data %>% group_by(Sequencing) %>% summarise(Mean_Reads = mean(Reads_in_Thousands))
ggplot() +
  geom_bar(data = means, aes(x = Sequencing, y = Mean_Reads, fill = Sequencing),
           stat = "identity", width = 0.6, alpha = 0.8) +
  geom_point(data = data, aes(x = Sequencing, y = Reads_in_Thousands),
             color = "black", size = 3, position = position_jitter(width = 0.1, height = 0)) +
  scale_fill_manual(values = c("shallow" = "gray", "deep" = "darkgreen")) +
  labs(title = "Number of Reads per Cell",
       x = "",
       y = "Number of reads per cell (*10Â³)") +
  theme_minimal() +
  theme(legend.position = "none")

## C ####
#it will generate a .pdf
data<-read.table("data.txt", row.names = 1, header = TRUE)

files <-dir()
files_anchors <- files[grep("Number_of_anchors_",files)]
files_kBET <- files[grep("kBET_batch_estimate_subset_bothDatasets_",files)]
files_kBET<-files_kBET[grep(".RDS",files_kBET)]

rejection_rate<-c()
mean_anchors <- c()
mean_anchors_scaled <- c()
for (i in rownames(data)){
  n_anchors <- read.table(paste0(files_anchors[grep(i,files_anchors)]), row.names = 1)
  mean_anchors <- c(mean_anchors,mean(n_anchors[,1]))
  mean_anchors_scaled <- c(mean_anchors_scaled,mean(n_anchors[,1])/data[i,2])
  kbet<-readRDS(paste0(files_kBET[grep(i,files_kBET)]))
  rejection_rate<-c(rejection_rate,median(kbet$stats$kBET.observed))
}
colors_ditto<-dittoColors()

files_knn <- files[grep("Percentages_of_cells_in_",files)]
files_knn<-files_knn[grep(".RDS",files_knn)]
pdf('plot_percentages_of_cells_with_mixed_neighborhood_in_kNN_allSamples.pdf', width=8)
par(cex=0.5,cex.lab=1.2)
i=1
knn_perc<-readRDS(paste0(files_knn[grep(rownames(data)[i],files_knn)]))
plot(seq(5,200,5),knn_perc, xlab="k",ylab="fraction of cells with mixed neighborhood", pch=0,cex=0.8,col=colors_ditto[i], ylim=c(0,1.05),xaxt='n',yaxt='n')
legend("top",legend=paste0(data[,1]," - ",rownames(data[,])),fill=colors_ditto,horiz = FALSE,cex=0.915,pch = c(0:12),ncol=6)
axis(2,cex.axis=1.5,)
axis(1,cex.axis=1.5)
print(lines(seq(5,200,5),knn_perc,col=colors_ditto[i]))
for (i in 2:nrow(data)){
  knn_perc<-readRDS(paste0(files_knn[grep(rownames(data)[i],files_knn)]))
  print(points(seq(5,200,5),knn_perc, pch=i-1,col=colors_ditto[i]))
  print(lines(seq(5,200,5),knn_perc,col=colors_ditto[i]))
}
abline(v=60,col="blue4",lty=2)
abline(h=0.45,col="red3",lty=2)
dev.off()

knn_perc_k60<-c()
for (i in 1:nrow(data)){
  knn_perc<-readRDS(paste0(files_knn[grep(rownames(data)[i],files_knn)]))
  knn_perc_k60<-c(knn_perc_k60,knn_perc["60"])
}

## D ####
#it will generate a .pdf
data_plot1<-data.frame(data, mean_anchors = mean_anchors, kBET_rejection_rate=rejection_rate, knn_perc_k60=knn_perc_k60)
pdf('Mean_ancors_vs_k60_fraction_of_cells.pdf')
plot(data_plot1$knn_perc_k60, data_plot1$mean_anchors, col=colors_ditto, pch=c(0:12),cex=1.5,lwd=2, xlab ="fraction of cells with mixed neighborhood", ylab="Mean anchors")
legend("topleft",legend=paste0(data[,1]," - ",rownames(data[,])),fill=colors_ditto,horiz = FALSE,cex=0.6,pch = c(0:12))
abline(v=0.45,col="red3",lty=2)
abline(h=2000,col="red3",lty=2)
dev.off()

## E ####
#it will generate a .pdf
data_plot1<-data.frame(data,mean_anchors = mean_anchors, mean_anchors_scaled = mean_anchors_scaled, kBET_rejection_rate=rejection_rate)
pdf('Mean_anchors_vs_kBET_RejectionRate.pdf')
plot(data_plot1$kBET_rejection_rate, data_plot1$mean_anchors, col=colors_ditto, pch=c(0:12),cex=1.5,lwd=2, xlab ="kBET rejection rate", ylab="Mean anchors")
legend("topright",legend=paste0(data[,1]," - ",rownames(data)),fill=colors_ditto,horiz = FALSE,cex=0.6,pch = c(0:12))
abline(v=0.5,col="red3",lty=2)
abline(h=2000,col="red3",lty=2)
dev.off()


rm(list = c("data", "data_plot1", "kbet", "n_anchors", "means", "colors_ditto", 
            "files", "files_anchors", "files_kBET", "files_knn", "i", 
            "knn_perc", "knn_perc_k60", "mean_anchors", "mean_anchors_scaled", 
            "rejection_rate"))
gc()

```

# Fig 4 ####
```{r}
## A ####
#Load the data
umap_data <- read.csv("UMAP_Metadata_Fig4a.csv")
# Plot by sample
ggplot(umap_data, aes(x = UMAP_1, y = UMAP_2, color = Sample_Group)) + geom_point(size = 0.1, alpha = 0.8) +
  scale_color_manual(values = c("grey",'#2AB45D','#28BAA3','#489DD7','#0077B6','#00B9DE','#C574AF','#F067A6')) +
  theme_minimal()
# Plot by Major group
ggplot(umap_data, aes(x = UMAP_1, y = UMAP_2, color = Major)) +
  geom_point(size = 0.1, alpha = 0.8) +
  scale_color_manual(values = c("#20B1BD", "#6C6DAF", "#DA5E6B")) +
  theme_minimal()
# Plot by Minor group
ggplot(umap_data, aes(x = UMAP_1, y = UMAP_2, color = Minor)) +
  geom_point(size = 0.1, alpha = 0.8) +
  scale_color_manual(values = c('#5BBFAB','#33C0D4',"#1295C7","#1E2D59", '#2A2768','#644E9F','#8459A4','#BB5BA2','#EC697D','#F09F8A','#DF5E36','#9E1D20')) +
  theme_minimal()

## B ####
# Data for accuracy values
data <- data.frame(
  Row = c("LRP", "LRP", "LRP", "Martinotti", "Martinotti", "Martinotti", "Non-Martinotti", "Non-Martinotti", "Non-Martinotti"),
  Column = c("LRP", "Martinotti", "Non-Martinotti", "LRP", "Martinotti", "Non-Martinotti", "LRP", "Martinotti", "Non-Martinotti"),
  Accuracy = c(0.954, 0.027, 0.018, 0.019, 0.926, 0.055, 0.005, 0.059, 0.926))  # Accuracy values from the plot
  
# Create the count matrix from accuracy values
count_matrix <- reshape2::dcast(data, Row ~ Column, value.var = "Accuracy")
  
#Plot
  ggplot(data, aes(x = Column, y = Row, fill = Accuracy)) +
    geom_tile(color = "black") +
    geom_text(aes(label = round(Accuracy, 3)), color = "white", size = 6) +
    scale_fill_gradient(low = "black", high = "orange", limits = c(0, 1)) +
    labs(title = NULL, x = NULL, y = NULL) +
    theme_minimal() +
    theme(axis.text = element_text(size = 12))
  
## C ####
#Load the data
umap_data <- read.csv("UMAP_Metadata_Fig4c.csv")
# Plot by sample
ggplot(umap_data, aes(x = UMAP_1, y = UMAP_2, color = metadata)) + geom_point(size = 0.1, alpha = 0.8) +
  scale_color_manual(values = c("#059E74","#F0E443", '#E6A024','#5BB4E5', '#D36127')) +
  theme_minimal()

## D ####
stacked_bar_data <- read.csv("BarPlot_Fig4d.csv")
stacked_bar_data$Subclass <- factor(stacked_bar_data$Subclass, levels = c("Other", "052 Pvalb Gaba", "045 OB-STR-CTX Inh IMN", "056 Sst Chodl Gaba", "053 Sst Gaba"))
# Plot with ggplot
ggplot(stacked_bar_data, aes(x = Version, y = percentage, fill = Subclass)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = c("grey", "#F0E443", "#059E74", "#5BB4E5", "#E6A024")) +  # Use specific colors
  theme_minimal() +
  labs(title = "Fraction of Cells by Subclass",
       x = "Version",  # Remove x-axis label
       y = "Fraction of cells",
       fill = "Subclass")

## E ####
#Load the data
umap_data <- read.csv("UMAP_Metadata_Fig4e.csv")
# Plot by sample
ggplot(umap_data, aes(x = UMAP_1, y = UMAP_2, color = metadata)) + geom_point(size = 0.1, alpha = 0.8) +
  scale_color_manual(values = c("#CCCCCC", "#989898",
                                "#2EBAD1", "#5BC2AE", "#60BD6E",
                                "#001d3d","#2a6f97", "#0d00a4", "#3C3CF0", "#3f8efc", "#449dd1", "#00a7e1", "#7fc8f8", "#81c3d7", "#1B1B75", "#023e8a", 
                                "#6E2DB7", "#d283ff", 
                                "#52b69a", "#007f5f", "#006400", "#6a994e", "#7cb518", "#a5be00", 
                                "#F0B572", "#e9c46a", 
                                "#A45F10", 
                                "#D14545", 
                                "#AC4758", 
                                "#ea698b", "#ffacc5", "#ff5d8f", "#c05299", "#ffa69e", "#f7cad0")) +
  theme_minimal()


rm(list = c("count_matrix", "data", "stacked_bar_data", "umap_data"))
gc()
```

# Fig 5 ####
```{r}
## A ####
# Load the data
umap_data <- read.csv("UMAP_Metadata_Fig5a.csv")
# Plot Martinotti Cells
ggplot(umap_data, aes(x = UMAP_1, y = UMAP_2, color = Major)) + geom_point(size = 0.1, alpha = 0.8) +
  scale_color_manual(values = c("#8D8DC1", "grey")) +
  theme_minimal()
# Plot Developmental Martinotti clusters
ggplot(umap_data, aes(x = UMAP_1, y = UMAP_2, color = final_clusters_renamed)) + geom_point(size = 0.1, alpha = 0.8) +
  scale_color_manual(values = c("#001d3d","#2a6f97", "#0d00a4", "#3C3CF0", "#3f8efc", "#449dd1", "#00a7e1", "#7fc8f8", "#81c3d7", "#1B1B75", "#023e8a", "#6E2DB7", "#d283ff", "#52b69a", "#007f5f", "#006400", "#6a994e", "#7cb518", "#a5be00", "grey","grey", "grey", "grey", "grey", "grey", "grey", "grey", "grey", "grey", "grey")) +
  labs(color = "Developmental clusters") +
  theme_minimal()
# Plot Adult supertypes matches
umap_data <- read.csv("UMAP_Metadata_Fig5a_Adult.csv")
ggplot(umap_data, aes(x = UMAP_1, y = UMAP_2, color = metadata)) + geom_point(size = 0.1, alpha = 0.8) +
  scale_color_manual(values = c("#81c3d7", "#3C3CF0", "#3f8efc", "#001d3d", "#00a7e1", "#6a994e", "#A01061", "#B23C25", "#2a6f97", "#6E2DB7", "#52b69a",
                                "#ffa69e", "#5111B2", "#B58C00", "#d283ff", "#2E004B", "#A45F10", "#007f5f", "#006400",
                                "grey50", "grey")) +
  labs(color = "Adult supertypes") +
  theme_minimal()

## B ####
'%!in%' <- function(x,y)!('%in%'(x,y))
# Load the data
data <- read.csv("RiverPlotMC_Fig5b.csv", row.names = 1)
# Plot river plots from developmental MC clusters to adult supertypes
data$supertype_name <- paste("Sst_",unlist(lapply(as.character(data$supertype_name), function(x) strsplit(x," ")[[1]][3])),sep="")
data2<-data.frame(final_clusters_renamed=rep("MC0.0",2000),supertype_name=rep("Sst_Gaba_0",2000),final_clusters_palette=rep("gray",2000))
data<-rbind(data,data2)
data$final_clusters_renamed <- factor(data$final_clusters_renamed, levels = c("MC2.1","MC1.1","MC3.2","MC3.3","MC3.4","MC1.5","MC1.7","MC1.9","MC0.0","MC1.2","MC3.1","MC1.4","MC3.5","MC2.2","MC1.3","MC3.6","MC1.6","MC1.8","MC1.10","MC1.11"))
data$supertype_name <- factor(data$supertype_name, levels = c("Sst_Gaba_10","Sst_Gaba_4","Sst_Gaba_18","Sst_Gaba_19","Sst_Gaba_6","Sst_Gaba_3","Sst_Gaba_5","Sst_Gaba_1","Sst_Gaba_0","Sst_Gaba_9","Sst_Gaba_11","Sst_Gaba_16","Sst_Gaba_15","Sst_Gaba_13","Sst_Gaba_2","Sst_Gaba_12","Sst_Gaba_7","Sst_Gaba_8","Sst_Gaba_17","Sst_Gaba_14"))
edges  <- data.frame(data$final_clusters_renamed, data$supertype_name)
edges<-edges %>% group_by(across(everything())) %>% dplyr::count()
colnames(edges) <- c("Cluster", "Supertype","Freq")  
edges <- as.data.frame(edges)
tmp<-data[!duplicated(data$final_clusters_renamed),]
color1<-tmp$final_clusters_palette
names(color1)<-tmp$final_clusters_renamed
edges2<-edges
dat_ggforce <- edges2  %>%
  gather_set_data(1:2) %>%        # <- ggforce helper function
  arrange(x,Cluster,desc(Supertype))
p<-ggplot(dat_ggforce, aes(x = x, id = id, split = y, value = Freq)) +
  geom_parallel_sets(aes(fill = Cluster), alpha = 0.45, axis.width = 0.2, n=100, strength = 0.5) +
  geom_parallel_sets_axes(axis.width = 0.05, fill = "white",color = "white", size = 0.15) +
  geom_parallel_sets_labels(colour = 'black', size = 3, angle = 0) +
  scale_fill_manual(values  = color1) + 
  theme(panel.background = element_rect(fill='white', colour='black')) + theme(legend.position="none") + theme(axis.title.x=element_blank(),axis.text.x=element_blank(),axis.ticks.x=element_blank(),axis.title.y=element_blank(),axis.text.y=element_blank(),axis.ticks.y=element_blank())
q<-ggplot_build(p)
tmp<-q$data[[3]]
tmp<-tmp[which(tmp$label %!in% c("Sst_Gaba_0","MC0.0")),]
q$data[[3]] <- tmp
tmp<-q$data[[1]]
tmp[which(tmp$split == "MC0.0"),"alpha"] <-0
q$data[[1]] <-tmp
q <- ggplot_gtable(q)
plot(q)

## C ####
#Load the data
umap_data <- read.csv("UMAP_Metadata_Fig5c.csv")
# Plot non-Martinotti Cells
ggplot(umap_data, aes(x = UMAP_1, y = UMAP_2, color = Major)) + geom_point(size = 0.1, alpha = 0.8) +
  scale_color_manual(values = c("grey", "#E0767F")) +
  theme_minimal()
# Plot Developmental non-Martinotti clusters
ggplot(umap_data, aes(x = UMAP_1, y = UMAP_2, color = final_clusters_renamed)) + geom_point(size = 0.1, alpha = 0.8) +
  scale_color_manual(values = c("grey","grey","grey","grey","grey","grey","grey","grey","grey","grey","grey", "grey","grey", "grey","grey","grey","grey","grey","grey",
                             "#F0B572", "#e9c46a", "#A45F10", "#D14545", "#AC4758", "#ea698b", "#ffacc5", "#ff5d8f", "#c05299", "#ffa69e", "#f7cad0")) +
  labs(color = "Developmental clusters") +
  theme_minimal()
# Plot Adult supertypes matches
umap_data <- read.csv("UMAP_Metadata_Fig5c_Adult.csv")
ggplot(umap_data, aes(x = UMAP_1, y = UMAP_2, color = metadata)) + geom_point(size = 0.1, alpha = 0.8) +
  scale_color_manual(values = c("grey", "#AC4758", "#EA698B","#FFACC9", "#D14545","#FFACC1", "#FF5D8F", "#AC4758", "#A01061", "#F0B572",
                                "#2a6f97", "#F0B411","#52b69a","#ffa69e","#FFDDD5","#FFACC5","#D15545", "#F1C572",
                                "#A45F10", "#AC4AA8", "#AC4AA9", "grey50", "#F0E443", "#F0E443")) +
  labs(color = "Adult Supertypes") +
  theme_minimal()

## D ####
'%!in%' <- function(x,y)!('%in%'(x,y))
# Load the data
data <- read.csv("RiverPlotnMC_Fig5d.csv", row.names = 1)
# Plot river plots from developmental nMC clusters to adult supertypes
data$supertype_name <- paste("Sst_",unlist(lapply(as.character(data$supertype_name), function(x) strsplit(x," ")[[1]][3])),sep="")
data2<-data.frame(final_clusters_renamed=rep("nMC0.0",2000),supertype_name=rep("Sst_Gaba_0",1000),final_clusters_palette=rep("white",2000))
data<-rbind(data,data2)
data$final_clusters_renamed <- factor(data$final_clusters_renamed, levels = c("nMC1.1","nMC4.0","nMC3.0","nMC5.2","nMC5.1","nMC5.3","nMC5.4","nMC5.6","nMC1.2","nMC0.0","nMC2.0","nMC5.5"))
data$supertype_name <- factor(data$supertype_name, levels = c("Sst_Gaba_14","Sst_Gaba_13","Sst_Gaba_1","Sst_Gaba_5","Sst_Gaba_8","Sst_Gaba_10","Sst_Gaba_16","Sst_Gaba_6","Sst_Gaba_18","Sst_Gaba_19","Sst_Gaba_2","Sst_Gaba_4","Sst_Gaba_3","Sst_Gaba_15","Sst_Gaba_0","Sst_Gaba_17","Sst_Gaba_12"))
edges  <- data.frame(data$final_clusters_renamed, data$supertype_name)
edges<-edges %>% group_by(across(everything())) %>% dplyr::count()
colnames(edges) <- c("Cluster", "Supertype","Freq")  
edges <- as.data.frame(edges)
tmp<-data[!duplicated(data$final_clusters_renamed),]
color1<-tmp$final_clusters_palette
names(color1)<-tmp$final_clusters_renamed
edges2<-edges
dat_ggforce <- edges2  %>%
  gather_set_data(1:2) %>%        # <- ggforce helper function
  arrange(x,Cluster,desc(Supertype))
p<-ggplot(dat_ggforce, aes(x = x, id = id, split = y, value = Freq)) +
  geom_parallel_sets(aes(fill = Cluster), alpha = 0.45, axis.width = 0.2,
                     n=100, strength = 0.5) +
  geom_parallel_sets_axes(axis.width = 0.05, fill = "white",
                          color = "white", size = 0.15) +
  geom_parallel_sets_labels(colour = 'black', size = 3, angle = 0) +
  scale_fill_manual(values  = color1) + 
  theme(panel.background = element_rect(fill='white', colour='black')) + theme(legend.position="none") + theme(axis.title.x=element_blank(),axis.text.x=element_blank(),axis.ticks.x=element_blank(),axis.title.y=element_blank(),axis.text.y=element_blank(),axis.ticks.y=element_blank())
q<-ggplot_build(p)
tmp<-q$data[[3]]
tmp<-tmp[which(tmp$label %!in% c("Sst_Gaba_0","nMC0.0")),]
q$data[[3]] <- tmp
tmp<-q$data[[1]]
tmp[which(tmp$split == "nMC0.0"),"alpha"] <- 0
q$data[[1]] <-tmp
q <- ggplot_gtable(q)
plot(q)

## E ####
# Load the data
summary_df <- read.csv("StackedBarMajorType_Fig5e.csv", row.names = 1)
supertype_order <- c("0222 Sst Gaba_9", "0224 Sst Gaba_11", "0220 Sst Gaba_7", "0217 Sst Gaba_4", "0228 Sst Gaba_15", "0223 Sst Gaba_10", "0215 Sst Gaba_2", "0232 Sst Gaba_19", "0231 Sst Gaba_18", "0219 Sst Gaba_6","0226 Sst Gaba_13", "0229 Sst Gaba_16", "0216 Sst Gaba_3", "0225 Sst Gaba_12", "0218 Sst Gaba_5", "0214 Sst Gaba_1", "0221 Sst Gaba_8", "0230 Sst Gaba_17", "0227 Sst Gaba_14")
summary_df$Supertype_Other <- factor(summary_df$Supertype_Other, levels = supertype_order)
# Bar plot
ggplot(summary_df, aes(x = Supertype_Other, y = proportion, fill = major_label_transferAnchors_BaseAtlas)) +
  geom_bar(stat = "identity", position = "stack") +
  scale_fill_manual(values = c("Martinotti" = "#8E8EC1", "Non-Martinotti" = "#E0777F"))+
  labs(title = "Distribution of Supertypes by Main types",
       x = "Adult supertypes",
       y = "Proportion of Cells",
       fill = "Main types") +
  scale_y_continuous(breaks = seq(0, 1, by = 0.2)) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
# Dot plot
ggplot(summary_df, aes(x = Supertype_Other, y = major_label_transferAnchors_BaseAtlas)) +
  geom_point(aes(size = cell_count, color = proportion)) +
  scale_color_gradient(low = "grey", high = "red")+
  labs(x = "Adult Supertypes", y = "Main types", color = "Proportion", size = "Cell Count") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))

## F ####
# Load the data
plot_data <- read.csv("DevData_Fig5f.csv", row.names = 1)
plot_data <- plot_data %>% filter(!group == "P1")%>% mutate(Supertype_Other = factor(Supertype_Other, levels = rev(sort(unique(Supertype_Other), decreasing = FALSE, na.last = TRUE)))) %>% 
  mutate(group = factor(group, levels = rev(sort(unique(group), decreasing = FALSE, na.last = TRUE))))
# Bar plot
ggplot(plot_data, aes(x = Supertype_Other, y = relative_proportion_toP1,  fill = group)) +
  geom_bar(stat = "identity", position = position_dodge()) +
  labs( title = "Proportion of Supertypes by Group", x = "Supertype", y = "Relative Proportion to P1", fill = "Group") +
  scale_fill_manual(values = c("#414042", "#939598")) +
  coord_flip() +
  guides(fill = guide_legend(reverse = TRUE)) +
  theme_minimal()

rm(list = c("dat_ggforce", "data", "data2", "edges", "edges2", "p", "plot_data", "q", "summary_df", "tmp", "umap_data", "color1", "supertype_order", "%!in%"))
gc()

```

# Fig 6 ####
```{r}
## A ####
umap_data <- read.csv("LRP_UMAP_Metadata.csv")
#Developmental clusters
ggplot(umap_data, aes(x = UMAP_1, y = UMAP_2, color = clusters)) +
  geom_point(size = 0.1, alpha = 0.8) +
  scale_color_manual(values = c("#2EBAD1", "#5BC2AE")) +
  theme_minimal() +
  labs(title = "UMAP by Clusters", x = "UMAP 1", y = "UMAP 2")
#Adults supertypes
ggplot(umap_data, aes(x = UMAP_1, y = UMAP_2, color = supertype_name)) +
  geom_point(size = 0.1, alpha = 0.8) +
  scale_color_manual(values = c("grey", "grey", "grey", "grey", "grey", 
                                "grey", "grey", "grey", "grey", "#AC4758", 
                                "grey", "#2EBAD1", "grey", "grey", "grey")) +
  theme_minimal() +
  labs(title = "UMAP by Supertype Name", x = "UMAP 1", y = "UMAP 2")



## B ####
#More sections and supertypes distribution can be found here:
# https://knowledge.brain-map.org/abcatlas#AQEBSlpQSDhSMUg2MEM0MU4wMlVCMgACT0E3SjJLQlZOREwyMjBORE1LOQADAAQBAAKE3AUMhM4qGwOFrEnShbykFwAFAAYBAQJGUzAwRFhWMFQ5UjFYOUZKNFFFAAN%2BAAAABAAACEJBNURQNzNVUjZaNEJaTkY5OEgACTVDMDIwMUpTVkUwNFdZNkRNVkMACgALAW5vbmUAAm5vbmUAAwEEAQACIzAwMDAwMAADyAEABQEBAiMwMDAwMDAAA8gBAAAAAgEA

spatial_data <- read.csv("spatial_data_Zhuang-ABCA-2.027.csv", row.names = 1)
ggplot(spatial_data, aes(x, y)) + 
  geom_point(colour = "gray90", size = 0.6) +  # Background points
  theme_void() +
  geom_point(data = spatial_data[spatial_data$supertype_label %in% supertype,], aes(x, y, colour = supertype_label)) +  
  scale_colour_manual(values = c("Sst Chodl Gaba_2" = "red", "Sst Chodl Gaba_4" = "blue")) +  # Define colors for supertypes
  scale_y_reverse() + 
  ggtitle(paste0("Section: ", section, ", Supertypes: ", paste(supertype, collapse = " and "))) + 
  theme(plot.title = element_text(hjust = 1))


## C ####
plot_data <- read_csv("LRP_Pseudotime_Data.csv")

#Pseudotime
ggplot(plot_data, aes(x = DC1, y = DC2, color = Pseudotime)) +
  geom_point(size = 1) +
  scale_color_viridis_c(option = "plasma") +
  labs(title = "Pseudotime Plot on DC1-DC2", x = "DC1", y = "DC2") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5))

#Stage
ggplot(plot_data, aes(x = DC1, y = DC2, color = time_point)) +
  geom_point(size = 1) +
  scale_color_manual(values = c('#F067A6', '#7FD2E8', '#489DD7')) +
  labs(title = "Time Point Plot on DC1-DC2", x = "DC1", y = "DC2") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5))

#Clusters
ggplot(plot_data, aes(x = DC1, y = DC2, color = clusters)) +
  geom_point(size = 1) +
  scale_color_manual(values = c("#2EBAD1", "#5BC2AE")) +
  labs(title = "Cluster Plot on DC1-DC2", x = "DC1", y = "DC2") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5))

## D ####
#LRP1 genes
LRP_subset <- readRDS('LRP1_genes.RDS')
genes <- c("Mafb", "Crhbp", "Lingo1","Car10","Igf1","Lypd6b","Ncam2","Gad2","Oxtr",'Clmp') 
#LRP2 genes
LRP_subset <- readRDS('LRP2_genes.RDS') 
genes <- c("Cited1","Hhip","Igsf21","Mgll","Selenom","Atp2b4","Brinp1","Elmod1","Rarb", 'Hs6st2') 
#Common genes
LRP_subset <- readRDS('Common_genes.RDS') 
genes <- c("Sox4", "Olfm3", "Cntnap4")

#Arrange data
plot_data <- FetchData(LRP_subset, vars = c("Pseudotime", "clusters", genes))
cluster1_data <- plot_data %>% filter(clusters == "LRP1.0") %>% arrange(Pseudotime)
cluster2_data <- plot_data %>% filter(clusters == "LRP2.1") %>% arrange(Pseudotime)
ordered_data <- rbind(cluster1_data, cluster2_data)
heatmap_data <- as.matrix(ordered_data[, genes, drop = FALSE])
heatmap_data <- t(heatmap_data)  # Transpose so that genes are rows and cells are columns
rownames(heatmap_data) <- genes
colnames(heatmap_data) <- rownames(ordered_data)
annotation_col <- data.frame(Cluster = ordered_data$clusters, Pseudotime = ordered_data$Pseudotime)
rownames(annotation_col) <- rownames(ordered_data)

pheatmap(
  heatmap_data,
  cluster_rows = FALSE,            # Keep gene order as specified
  cluster_cols = FALSE,            # Avoid reordering cells
  annotation_col = annotation_col, # Color columns by cluster and pseudotime
  gaps_col = nrow(cluster1_data),  # Add gap between clusters
  color = colorRampPalette(colors = c("#110807", "#926026", "#EE9A3A", "#E5CC4D"))(2000),  # Custom color palette
  main = "Gene Expression Across Clusters by Pseudotime"
)

rm(list = c("desired_order", "genes", "section", "supertype", "LRP"))
gc()

```

# Fig 7
```{r}
## A ####
LRP_numbers <- data.frame(
  E16 = c(304, 128),
  P1 = c(1348,   871),
  P5 = c(1980,   2206))
rownames(LRP_numbers) <- c("LRP1", "LRP2.1")
sst_chodl_props <- sweep(LRP_numbers, 2, colSums(LRP_numbers), FUN = "/")
sst_chodl_long <- as.data.frame(sst_chodl_props) %>%
  rownames_to_column(var = "Cell_Type") %>%
  melt(id.vars = "Cell_Type", variable.name = "Time_Point", value.name = "Proportion")
sst_chodl_long$Time_Point <- factor(sst_chodl_long$Time_Point, levels = c("E16",'P1',"P5"))

#plot
ggplot() +
  geom_bar(data = sst_chodl_long, aes(x = Time_Point, y = Proportion, fill = Cell_Type), 
           stat = "identity", position = "fill", alpha = 0.5) +  # Stacked bar plot
  scale_y_continuous(labels = scales::percent) +  # Y-axis as percentage
  labs(title = "Proportion of LRP1 and LRP2.1 Across Time Points",
       y = "Proportion",
       x = "Time Point") +
  scale_fill_manual(values = c("LRP1" = "blue2", "LRP2.1" = "#57C0AC")) +  # Custom colors
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

## B ####
sst_chodl_counts <- data.frame(
  E18_P5_Gao =   c(18, 8),
  P6_P14_Gao =   c(89,  3),
  P15_P56_Gao =   c(57,  2))
rownames(sst_chodl_counts) <- c("LRP1", "LRP2.1")
sst_chodl_props <- sweep(sst_chodl_counts, 2, colSums(sst_chodl_counts), FUN = "/")
sst_chodl_long <- as.data.frame(sst_chodl_props) %>%
  rownames_to_column(var = "Cell_Type") %>%
  melt(id.vars = "Cell_Type", variable.name = "Time_Point", value.name = "Proportion")
sst_chodl_long$Time_Point <- factor(sst_chodl_long$Time_Point, levels = c('E18_P5_Gao','P6_P14_Gao','P15_P56_Gao'))

 
ggplot() +
  geom_bar(data = sst_chodl_long, aes(x = Time_Point, y = Proportion, fill = Cell_Type), 
           stat = "identity", position = "fill", alpha = 0.5) +  # Stacked bar plot
  scale_y_continuous(labels = scales::percent) +  # Y-axis as percentage
  labs(title = "Proportion of LRP1 and LRP2.1 Across Time Points",
       y = "Proportion",
       x = "Time Point") +
  scale_fill_manual(values = c("LRP1" = "blue2", "LRP2.1" = "#57C0AC")) +  # Custom colors
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

## F ####
LRP_numbers <- data.frame(
  P5 = c(1401, 1126),
  P7 = c(1025,   371),
  P21 = c(1701,   373))
rownames(LRP_numbers) <- c("LRP1", "LRP2.1")
sst_chodl_props <- sweep(LRP_numbers, 2, colSums(LRP_numbers), FUN = "/")
sst_chodl_long <- as.data.frame(sst_chodl_props) %>%
  rownames_to_column(var = "Cell_Type") %>%
  melt(id.vars = "Cell_Type", variable.name = "Time_Point", value.name = "Proportion")
sst_chodl_long$Time_Point <- factor(sst_chodl_long$Time_Point, levels = c("P5",'P7',"P21"))


#plot
ggplot() +
  geom_bar(data = sst_chodl_long, aes(x = Time_Point, y = Proportion, fill = Cell_Type), 
           stat = "identity", position = "fill", alpha = 0.5) +  # Stacked bar plot
  scale_y_continuous(labels = scales::percent) +  # Y-axis as percentage
  labs(title = "Proportion of LRP1 and LRP2.1 Across Time Points",
       y = "Proportion",
       x = "Time Point") +
  scale_fill_manual(values = c("LRP1" = "blue2", "LRP2.1" = "#57C0AC")) +  # Custom colors
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

## G ###
sst_chodl_counts <- data.frame(
  Wu_P2 = c(49,  21),
  Wu_P7 = c(119, 21),
  Wu_P20 =  c(113,  1),
  Wu_P14wt =    c(108,  1),
  Wu_P14_Bax_ko =    c(151,  55))
rownames(sst_chodl_counts) <- c("LRP1", "LRP2.1")
sst_chodl_props <- sweep(sst_chodl_counts, 2, colSums(sst_chodl_counts), FUN = "/")
verall_proportion <- sst_chodl_long %>% group_by(Time_Point) %>% summarize(Proportion = sum(Proportion), .groups = 'drop') 
sst_chodl_long <- as.data.frame(sst_chodl_props) %>% rownames_to_column(var = "Cell_Type") %>%
  melt(id.vars = "Cell_Type", variable.name = "Time_Point", value.name = "Proportion")
sst_chodl_long$Time_Point <- factor(sst_chodl_long$Time_Point, levels = c('Wu_P2',"Wu_P7", "Wu_P20", "Wu_P14wt", "Wu_P14ko"))

#plot
ggplot() +
  geom_bar(data = sst_chodl_long, aes(x = Time_Point, y = Proportion, fill = Cell_Type), 
           stat = "identity", position = "fill", alpha = 0.5) +  # Stacked bar plot
  scale_y_continuous(labels = scales::percent) +  # Y-axis as percentage
  labs(title = "Proportion of LRP1 and LRP2.1 Across Time Points",
       y = "Proportion",
       x = "Time Point") +
  scale_fill_manual(values = c("LRP1" = "blue2", "LRP2.1" = "#57C0AC")) +  # Custom colors
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

## H ####
sst_chodl_counts <- data.frame(
  GW18_20 = c(32,  31),
  GW22_25 = c(32, 13),
  Postnatal =  c(41,  12),
  GW14_27 =    c(191,  57),
  GW28_40 =    c(140,  10),
  PN_010 =    c(90,  3),
  PN_10 =    c(50,  0))
rownames(sst_chodl_counts) <- c("LRP1", "LRP2.1")
sst_chodl_props <- sweep(sst_chodl_counts, 2, colSums(sst_chodl_counts), FUN = "/")
verall_proportion <- sst_chodl_long %>% group_by(Time_Point) %>% summarize(Proportion = sum(Proportion), .groups = 'drop') 
sst_chodl_long <- as.data.frame(sst_chodl_props) %>% rownames_to_column(var = "Cell_Type") %>%
  melt(id.vars = "Cell_Type", variable.name = "Time_Point", value.name = "Proportion")
sst_chodl_long$Time_Point <- factor(sst_chodl_long$Time_Point, levels = c('GW18_20',"GW22_25", "Postnatal", "GW14_27", "GW28_40", 'PN_010', 'PN_10'))

#plot
ggplot() +
  geom_bar(data = sst_chodl_long, aes(x = Time_Point, y = Proportion, fill = Cell_Type), 
           stat = "identity", position = "fill", alpha = 0.5) +  # Stacked bar plot
  scale_y_continuous(labels = scales::percent) +  # Y-axis as percentage
  labs(title = "Proportion of LRP1 and LRP2.1 Across Time Points",
       y = "Proportion",
       x = "Time Point") +
  scale_fill_manual(values = c("LRP1" = "blue2", "LRP2.1" = "#57C0AC")) +  # Custom colors
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

## I ###
sst_chodl_counts <- data.frame(
  MPT_1 = c(507,  189),
  MPT_3 = c(405, 46),
  MPT_6 =  c(2115,  272),
  MPT_12_18 =  c(908,  117))
rownames(sst_chodl_counts) <- c("LRP1", "LRP2.1")
sst_chodl_props <- sweep(sst_chodl_counts, 2, colSums(sst_chodl_counts), FUN = "/")
verall_proportion <- sst_chodl_long %>% group_by(Time_Point) %>% summarize(Proportion = sum(Proportion), .groups = 'drop') 
sst_chodl_long <- as.data.frame(sst_chodl_props) %>% rownames_to_column(var = "Cell_Type") %>%
  melt(id.vars = "Cell_Type", variable.name = "Time_Point", value.name = "Proportion")
sst_chodl_long$Time_Point <- factor(sst_chodl_long$Time_Point, levels = c('MPT_1',"MPT_3", "MPT_6", "MPT_12_18"))

#plot
ggplot() +
  geom_bar(data = sst_chodl_long, aes(x = Time_Point, y = Proportion, fill = Cell_Type), 
           stat = "identity", position = "fill", alpha = 0.5) +  # Stacked bar plot
  scale_y_continuous(labels = scales::percent) +  # Y-axis as percentage
  labs(title = "Proportion of LRP1 and LRP2.1 Across Time Points",
       y = "Proportion",
       x = "Time Point") +
  scale_fill_manual(values = c("LRP1" = "blue2", "LRP2.1" = "#57C0AC")) +  # Custom colors
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

## J ####
pca_data <- read_csv("PCA_Metadata.csv", show_col_types = FALSE)
pca_data$PC_1 <- as.numeric(pca_data$PC_1)
pca_data$PC_2 <- as.numeric(pca_data$PC_2)
ggplot(pca_data, aes(x = PC_1, y = PC_2, color = cluster_prediction)) +
  geom_point(size = 1) +  # Customize point size
  scale_color_manual(values = c('#00B9DE','#2AB45D')) +  # Define custom colors
  ggtitle("PCA Plot Colored by Stage") +
  theme_minimal()


rm(list = c("annotation_col", "cluster1_data", "cluster2_data", "heatmap_data", 
            "LRP_numbers", "lrp2_only", "meta", "metadata", "metadata_list", 
            "ordered_data", "overall_proportion", "plot_data", "sst_chodl_counts", 
            "sst_chodl_long", "sst_proportion", "sst_chodl_props", "sub", "cells"))
```



# Ext Fig 2 ####
```{r}
## B ####
#it gives a .pdf output
data<-read.table("data.txt", row.names = 1, header = TRUE)

files <- dir()
files_anchors <- files[grep("Number_of_anchors_",files)]
files_kBET <- files[grep("kBET_batch_estimate_",files)]
files_kBET <- files_kBET[grep(".RDS",files_kBET)]

rejection_rate <- c()
mean_anchors <- c()
mean_anchors_scaled <- c()
for (i in rownames(data)){
  n_anchors <- read.table(paste0(files_anchors[grep(i,files_anchors)]), row.names = 1)
  mean_anchors <- c(mean_anchors,mean(n_anchors[,1]))
  mean_anchors_scaled <- c(mean_anchors_scaled,mean(n_anchors[,1])/data[i,2])
  kbet<-readRDS(paste0(files_kBET[grep(i,files_kBET)]))
  rejection_rate<-c(rejection_rate,median(kbet$stats$kBET.observed))
}
colors_ditto <- dittoColors()

files_knn <- files[grep("Percentages_of_cells_in_",files)]
files_knn <- files_knn[grep(".RDS",files_knn)]
pdf('plot_percentages_of_cells_with_mixed_neighborhood_in_kNN_allSamples.pdf', width=8)
par(cex=0.5,cex.lab=1.2)
i=1
knn_perc<-readRDS(paste0(files_knn[grep(rownames(data)[i],files_knn)]))
plot(seq(5,200,5),knn_perc, xlab="k",ylab="fraction of cells with mixed neighborhood", pch=0,cex=0.8,col=colors_ditto[i], ylim=c(0,1.05),xaxt='n',yaxt='n')
legend("top",legend=paste0(data[,1]," - ",rownames(data[,])),fill=colors_ditto,horiz = FALSE,cex=0.915,pch = c(0:12),ncol=6)
axis(2,cex.axis=1.5,)
axis(1,cex.axis=1.5)
print(lines(seq(5,200,5),knn_perc,col=colors_ditto[i]))
for (i in 2:nrow(data)){
  knn_perc<-readRDS(paste0(files_knn[grep(rownames(data)[i],files_knn)]))
  print(points(seq(5,200,5),knn_perc, pch=i-1,col=colors_ditto[i]))
  print(lines(seq(5,200,5),knn_perc,col=colors_ditto[i]))
}
abline(v=60, col = "blue4", lty=2)
abline(h=0.45, col = "red3", lty=2)
dev.off()

knn_perc_k60<-c()
for (i in 1:nrow(data)){
  knn_perc<-readRDS(paste0(files_knn[grep(rownames(data)[i],files_knn)]))
  knn_perc_k60<-c(knn_perc_k60,knn_perc["60"])
}

## C ##
data <- read_csv('DS_anchors.csv')
ggplot(data, aes(x = reorder(dataset_id, -Anchors), y = Anchors)) +
  stat_summary_bin(aes(y = Anchors), fun = "mean", geom = "bar")+
  stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.2, color = "black")+
  geom_jitter()+
  geom_hline(yintercept = 15000, linetype = "dashed", color = "red")

## D ####
#it gives a .pdf output
data_plot1 <- data.frame(data, mean_anchors = mean_anchors, kBET_rejection_rate=rejection_rate, knn_perc_k60=knn_perc_k60)
pdf('Mean_ancors_vs_k60_fraction_of_cells.pdf')
plot(data_plot1$knn_perc_k60, data_plot1$mean_anchors, col=colors_ditto, pch=c(0:12),cex=1.5,lwd=2, xlab ="fraction of cells with mixed neighborhood", ylab="Mean anchors")
legend("topleft",legend=paste0(data[,1]," - ",rownames(data[,])),fill=colors_ditto,horiz = FALSE, cex = 0.6, pch = c(0:12))
abline(v=0.45, col="red3", lty=2)
abline(h=2000, col="red3", lty=2)
dev.off()

## E ####
#it gives a .pdf output
data_plot1<-data.frame(data,mean_anchors = mean_anchors, mean_anchors_scaled = mean_anchors_scaled, kBET_rejection_rate=rejection_rate)
pdf('Mean_anchors_vs_kBET_RejectionRate.pdf')
plot(data_plot1$kBET_rejection_rate, data_plot1$mean_anchors, col=colors_ditto, pch=c(0:12),cex=1.5,lwd=2, xlab ="kBET rejection rate", ylab="Mean anchors")
legend("topright",legend=paste0(data[,1]," - ",rownames(data)),fill=colors_ditto,horiz = FALSE,cex=0.6,pch = c(0:12))
abline(v=0.5, col = "red3",lty = 2)
abline(h=2000, col = "red3",lty = 2)
dev.off()

rm(list = c("data", "data_plot1", "kbet", "n_anchors", "colors_ditto", 
            "files", "files_anchors", "files_kBET", "files_knn", "i", 
            "knn_perc", "knn_perc_k60", "mean_anchors", "mean_anchors_scaled", 
            "rejection_rate"))
gc()
```

# Ext Fig 3 ####
```{r}
## A ####
conserved_clusters <- readRDS('MN_clusters_conserved_finalClusters.RDS')
cluster_number<-list()
for (perc.sub in seq(0.05, 0.95, 0.05)){
  cluster_number[[as.character(perc.sub)]]<-unlist(lapply(conserved_clusters[[as.character(perc.sub)]], function(x) {
    sum(diag(x) >0.9)
  }))
}
cluster_number_all<-35

avg<-rev(unlist(lapply(cluster_number, mean))/cluster_number_all)
x<-round(seq(0.05, 0.95, 0.05)*51205)
err <- unlist(lapply(cluster_number, function(x) sd(x/cluster_number_all)/sqrt(length(x))))
pdf('Number_ofcluster_finalClustersMarkers_MAST_v2.pdf', width = 8, height = 6)
par(las=2, mar=c(6, 5, 4, 3))
plot(seq(0.05, 0.95, 0.05), rev(unlist(lapply(cluster_number, mean))/cluster_number_all), pch=20, xaxt='n', xlab="", ylab="Fraction of conserved cluster", ylim=c(0.54,1.02))
axis(1, at=seq(0.05, 0.95, 0.05), labels = rev(x), cex.lab=1.2, cex.axis=1.2)
lines(seq(0.05, 0.95, 0.05), avg)
title(xlab="Number of cells",mgp=c(4,1,0))
abline(v=0.8, col="red3", lty=2) #inflection point
arrows(seq(0.05, 0.95, 0.05), avg-rev(err), seq(0.05, 0.95, 0.05), avg+rev(err), length=0.05, angle=90, code=3)

cluster_number<-list()
for (perc.sub in seq(0.05, 0.95, 0.05)){
  cluster_number[[as.character(perc.sub)]]<-unlist(lapply(conserved_clusters[[as.character(perc.sub)]], function(x) {
    sum(diag(x) >0.88)
  }))
}
avg<-rev(unlist(lapply(cluster_number, mean))/cluster_number_all)
x<-round(seq(0.05, 0.95, 0.05)*51205)
err <- unlist(lapply(cluster_number, function(x) sd(x/cluster_number_all)/sqrt(length(x))))
points(seq(0.05, 0.95, 0.05), rev(unlist(lapply(cluster_number, mean))/cluster_number_all), pch=20, col="red4")
lines(seq(0.05, 0.95, 0.05), avg,col="red4")
arrows(seq(0.05, 0.95, 0.05), avg-rev(err), seq(0.05, 0.95, 0.05), avg+rev(err), length=0.05, angle=90, code=3, col = "red4")

cluster_number<-list()
for (perc.sub in seq(0.05, 0.95, 0.05)){
  cluster_number[[as.character(perc.sub)]]<-unlist(lapply(conserved_clusters[[as.character(perc.sub)]], function(x) {
    sum(diag(x) >0.86)
  }))
}
avg<-rev(unlist(lapply(cluster_number, mean))/cluster_number_all)
x<-round(seq(0.05, 0.95, 0.05)*51205)
err <- unlist(lapply(cluster_number, function(x) sd(x/cluster_number_all)/sqrt(length(x))))
points(seq(0.05, 0.95, 0.05), rev(unlist(lapply(cluster_number, mean))/cluster_number_all), pch=20, col="blue4")
lines(seq(0.05, 0.95, 0.05), avg,col="blue4")
arrows(seq(0.05, 0.95, 0.05), avg-rev(err), seq(0.05, 0.95, 0.05), avg+rev(err), length=0.05, angle=90, code=3, col = "blue4")

legend("bottom",x.intersp = c(0.5,0.5,0.5),pch = 20,lty=1, legend = c(expression(AUROC > 0.9),expression(AUROC > 0.88),expression(AUROC > 0.86)), col = c("black","red4","blue4"), horiz = TRUE,inset=c(0,1), xpd=TRUE, bty="n" )
dev.off()

## B ###
majorClassMarkers <- readRDS("allDEGs_final_clusters_subsets_Atlas_v2.RDS")
DEGs_list <- readRDS("allDEGs_final_clusters_subsets.RDS")

DEGs_number<-list()
for (perc.sub in seq(0.05, 0.95, 0.05)){
  DEGs_number[[as.character(perc.sub)]]<-unlist(lapply(DEGs_list[[as.character(perc.sub)]], function(x) {
    x <- subset(x,pct.1 > 0.2 & avg_log2FC>log2(1.8))
    dim(x)[1]
  }))
}
DEGs_number_all <- dim(subset(majorClassMarkers,pct.1 > 0.2 & avg_log2FC>log2(1.8)))[1]

avg<-rev(unlist(lapply(DEGs_number, mean))/DEGs_number_all)
x<-round(seq(0.05, 0.95, 0.05)*51205)
err <- unlist(lapply(DEGs_number, function(x) sd(x/DEGs_number_all)/sqrt(length(x))))
pdf('Number_ofDEGs_finalClustersMarkers_MAST.pdf', width = 8, height = 6)
par(las=2, mar=c(6, 5, 4, 3))
plot(seq(0.05, 0.95, 0.05), rev(unlist(lapply(DEGs_number, mean))/DEGs_number_all), pch=20, xaxt='n', xlab="", ylab="Fraction of conserved DEGs", ylim=c(0.48,1.02))
axis(1, at=seq(0.05, 0.95, 0.05), labels = rev(x), cex.lab=1.2, cex.axis=1.2)
lines(seq(0.05, 0.95, 0.05), avg)
title(xlab="Number of cells",mgp=c(4,1,0))
abline(v=0.75, col="red3", lty=2) #inflection point
arrows(seq(0.05, 0.95, 0.05), avg-rev(err), seq(0.05, 0.95, 0.05), avg+rev(err), length=0.05, angle=90, code=3)

DEGs_number<-list()
for (perc.sub in seq(0.05, 0.95, 0.05)){
  DEGs_number[[as.character(perc.sub)]]<-unlist(lapply(DEGs_list[[as.character(perc.sub)]], function(x) {
    x <- subset(x,pct.1 > 0.2 & avg_log2FC>log2(2))
    dim(x)[1]
  }))
}
DEGs_number_all <- dim(subset(majorClassMarkers,pct.1 > 0.2 & avg_log2FC>log2(2)))[1]
avg<-rev(unlist(lapply(DEGs_number, mean))/DEGs_number_all)
x<-round(seq(0.05, 0.95, 0.05)*51205)
err <- unlist(lapply(DEGs_number, function(x) sd(x/DEGs_number_all)/sqrt(length(x))))
points(seq(0.05, 0.95, 0.05), rev(unlist(lapply(DEGs_number, mean))/DEGs_number_all), pch=20, col="red4")
lines(seq(0.05, 0.95, 0.05), avg,col="red4")
arrows(seq(0.05, 0.95, 0.05), avg-rev(err), seq(0.05, 0.95, 0.05), avg+rev(err), length=0.05, angle=90, code=3, col = "red4")

DEGs_number <- list()
for (perc.sub in seq(0.05, 0.95, 0.05)){
  DEGs_number[[as.character(perc.sub)]]<-unlist(lapply(DEGs_list[[as.character(perc.sub)]], function(x) {
    x <- subset(x,pct.1 > 0.2 & avg_log2FC>log2(2.2))
    dim(x)[1]
  }))
}
DEGs_number_all <- dim(subset(majorClassMarkers,pct.1 > 0.2 & avg_log2FC>log2(2.2)))[1]
avg<-rev(unlist(lapply(DEGs_number, mean))/DEGs_number_all)
x<-round(seq(0.05, 0.95, 0.05)*51205)
err <- unlist(lapply(DEGs_number, function(x) sd(x/DEGs_number_all)/sqrt(length(x))))
points(seq(0.05, 0.95, 0.05), rev(unlist(lapply(DEGs_number, mean))/DEGs_number_all), pch=20, col="blue4")
lines(seq(0.05, 0.95, 0.05), avg,col="blue4")
arrows(seq(0.05, 0.95, 0.05), avg-rev(err), seq(0.05, 0.95, 0.05), avg+rev(err), length=0.05, angle=90, code=3, col = "blue4")

legend("bottom",x.intersp = c(0.5,0.5,0.5),pch = 20,lty=1, legend = c(expression(log[2](FC) > log[2](1.8)),expression(log[2](FC) > log[2](2.0)),expression(log[2](FC) > log[2](2.2))), col = c("black","red4","blue4"), horiz = TRUE,inset=c(0,1), xpd=TRUE, bty="n" )
dev.off()

## C ####
diff_mean <- readRDS('diff_mean_subsampling_DEGs.RDS')
x<-round(seq(0.05, 0.95, 0.05)*51205)
avg<-rev(unlist(lapply(diff_mean, mean)))
err <- unlist(lapply(diff_mean, function(x) sd(x)/sqrt(length(x))))
pdf('Differences_MeanExpression_of_finalClustersMarkers_MAST_v2.pdf', width = 8, height = 6)
par(las=2, mar=c(6, 5, 3, 3))
plot(seq(0.05, 0.95, 0.05), rev(unlist(lapply(diff_mean, mean))), pch=20, xaxt='n', xlab="", ylab="Absolute Difference (mean expression)")
axis(1, at=seq(0.05, 0.95, 0.05), labels = rev(x), cex.lab=1.2, cex.axis=1.2)
lines(seq(0.05, 0.95, 0.05), rev(unlist(lapply(diff_mean, mean))))
title(xlab="Number of cells",mgp=c(4,1,0))
abline(v=0.75, col="red3", lty=2) #inflection point
arrows(seq(0.05, 0.95, 0.05), avg-rev(err), seq(0.05, 0.95, 0.05), avg+rev(err), length=0.05, angle=90, code=3)
dev.off()

## D ####
HausdorffDistance <- readRDS('HausdorffDistance_q50.RDS')
x<-round(seq(0.05, 0.95, 0.05)*51205)
avg<-rev(unlist(lapply(HausdorffDistance, mean)))
err <- unlist(lapply(HausdorffDistance, function(x) sd(x)/sqrt(length(x))))
pdf('HausdorffDistance_of_subsampling_q50_v2.pdf', width = 8, height = 6)
par(las=2, mar=c(6, 5, 3, 3))
plot(seq(0.05, 0.95, 0.05), rev(unlist(lapply(HausdorffDistance, mean))), pch=20, xaxt='n', xlab="", ylab="Robust Hausdorff Distance")
axis(1, at=seq(0.05, 0.95, 0.05), labels = rev(x), cex.lab=1.2, cex.axis=1.2)
lines(seq(0.05, 0.95, 0.05), rev(unlist(lapply(HausdorffDistance, mean))))
title(xlab="Number of cells",mgp=c(4,1,0))
abline(v=0.8, col="red3",lty=2) #inflection point
arrows(seq(0.05, 0.95, 0.05), avg-rev(err), seq(0.05, 0.95, 0.05), avg+rev(err), length=0.05, angle=90, code=3)
dev.off()

## E ####
#load data
jc_to_plot_all <- readRDS("Jaccard_allClusters.RDS")
colors_plot<-readRDS("palette_clusters.RDS")

pdf('JC_ClusterStability_plot_all_clusters.pdf', width=15, height = 10)
jc_to_plot_all %>% ggplot2::ggplot(ggplot2::aes(x = cluster, y = jaccard, fill = cluster)) +
  geom_flat_violin(position = position_nudge(x = .2, y = 0), alpha = .7, width = 3) +
  ggplot2::geom_point(ggplot2::aes(y = jaccard, color = cluster), position = position_jitter(width = .15), size = .5, alpha = 0.7) +
  ggplot2::geom_boxplot(width = .2, outlier.shape = NA, alpha = 0.5) +
  ggplot2::theme_classic() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + ylim(0,1) +
  ggplot2::theme(legend.position="none") +  scale_colour_manual(values=colors_plot) +  scale_fill_manual(values=colors_plot) + geom_hline(yintercept = c(0.75), linetype = 2,color="red2")
dev.off()


rm(list = c("cluster_number", "conserved_clusters", "DEGs_list", "DEGs_number", 
            "diff_mean", "HausdorffDistance", "jc_to_plot_all", "majorClassMarkers", 
            "avg", "cluster_number_all", "colors_plot", "DEGs_number_all", 
            "err", "perc.sub", "x"))
gc()

```

# Ext Fig 4 ####
```{r}
## A ####
#Load the data
cellpercentages <- read.csv("CellPercentages_ExFig4a.csv", row.names = 1)
#Bar plot
library(ggplot2)
ggplot(cellpercentages, aes(x = final_clusters, y = percentage, fill = final_clusters)) +
  geom_bar(stat = "identity", position = "stack") +
  labs(title = "Percentage of Cells by cluster",
       x = "Cluster",
       y = "Percentage (%)",
       fill = NULL) +
  theme_minimal() +
  scale_fill_manual(values = c("#30BBD1","#5DC3AF", 'darkgreen')) +
  theme(legend.position = "right")

## B ####
#Load the data
umap_data <- read.csv("UMAP_Metadata_ExFig4b.csv")
# Plot by cluster identity
ggplot(umap_data, aes(x = UMAP_1, y = UMAP_2, color = metadata)) + geom_point(size = 0.5, alpha = 0.8) + scale_color_manual(values = c("#30BBD1","#5DC3AF", 'darkgreen')) + theme_minimal()

## C ####
#Load the data
hp <- readRDS("HeatMapA_ExtFig4c.rds")
lgd_list <- readRDS("HeatMapB_ExtFig4c.rds")
#Dot plot
col_fun = circlize::colorRamp2(c(-1, 0, 2.5), c("blue2","gray90","red2"))
draw( hp, annotation_legend_list = lgd_list)

## D ####
#Load the data
hp <- readRDS("HeatMapA_ExtFig4d.rds")
lgd_list <- readRDS("HeatMapB_ExtFig4d.rds")
#Dot plot
col_fun = circlize::colorRamp2(c(-1, 0, 2.5), c("blue2","gray90","red2"))
draw( hp, annotation_legend_list = lgd_list)

## E ####
#Load the data
hp <- readRDS("HeatMapA_ExtFig4e.rds")
lgd_list <- readRDS("HeatMapB_ExtFig4e.rds")
#Dot plot
col_fun = circlize::colorRamp2(c(-1, 0, 2.5), c("blue2","gray90","red2"))
draw( hp, annotation_legend_list = lgd_list)


rm(list = c("cellpercentages", "hp", "lgd_list", "umap_data", "col_fun"))
gc()


```

# Ext Fig 5 ####
```{r}
## A ####
# Load the data
PercentageMapping <- read.csv("BarPlot_ExtFig5a.csv")
# Bar plot
ggplot(PercentageMapping, aes(x = major_label_transferAnchors_BaseAtlas, y = Percentage, fill = Category)) +
  geom_bar(stat = "identity") +
  labs(title = "Percentage of Cells by Subclass Bootstrapping Probability Range",
       x = "Major",
       y = "Percentage of Cells") +
  scale_fill_manual(values = c(">=0.9" = "blue", "<0.9" = "lightblue")) +
  theme_minimal() +
  theme(legend.title = element_blank())
# Load the data
PercentageMapping <- read.csv("BarPlot_ExtFig5a2.csv")
# Bar plot
ggplot(PercentageMapping, aes(x = Major, y = proportion, fill = supertype_categoryAllOther)) +
  geom_bar(stat = "identity", position = "stack") +
  labs(
    title = "Proportion of Supertypes by Group",
    x = "Group",
    y = "Proportion",
    fill = "Types"
  ) +
  scale_fill_manual(values = c("grey",
                               "#F0E443",
                               "#E6A024")) +
  theme_minimal()

## B,C ####
# Load the data
PercentageMapping <- read.csv("BarPlot_ExtFig5bc.csv")
#Bar plots
ggplot(PercentageMapping, aes(x = group, y = Percentage, fill = Category)) +
  geom_bar(stat = "identity") +  # Use position = "dodge" for side-by-side bars
  labs(title = "Percentage of Cells by Supertype Bootstrapping Probability Range",
       x = "group",
       y = "Percentage of Cells") +
  scale_fill_manual(values = c(">=0.9" = "blue", "<0.9" = "lightblue")) +
  facet_wrap(~ Major) +  # Create separate panels for each group
  theme_minimal() +
  theme(legend.title = element_blank())


## D ####
# Load the data
PercentageMapping <- read.csv("BarPlot_ExtFig5d.csv")
#Bar plots
ggplot(PercentageMapping, aes(x = group, y = proportion, fill = supertype_categoryAllOther)) +
  geom_bar(stat = "identity", position = "stack") +
  labs( title = "Proportion of Supertypes by Group", x = "Group", y = "Proportion", fill = "Supertype") +
  scale_fill_manual(values = c("#81c3d7","#3C3CF0","#3f8efc","#001d3d","#00a7e1","#6a994e","#A01061","#B23C25","#2a6f97","#6E2DB7","#52b69a","#ffa69e",
                               "#5111B2","#B58C00","#d283ff","#2E004B","#A45F10","#007f5f","#006400","grey")) +
  theme_minimal()


rm(list = c("PercentageMapping"))
gc()


```

# Ext Fig 6 ####
```{r}
## A ####
umap_data <- read.csv("UMAP_Metadata.csv")
#2021 clusters
ggplot(umap_data, aes(x = UMAP_1, y = UMAP_2, color = cluster_label)) +
     geom_point(size = 0.5, alpha = 0.8) +  # Increase size if points are too small
     scale_color_manual(values = c("#1096C5","#6158A6", '#F19B87','#9D1F2B')) +
     theme_minimal()

#2021 MapMycell
ggplot(umap_data, aes(x = UMAP_1, y = UMAP_2, color = supertype_name_Yao2023)) +
     geom_point(size = 0.5, alpha = 0.8) +  # Increase size if points are too small
     scale_color_manual(values = c("#1096C5","#6158A6", '#F19B87','#9D1F2B')) +
     theme_minimal()

## D-H ####
data <- data.frame(
  Matrix = c("Sst/Nos1/Dach1", "Sst/Nos1/Dach1", "Sst/Nos1/Dach1", "Sst/Nos1/Dach1",
    "Sst/Nos1/Mafb", "Sst/Nos1/Mafb", "Sst/Nos1/Mafb", "Sst/Nos1/Mafb"),
  Row = c("LRP1", "LRP1", "LRP2.1", "LRP2.1", "LRP1", "LRP1", "LRP2.1", "LRP2.1"),
  Column = c("LRP1", "LRP2.1", "LRP1", "LRP2.1", "LRP1", "LRP2.1", "LRP1", "LRP2.1"),
  Accuracy = c(0.905, 0.095, 0.2, 0.8, 0.737, 0.263, 0.072, 0.928))

ggplot(data, aes(x = Column, y = Row, fill = Accuracy)) +
geom_tile(color = "black") + 
geom_text(aes(label = round(Accuracy, 3)), color = "white", size = 6) + 
scale_fill_gradient(low = "black", high = "orange", limits = c(0, 1)) + 
labs(title = NULL, x = NULL, y = NULL) + 
facet_wrap(~Matrix) + 
theme_minimal() +
theme( axis.text = element_text(size = 12), strip.text = element_text(size = 14, face = "bold")) 


## G-Dach1 ####
data <- read.csv("Dach1_GFP_quantification.csv")
data$Stage <- factor(data$Stage, levels = c('P5',"P7", "P21"))
data$Celltype <- factor(data$Celltype, levels = c('YFP-Dach1-low',"YFP-Dach1-high"))

summary_data2 <- data %>%
  group_by(Stage, Celltype) %>%
  summarise(total_count = sum(cellcount), .groups = 'drop') %>%
  group_by(Stage) %>%
  mutate(relative_count = total_count / sum(total_count) * 100)
# Plotting the relative distribution as a barplot
ggplot(summary_data2, aes(x = Stage, y = relative_count, fill = Celltype)) +
  geom_bar(stat = "identity", position = "stack") +
  labs(title = "Relative Distribution of LRP1/LRP2 by Stage",
       y = "Relative Percentage (%)",
       x = "Region") +
  theme_minimal() +
  scale_fill_manual(values = c("YFP-Dach1-high" = "#51B885", "YFP-Dach1-low" = "#81D2E7")) +
  geom_text(aes(label = total_count), position = position_stack(vjust = 0.5), color = "black", size = 6)+
  geom_text(aes(label = paste0(round(relative_count, 1), "%")), position = position_stack(vjust = 0.9), color = "black", size = 4) 

## K-Mafb ####
data <- read.csv("Mafb__GFP.csv")
data$Stage <- factor(data$Stage, levels = c('P5',"P7", "P21"))
data$Celltype <- factor(data$Celltype, levels = c('YFP-MafB-high',"YFP-MafB-low"))

summary_data2 <- data %>%
  group_by(Stage, Celltype) %>%
  summarise(total_count = sum(cellcount), .groups = 'drop') %>%
  group_by(Stage) %>%
  mutate(relative_count = total_count / sum(total_count) * 100)
# Plotting the relative distribution as a barplot
ggplot(summary_data2, aes(x = Stage, y = relative_count, fill = Celltype)) +
  geom_bar(stat = "identity", position = "stack") +
  labs(title = "Relative Distribution of LRP1/LRP2 by Stage",
       y = "Relative Percentage (%)",
       x = "Region") +
  theme_minimal() +
  scale_fill_manual(values = c("YFP-MafB-high" = "#81D2E7", "YFP-MafB-low" = "#51B885")) +
  geom_text(aes(label = total_count), position = position_stack(vjust = 0.5), color = "black", size = 6)+
  geom_text(aes(label = paste0(round(relative_count, 1), "%")), position = position_stack(vjust = 0.9), color = "black", size = 4) 


## L ###
data_relative <- read.csv("data_relative.csv")
ggplot(data_relative, aes(x = supertype_name, y = Proportion, fill = Condition)) +
  geom_bar(stat = "identity", position = "dodge") +
  scale_y_continuous(labels = scales::percent_format()) +
  labs(
    x = "Supertype",
    y = "Relative Proportion",
    fill = "Condition",
    title = "SST+ Interneuron Relative Proportions"
  ) +
  theme_minimal() +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1))


## M ####
pca_data <- read_csv("PCA_Metadata.csv", show_col_types = FALSE)
pca_data$PC_1 <- as.numeric(pca_data$PC_1)
pca_data$PC_2 <- as.numeric(pca_data$PC_2)
pca_data$Stage <- factor(pca_data$Stage, levels = c("1MPT", "3MPT", "6MPT", "2MPT", "8MPT"))
ggplot(pca_data, aes(x = PC_1, y = PC_2, color = Stage)) +
  geom_point(size = 1) +  # Customize point size
  scale_color_manual(values = c('#2AB45D','#0077B6','#28BAA3','#00B9DE','#C574AF')) +  # Define custom colors
  ggtitle("PCA Plot Colored by Stage") +
  theme_minimal()


```

