---
title: "Module 1 QCs_and_filtering_of_scRNAseq_datasets"
---

## Installation needed:
```{r}
if (!requireNamespace("pacman", quietly = TRUE)) install.packages("pacman")
pacman::p_load(
  BiocManager, remotes, devtools, Seurat, dittoSeq, ggplot2, gtools, dplyr, R.utils, DoubletFinder,
  dismo, missMDA, hdf5r
)

#remotes::install_github("chris-mcginnis-ucsf/DoubletFinder") #install if needed
#BiocManager::install("dittoSeq")
```


## Load required packages
```{r}
library(Seurat)
library(dittoSeq)
library(ggplot2)
library(gtools)
library(dplyr)
library(dittoSeq)
library(R.utils)
library(DoubletFinder)
#Define Seurat parameters
seurat_default_options <- list(
  Seurat.memsafe = TRUE,
  Seurat.warn.umap.uwot = TRUE,
  Seurat.checkdots = "warn",
  Seurat.limma.wilcox.msg = TRUE,
  Seurat.Rfast2.msg = TRUE,
  Seurat.warn.vlnplot.split = TRUE
)
#Source the external script 'Seurat_Utils' for further analysis (10-fold Singular Value Decomposition (SVD) cross validation to predict number of PCs)
source("Seurat_Utils.R") 
```

## Set the directory with your dataset
```{r}
#Set your personal working directory where your count matrices folder is located 
setwd("") # set your working directory 
```


## read in the data you loaded
```{r}
#Loads scRNA-seq count matrices from CellRanger output. Ensure the folder containing your files is located in your working directory
#option 1 : Cellranger output files
#Dataset <- Read10X("L:/GBW-0086_LYLIM/Manuscripts/2025_Atlas_development/Datasets/scRNAseq_datasets/P1/Lim_lab_P1_Live/Deep sequencing/filtered_feature_bc_matrix") 

#Option 2: If your data is in an h5 file format, use the following line instead:
Dataset <- Read10X_h5("/Volumes/lvs/GBW-0086_LYLIM/Manuscripts/2025_Atlas_development/GitHub codes/Submission/Module 1/Dataset_examples/h5 files/P1_DiBella.h5", use.names = TRUE, unique.features = TRUE)

#Create a Seurat object from the count matrices and specify the associated project: 
seurat_object <- CreateSeuratObject(counts = Dataset, project = "P1", min.cells = 10, min.features = 0)
#Add a metadata column to define the sample name:
seurat_object$sample <- "P1_Dibella" 

rm(Dataset)
gc()
```

## Reads alignement QC (usually performed already by CellRanger) - not always necessary (this is already done in most of the GEO datasets)
```{bash}
###This part process BAM files to extract counts and cell barcodes
# samtools view -@ 30 sample_alignments.bam |  grep CB:Z: | sed 's/.*CB:Z:\([ACGT]*\).*/\1/' | sort | uniq -c > all_reads_per_barcode.txt
# samtools view -F 0x1404 -q 30 sample_alignments.bam |  grep CB:Z: | sed 's/.*CB:Z:\([ACGT]*\).*/\1/' | sort | uniq -c > confident_reads_per_barcode.txt
# # sed 's/^[[:space:]]*//' all_reads_per_barcode.txt > all_reads_per_barcode_correct.txt
# sed 's/^[[:space:]]*//' confident_reads_per_barcode.txt > confident_reads_per_barcode_correct.txt

###This second part analyse the output of the previous one, removing low confidentially aligned reads
# read_per_bc<-read.delim("all_reads_per_barcode_correct.txt",header=F, sep =" ")
# colnames(read_per_bc)<-c("reads_per_bc","bc")
# conf_read_per_bc<-read.delim("confident_reads_per_barcode_correct.txt",header=F, sep =" ")
# colnames(conf_read_per_bc)<-c("conf_reads_per_bc","bc")
# read_per_bc_stats<-merge(read_per_bc,conf_read_per_bc,by="bc")
# read_per_bc_stats$bc<-paste0(read_per_bc_stats$bc,"-1")
# read_per_bc_stats<-read_per_bc_stats[which(read_per_bc_stats$bc %in% colnames(seurat_object)),]
# read_per_bc_stats$perc_conf<-read_per_bc_stats$conf_reads_per_bc/read_per_bc_stats$reads_per_bc
# rownames(read_per_bc_stats)<-read_per_bc_stats$bc
# read_per_bc_stats<-read_per_bc_stats[colnames(seurat_object),]
# read_per_bc_stats$bc<-NULL
# read_per_bc_stats$reads_per_bc<-NULL
# read_per_bc_stats$conf_reads_per_bc<-NULL# 
#seurat_object<-AddMetaData(object = seurat_object, metadata = read_per_bc_stats)
```

## Calculate percentage of mitochondiral and ribosomal genes
```{r}
# Calculate the percentage of mitochondrial and ribosomal gene expression. These metrics indicate cell quality (e.g., high mitochondrial content suggests cell stress or death).
seurat_object[["percent.mt"]] <- PercentageFeatureSet(seurat_object, pattern = "^mt-")
seurat_object[["percent.ribo"]] <- PercentageFeatureSet(seurat_object, pattern = "^Rp[Sl]")

# Visualize the distributions of mitochondrial and ribosomal gene percentages, the number of expressed genes per cell, and the number of UMIs per cell
VlnPlot(seurat_object, features = c("nFeature_RNA", "nCount_RNA", "percent.mt","percent.ribo"), ncol = 2, pt.size = 0, group.by = "sample")
```

## # Normalizes data, 
```{r}
seurat_obj<-seurat_object
# Normalizes data, identifies variable genes, and performs PCA for dimensionality reduction.
seurat_obj <- NormalizeData(seurat_obj)
seurat_obj <- FindVariableFeatures(seurat_obj, selection.method = "vst", nfeatures = 2000)
seurat_obj <- ScaleData(seurat_obj, vars.to.regress = "percent.mt")

rm(seurat_object)
gc()
```

## Doublets prediction with DoubletFinder
```{r}
seurat_obj <- RunPCA(seurat_obj, verbose = FALSE)
seurat_obj <- FindNeighbors(seurat_obj, reduction = "pca", dims = 1:5) 
#Note dim = 1:5, but typically 1:10 for us 1:5 works well
seurat_obj <- FindClusters(seurat_obj, resolution = 0.6) #resolution can also be adjusted
# pK Identification 
sweep.res.list_pbmc <- paramSweep(seurat_obj, PCs = 1:10, sct = FALSE)
sweep.stats_pbmc <- summarizeSweep(sweep.res.list_pbmc, GT = FALSE)
bcmvn_pbmc <- find.pK(sweep.stats_pbmc)
pK_pbmc=as.numeric(as.character(bcmvn_pbmc$pK))
BCmetric_pbmc=bcmvn_pbmc$BCmetric
pK_choose_pbmc = pK_pbmc[which(BCmetric_pbmc %in% max(BCmetric_pbmc))]
par(mar=c(5,4,4,8)+1,cex.main=1.2,font.main=2)
plot(x = pK_pbmc, y = BCmetric_pbmc, pch = 16,type="b",
     col = "blue",lty=1)
abline(v=pK_choose_pbmc,lwd=2,col='red',lty=2)
title("The BCmvn distributions")
text(pK_choose_pbmc,max(BCmetric_pbmc),as.character(pK_choose_pbmc),pos = 4,col = "red")

rm(list = c("bcmvn_pbmc", "sweep.res.list_pbmc", "sweep.stats_pbmc"))
gc()
```

## Review the previous graph and manually set the most accurate pk value (red line)
```{r}
pK_choose <- 0.29 #Insert your value here

# Homotypic Doublet Proportion Estimate (doublets originating from the same cluster)
homotypic.prop <- modelHomotypic(Idents(seurat_obj))
nExp_poi <- round(0.07*nrow((seurat_obj)@meta.data)) 
nExp_poi.adj <- round(nExp_poi*(1-homotypic.prop))
# Run DoubletFinder with varying classification stringencies 
seurat_obj <- doubletFinder(seurat_obj, PCs = 1:30, pN = 0.25, pK = pK_choose, nExp = nExp_poi, reuse.pANN = FALSE, sct = FALSE)
```

## Check reuse.pANN
```{r}
#Check the column name with the classification code generated
head(seurat_obj) #pause here after you check the heading
```
## Check reuse.pANN
```{r}
#Copy the 'DF.classification_xxx' number in the following reuse.pANN section
seurat_obj <- doubletFinder(seurat_obj, PCs = 1:30, pN = 0.25, pK = pK_choose, nExp = nExp_poi.adj, reuse.pANN = "DF.classifications_0.25_0.29_459", sct = FALSE) 
seurat_obj$doublet <- seurat_obj$DF.classifications_0.25_0.29_459 #copy/ replace 'DF.classifications_0.25_0.29_1289'
seurat_object <- seurat_obj

rm(list = c("seurat_obj", "BCmetric_pbmc", "homotypic.prop", "nExp_poi", 
            "nExp_poi.adj", "pK_choose", "pK_choose_pbmc", "pK_pbmc"))
gc()
```

## Cell and gene filtering (thresholds are user dependent)
```{r}
# set the minimum percentage of confidently mapped reads (not always necessary)
# lower.align <- mean(seurat_object[['perc_conf']][,1]) - (2*sd(seurat_object[['perc_conf']][,1]))

# Set the minimum number of UMIs per cell (calculated as the mean - 2x SD).
logUMI <- log1p(seurat_object[['nCount_RNA']])
lower.umi_WT1 <- exp(mean(logUMI[,1]) - (2*sd(logUMI[,1])))

# Remove cells based on UMI counts, percentage of mitochondrial genes and number of genes
Seurat_object_subset <- subset(seurat_object, subset = nFeature_RNA > 700 & percent.mt < 10 & nCount_RNA > lower.umi_WT1)

## Remove mito and ribo genes from the dataset - (not always necessary, you can regress them out during the following downstream analysis)
#mt_genes <- grep(pattern = "^mt-", x = rownames(Seurat_object_subset@assays$RNA@counts), value = TRUE)
#ribo_genes <- grep(pattern = "^Rp[Sl]", x = rownames(Seurat_object_subset@assays$RNA@counts), value = TRUE)
#keep_features <- !grepl(paste0("^", mt_genes, collapse = "|"), rownames(Seurat_object_subset@assays$RNA@counts))
#Seurat_object_subset <- subset(Seurat_object_subset, features = keep_features)
#keep_features <- !grepl(paste0("^", ribo_genes, collapse = "|"), rownames(Seurat_object_subset@assays$RNA@counts))
#Seurat_object_subset <- subset(Seurat_object_subset, features = keep_features)

rm(list = c("logUMI", "seurat_object", "lower.umi_WT1"))
gc()
```

## Predict Cell Cycle. 
```{r}
#Load the two external files containing gene names
s.genes <- readRDS("mouse_s.genes.rds")
g2m.genes <- readRDS("mouse_g2m.genes.rds")

#Assigns cells to cell cycle phases (S, G2M, or G1) based on marker genes
Seurat_object_subset <- CellCycleScoring(Seurat_object_subset, s.features = s.genes, g2m.features = g2m.genes)
ScoreDiffSeuratObject <- (Seurat_object_subset@meta.data[["S.Score"]] - Seurat_object_subset@meta.data[["G2M.Score"]])
Seurat_object_subset[['ccDiff']] <- ScoreDiffSeuratObject

rm(list = c("g2m.genes", "s.genes", "ScoreDiffSeuratObject"))
gc()
```

### Normalization and scaling
```{r}
Seurat_object_subset <- NormalizeData(Seurat_object_subset)
Seurat_object_subset <- FindVariableFeatures(Seurat_object_subset, nfeatures = 2000)
Seurat_object_subset <- ScaleData(Seurat_object_subset,vars.to.regress = c("nFeature_RNA",'percent.mt','ccDiff'), model.use = 'linear', block.size=dim(Seurat_object_subset)[1])
```


### PCA dimensionality
```{r}
Seurat_object_subset <- RunPCA(Seurat_object_subset, npcs = 60)
# Determine the optimal number of principal components (PCs) for downstream analysis
ElbowPlot(Seurat_object_subset, ndims = 60)
```


### Adjusting nPC based on ndims on the ElbowPlot 
```{r}
if (!file.exists("optimal_nPCs_WT1.RDS")) #Check if nPC results already exist
  { 
data.use.Seurat_object_subset <- PrepDR(object = Seurat_object_subset, genes.use = VariableFeatures(object = Seurat_object_subset), use.imputed = F, assay.type = "RNA")
path_data <- getwd()
nPCs.Seurat_object_subset <- PCA_estimate_nPC(data.use.Seurat_object_subset, whereto = paste0(path_data,"/optimal_nPCs_WT1.RDS"), k = 5, by.nPC = 2, from.nPC = 35, to.nPC = 45) 
# Adjust `from.nPC` and `to.nPC` after inspecting the ndims of the ElbowPlot

saveRDS(nPCs.Seurat_object_subset, file = "optimal_nPCs_WT1.RDS")
} else {
  # Load precomputed nPCs if they exist
  nPCs.Seurat_object_subset <- readRDS("optimal_nPCs_WT1.RDS")
}
## Adjust `from.nPC` and `to.nPC` after inspecting the ndims of the ElbowPlot. You can also change the number of k

rm(path_data)
rm(data.use.Seurat_object_subset)
gc()
```

### Compute UMAP/tSNE and clusters
```{r}
# Perform tSNE and UMAP dimensionality reduction using the top PCs
Seurat_object_subset <- RunTSNE(Seurat_object_subset, dims = 1:nPCs.Seurat_object_subset)
Seurat_object_subset <- RunUMAP(Seurat_object_subset, dims = 1:nPCs.Seurat_object_subset)

# Find neighbors and clusters based on reduced dimensions
Seurat_object_subset <- FindNeighbors(Seurat_object_subset, dims = 1:nPCs.Seurat_object_subset)
Seurat_object_subset <- FindClusters(Seurat_object_subset, resolution = 0.8)
```

## Diagnostic markers plots QC
```{r}
# check doublets on the UMAP (discard them if they tend to form clusters) 
dittoDimPlot(Seurat_object_subset, "doublet", reduction.use = "umap", split.by = "sample")
# This visualizes the clusters identified by Seurat's clustering on UMAP
dittoDimPlot(Seurat_object_subset, "seurat_clusters", reduction.use = "umap", split.by = "sample", do.label = T)

#Create diagnostic plots to check quality control metrics on UMAP (Clipping outlier values to better visualization on the graph, the data itself it is not clipped)
clipped_data <- Seurat_object_subset
clipped_data$nCount_RNA <- pmin(pmax(Seurat_object_subset$nCount_RNA, 0), 10000) #remove higher values to rescale the color map (only for plotting on the umap)
clipped_data$nFeature_RNA <- pmin(pmax(Seurat_object_subset$nFeature_RNA, 2315), 7500) #remove higher values to rescale the color map (only for plotting on the umap)

#Plot these QC metrics on the scaled UMAP
dittoDimPlot(clipped_data, "nCount_RNA", reduction.use = "umap", min.color = "lightgrey", max.color = "blue", min = 0, max = 20000, split.by = "sample")
dittoDimPlot(clipped_data, "nFeature_RNA", reduction.use = "umap", min.color = "lightgrey", max.color = "blue", split.by = "sample")
dittoDimPlot(Seurat_object_subset, "percent.mt", reduction.use = "umap", min.color = "lightgrey", max.color = "blue", split.by = "sample")
VlnPlot(Seurat_object_subset, features = c("nFeature_RNA", "nCount_RNA", "percent.mt","percent.ribo"), ncol = 2, pt.size = 0, group.by = "seurat_clusters")

rm(clipped_data)
gc()
```

## User define genes to visual which clusters will be selected. We exluded (1) Stress clusters, (2) negative markers - pyramidal, striatal, (3) positve markers - in this case SST, Lhx6 etc
```{r}
# Specificity plots (UMAP + Violin plot) - to extract cortical SST+ cells, we use Lhx6 and Sst 

# Stressed markers visualization (UMAP + Violin plot)
Stress <- c("Hspa5", "Mapk8", "Hsp90b1")
dittoDimPlot(Seurat_object_subset, Stress, reduction.use = "umap", min.color = "lightgrey", max.color = "blue")
VlnPlot(Seurat_object_subset, Stress, pt.size = 0, group.by = "seurat_clusters", ncol = 2)

# Cell-type contaminants plots, such as check for subcortical interneuron and pyramidal cells - these are to exclude
Contamin <- c("Meis2", "Nkx2-1")
dittoDimPlot(Seurat_object_subset, Contamin, reduction.use = "umap", min.color = "lightgrey", max.color = "blue", split.by = "sample")
VlnPlot(Seurat_object_subset, Contamin, pt.size = 0, group.by = "seurat_clusters", cols = dittoColors())

# Pyramidal cells markers (UMAP + Violin plot)
Pyramid <- c("Slc17a7", "Neurod6", "Tbr1")
dittoDimPlot(Seurat_object_subset, Pyramid, reduction.use = "umap", min.color = "lightgrey", max.color = "blue")
VlnPlot(Seurat_object_subset, Pyramid, pt.size = 0, group.by = "seurat_clusters", ncol = 2)

#Positive markers plots
Posmarkers <- c("Sst", "Lhx6", "Gad1", "Gad2")
dittoDimPlot(Seurat_object_subset, Posmarkers, reduction.use = "umap", min.color = "lightgrey", max.color = "blue")
VlnPlot(Seurat_object_subset, Posmarkers, pt.size = 0, group.by = "seurat_clusters", ncol = 2)

#use plot information to decide manually which clusters you want to keep based on 
```

# Remove contaminants and perform reclustering
```{r}
# Specify clusters you want to remove
#Seurat_object_cleaned <- subset(Seurat_object_subset, idents = c('8','12', '13', '14'), invert = T)
# - or -  Specify clusters you want to keep
Seurat_object_cleaned <- subset(Seurat_object_subset, idents = c('2', '4'), invert = F)

rm(Seurat_object_subset)
gc()

# Recluster selected cells
Seurat_object_cleaned <- NormalizeData(Seurat_object_cleaned)
Seurat_object_cleaned <- FindVariableFeatures(Seurat_object_cleaned,nfeatures = 2000)
Seurat_object_cleaned <- ScaleData(Seurat_object_cleaned,vars.to.regress = c("nFeature_RNA",'percent.mt','ccDiff'), model.use = 'linear', block.size=dim(Seurat_object_cleaned)[1])
Seurat_object_cleaned<-RunPCA(Seurat_object_cleaned, npcs = 100)

data.use.Seurat_object_cleaned <- PrepDR(object = Seurat_object_cleaned, genes.use = VariableFeatures(object = Seurat_object_cleaned), use.imputed = F, assay.type = "RNA")
path_data <- getwd()
nPCs.5_clean <- PCA_estimate_nPC(data.use.Seurat_object_cleaned, whereto = paste0(path_data,"/optimal_nPCs_5_clean.RDS"), by.nPC = 5, from.nPC = 25, to.nPC = 65) 
pdf('nPCs_estimate_clean.pdf')
nPCs.clean <- PCA_estimate_nPC(data.use.Seurat_object_cleaned, whereto = paste0(path_data,"/optimal_nPCs_clean.RDS"), by.nPC = 1, from.nPC = nPCs.5_clean-5, to.nPC = nPCs.5_clean + 5) 
#adjust from.nPC = XXX, to.nPC = XXX based on ElbowPlot ndims. You can also change the number of k
dev.off()

Seurat_object_cleaned <- RunUMAP(Seurat_object_cleaned, dims = 1:nPCs.clean)
Seurat_object_cleaned <- FindNeighbors(Seurat_object_cleaned, dims = 1:nPCs.clean)
Seurat_object_cleaned <- FindClusters(Seurat_object_cleaned, resolution = 0.8)

# plot clusters
dittoDimPlot(Seurat_object_cleaned, "seurat_clusters", reduction.use = "umap", do.label = T)
# Check again the expression of gene of interest
dittoDimPlot(Seurat_object_cleaned, "Sst", reduction.use = "umap", min.color = "lightgrey", max.color = "blue")
VlnPlot(Seurat_object_cleaned, "Sst", pt.size = 0, group.by = "seurat_clusters", cols = dittoColors())
# Quality control plots to asses different matrices
VlnPlot(Seurat_object_cleaned, features = c("nFeature_RNA", "nCount_RNA", "percent.mt","percent.ribo"), ncol = 2, pt.size = 0, group.by = "seurat_clusters")

# If contamination or unwanted clusters remain, this code can be re-run to clean the data further and perform reclustering.
```
#Reclean script
```{r}

Seurat_object_cleaned <- subset(Seurat_object_cleaned, idents = c('2', '4'), invert = F)

# plot clusters
dittoDimPlot(Seurat_object_cleaned, "seurat_clusters", reduction.use = "umap", do.label = T)
# Check again the expression of gene of interest
dittoDimPlot(Seurat_object_cleaned, "Sst", reduction.use = "umap", min.color = "lightgrey", max.color = "blue")
VlnPlot(Seurat_object_cleaned, "Sst", pt.size = 0, group.by = "seurat_clusters", cols = dittoColors())
# Quality control plots to asses different matrices
VlnPlot(Seurat_object_cleaned, features = c("nFeature_RNA", "nCount_RNA", "percent.mt", "percent.ribo"), ncol = 2, pt.size = 0, group.by = "seurat_clusters")
```

#Save final object
```{r}
#Save final seurat object
saveRDS(Seurat_object_cleaned, file = "Dataset_P1_Dibella.rds") 
```





